<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ian Brettell" />

<<<<<<< HEAD
<meta name="date" content="2021-01-27" />

<title>Fst variation across human traits</title>

=======
<meta name="date" content="2021-02-03" />

<title>Fst variation across human traits</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<<<<<<< HEAD
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
=======
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
<<<<<<< HEAD
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
=======
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
<<<<<<< HEAD
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
=======
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
<<<<<<< HEAD
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
=======
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Fst variation across human traits</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Fst variation across human traits</h1>
<h4 class="author">Ian Brettell</h4>
<<<<<<< HEAD
<h4 class="date">2021-01-27</h4>
=======
<h4 class="date">2021-02-03</h4>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638

</div>


<<<<<<< HEAD
<div id="setup" class="section level1">
<h1><span class="header-section-number">1</span> Setup</h1>
=======
<div id="setup" class="section level1" number="1">
<h1 number="1"><span class="header-section-number">1</span> Setup</h1>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<ul>
<li><span style="color: #4f0943;">Working directory</span> on EBI Cluster: <code>/hps/research1/birney/users/ian/hmn_fst</code></li>
<li><span style="color: #4f0943;">GitHub repository</span>: <a href="https://github.com/brettellebi/human_traits_fst" class="uri">https://github.com/brettellebi/human_traits_fst</a></li>
</ul>
<<<<<<< HEAD
<div id="conda-env-on-cluster" class="section level2">
<h2><span class="header-section-number">1.1</span> <code>conda</code> env on cluster</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># Create env on cluster with mamba</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="ex">mamba</span> create -y \</span>
<span id="cb1-3"><a href="#cb1-3"></a>  -n fst_env_rhel \</span>
<span id="cb1-4"><a href="#cb1-4"></a>  -c bioconda gatk4</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="ex">conda</span> activate fst_env_rhel</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="ex">mamba</span> install bcftools plink2 r-base r-essentials r-tidyverse r-units libgdal r-sf</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co"># Export</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="ex">conda</span> env export \</span>
<span id="cb1-9"><a href="#cb1-9"></a>  --no-builds \</span>
<span id="cb1-10"><a href="#cb1-10"></a>  -f envs/fst_env_rhel.yml</span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="co"># Activate</span></span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="ex">conda</span> activate fst_env_rhel</span></code></pre></div>
</div>
<div id="renv" class="section level2">
<h2><span class="header-section-number">1.2</span> <code>renv</code></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># Export env (to renv.lock file)</span></span>
<span id="cb2-2"><a href="#cb2-2"></a>renv<span class="op">::</span><span class="kw">init</span>()</span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="co"># To install packages on new system, or &#39;activate&#39; the env: </span></span>
<span id="cb2-4"><a href="#cb2-4"></a>renv<span class="op">::</span><span class="kw">restore</span>()</span></code></pre></div>
</div>
<div id="source-libraries-functions-and-plotting-parameters" class="section level2">
<h2><span class="header-section-number">1.3</span> Source libraries, functions and plotting parameters</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">library</span>(here)</span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;code&quot;</span>, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;source.R&quot;</span>))</span></code></pre></div>
</div>
<div id="download-1gk-data" class="section level2">
<h2><span class="header-section-number">1.4</span> Download 1GK data</h2>
<div id="download-from-ftp" class="section level3">
<h3><span class="header-section-number">1.4.1</span> Download from FTP</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">wget</span> \</span>
<span id="cb4-2"><a href="#cb4-2"></a>  -r -p -k \</span>
<span id="cb4-3"><a href="#cb4-3"></a>  --no-parent \</span>
<span id="cb4-4"><a href="#cb4-4"></a>  -cut-dirs=5 \</span>
<span id="cb4-5"><a href="#cb4-5"></a>  ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/</span></code></pre></div>
</div>
<div id="put-filenames-into-list" class="section level3">
<h3><span class="header-section-number">1.4.2</span> Put filenames into list</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1"></a><span class="fu">find</span> vcfs/ftp.1000genomes.ebi.ac.uk/ALL.chr*.vcf.gz \</span>
<span id="cb5-2"><a href="#cb5-2"></a>  <span class="op">&gt;</span> human_traits_fst/data/20200205_vcfs.list</span></code></pre></div>
</div>
<div id="merge-vcfs" class="section level3">
<h3><span class="header-section-number">1.4.3</span> Merge VCFs</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1"></a><span class="ex">java</span> -jar /nfs/software/birney/picard-2.9.0/picard.jar MergeVcfs \</span>
<span id="cb6-2"><a href="#cb6-2"></a>  I=human_traits_fst/data/20200205_vcfs.list \</span>
<span id="cb6-3"><a href="#cb6-3"></a>  O=vcfs/1gk_all.vcf.gz</span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co"># Exception in thread &quot;main&quot; java.lang.IllegalArgumentException: The contig entries in input file /hps/research1/birney/users/ian/rac_hyp/vcfs/ftp.1000genomes.ebi.ac.uk/ALL.chrMT.phase3_callmom-v0_4.20130502.genotypes.vcf.gz are not compatible with the others.</span></span>
<span id="cb6-5"><a href="#cb6-5"></a></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co"># So remove that one from list above</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="fu">sed</span> -i <span class="st">&#39;/MT/d&#39;</span> human_traits_fst/data/20200205_vcfs.list</span>
<span id="cb6-8"><a href="#cb6-8"></a></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co"># run MergeVCFs again</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="ex">java</span> -jar /nfs/software/birney/picard-2.9.0/picard.jar MergeVcfs \</span>
<span id="cb6-11"><a href="#cb6-11"></a>  I=human_traits_fst/data/20200205_vcfs.list \</span>
<span id="cb6-12"><a href="#cb6-12"></a>  O=vcfs/1gk_all.vcf.gz</span>
<span id="cb6-13"><a href="#cb6-13"></a>  </span>
<span id="cb6-14"><a href="#cb6-14"></a><span class="co"># Exception in thread &quot;main&quot; java.lang.IllegalArgumentException: The contig entries in input file /hps/research1/birney/users/ian/rac_hyp/vcfs/ftp.1000genomes.ebi.ac.uk/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz are not compatible with the others.</span></span>
<span id="cb6-15"><a href="#cb6-15"></a><span class="fu">sed</span> -i <span class="st">&#39;/chrY/d&#39;</span> human_traits_fst/data/20200205_vcfs.list</span>
<span id="cb6-16"><a href="#cb6-16"></a></span>
<span id="cb6-17"><a href="#cb6-17"></a><span class="co"># run MergeVCFs again</span></span>
<span id="cb6-18"><a href="#cb6-18"></a><span class="ex">java</span> -jar /nfs/software/birney/picard-2.9.0/picard.jar MergeVcfs \</span>
<span id="cb6-19"><a href="#cb6-19"></a>  I=human_traits_fst/data/20200205_vcfs.list \</span>
<span id="cb6-20"><a href="#cb6-20"></a>  O=vcfs/1gk_all.vcf.gz</span>
<span id="cb6-21"><a href="#cb6-21"></a><span class="co"># SUCCESS</span></span></code></pre></div>
</div>
</div>
<div id="obtain-gwas-data-from-the-gwas-catalog-httpswww.ebi.ac.ukgwas" class="section level2">
<h2><span class="header-section-number">1.5</span> Obtain GWAS data from the GWAS Catalog <a href="https://www.ebi.ac.uk/gwas" class="uri">https://www.ebi.ac.uk/gwas</a></h2>
<div id="pull-data-for-each-trait" class="section level3">
<h3><span class="header-section-number">1.5.1</span> Pull data for each trait</h3>
<p>[<strong>NOTE</strong>]{colour = “red”}: Uncheck <code>Include child trait data</code> before downloading.</p>
<p>All documents downloaded via ‘Download Catalog data’ link, then collated and saved here: <code>data/20210122_gwas_catalog.xlsx</code></p>
<div id="height" class="section level4">
<h4><span class="header-section-number">1.5.1.1</span> Height</h4>
=======
<div id="conda-env-on-cluster" class="section level2" number="1.1">
<h2 number="1.1"><span class="header-section-number">1.1</span> <code>conda</code> env on cluster</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create env on cluster with mamba</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> create -y <span class="kw">\</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">-n</span> fst_env_rhel <span class="kw">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">-c</span> bioconda gatk4</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate fst_env_rhel</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> install bcftools plink2 r-base r-essentials r-tidyverse r-units libgdal r-sf</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Export</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> env export <span class="kw">\</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--no-builds</span> <span class="kw">\</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="ex">-f</span> envs/fst_env_rhel.yml</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Activate</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate fst_env_rhel</span></code></pre></div>
</div>
<div id="renv" class="section level2" number="1.2">
<h2 number="1.2"><span class="header-section-number">1.2</span> <code>renv</code></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export env (to renv.lock file)</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">init</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># To install packages on new system, or &#39;activate&#39; the env: </span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span></code></pre></div>
</div>
<div id="source-libraries-functions-and-plotting-parameters" class="section level2" number="1.3">
<h2 number="1.3"><span class="header-section-number">1.3</span> Source libraries, functions and plotting parameters</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;code&quot;</span>, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;source.R&quot;</span>))</span></code></pre></div>
</div>
<div id="download-1kg-data" class="section level2" number="1.4">
<h2 number="1.4"><span class="header-section-number">1.4</span> Download 1KG data</h2>
<div id="download-from-ftp" class="section level3" number="1.4.1">
<h3 number="1.4.1"><span class="header-section-number">1.4.1</span> Download from FTP</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="kw">\</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="ex">-r</span> -p -k <span class="kw">\</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--no-parent</span> <span class="kw">\</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">-cut-dirs</span>=5 <span class="kw">\</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="ex">ftp</span>://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/</span></code></pre></div>
</div>
<div id="put-filenames-into-list" class="section level3" number="1.4.2">
<h3 number="1.4.2"><span class="header-section-number">1.4.2</span> Put filenames into list</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> vcfs/ftp.1000genomes.ebi.ac.uk/ALL.chr*.vcf.gz <span class="kw">\</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="op">&gt;</span> <span class="ex">human_traits_fst/data/20200205_vcfs.list</span></span></code></pre></div>
</div>
<div id="merge-vcfs" class="section level3" number="1.4.3">
<h3 number="1.4.3"><span class="header-section-number">1.4.3</span> Merge VCFs</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> -jar /nfs/software/birney/picard-2.9.0/picard.jar MergeVcfs <span class="kw">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="va">I=</span>human_traits_fst/data/20200205_vcfs.list <span class="kw">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">O=</span>vcfs/1kg_all.vcf.gz</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Exception in thread &quot;main&quot; java.lang.IllegalArgumentException: The contig entries in input file /hps/research1/birney/users/ian/rac_hyp/vcfs/ftp.1000genomes.ebi.ac.uk/ALL.chrMT.phase3_callmom-v0_4.20130502.genotypes.vcf.gz are not compatible with the others.</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co"># So remove that one from list above</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> -i <span class="st">&#39;/MT/d&#39;</span> human_traits_fst/data/20200205_vcfs.list</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># run MergeVCFs again</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> -jar /nfs/software/birney/picard-2.9.0/picard.jar MergeVcfs <span class="kw">\</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="va">I=</span>human_traits_fst/data/20200205_vcfs.list <span class="kw">\</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="va">O=</span>vcfs/1kg_all.vcf.gz</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Exception in thread &quot;main&quot; java.lang.IllegalArgumentException: The contig entries in input file /hps/research1/birney/users/ian/rac_hyp/vcfs/ftp.1000genomes.ebi.ac.uk/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz are not compatible with the others.</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> -i <span class="st">&#39;/chrY/d&#39;</span> human_traits_fst/data/20200205_vcfs.list</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co"># run MergeVCFs again</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> -jar /nfs/software/birney/picard-2.9.0/picard.jar MergeVcfs <span class="kw">\</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="va">I=</span>human_traits_fst/data/20200205_vcfs.list <span class="kw">\</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="va">O=</span>vcfs/1kg_all.vcf.gz</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co"># SUCCESS</span></span></code></pre></div>
</div>
</div>
<div id="obtain-gwas-data-from-the-gwas-catalog-httpswww.ebi.ac.ukgwas" class="section level2" number="1.5">
<h2 number="1.5"><span class="header-section-number">1.5</span> Obtain GWAS data from the GWAS Catalog <a href="https://www.ebi.ac.uk/gwas" class="uri">https://www.ebi.ac.uk/gwas</a></h2>
<div id="pull-data-for-each-trait" class="section level3" number="1.5.1">
<h3 number="1.5.1"><span class="header-section-number">1.5.1</span> Pull data for each trait</h3>
<p><span style="color: red;"><strong>NOTE</strong></span>: Uncheck <code>Include child trait data</code> before downloading.</p>
<p>All documents downloaded via ‘Download Catalog data’ link, then collated and saved here: <code>data/20210122_gwas_catalog.xlsx</code></p>
<div id="height" class="section level4" number="1.5.1.1">
<h4 number="1.5.1.1"><span class="header-section-number">1.5.1.1</span> Height</h4>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<ul>
<li>height: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0004339" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0004339</a>
<ul>
<li><strong>4912 SNPs</strong> from <strong>51 studies</strong></li>
</ul></li>
</ul>
</div>
<<<<<<< HEAD
<div id="bmi" class="section level4">
<h4><span class="header-section-number">1.5.1.2</span> BMI</h4>
=======
<div id="bmi" class="section level4" number="1.5.1.2">
<h4 number="1.5.1.2"><span class="header-section-number">1.5.1.2</span> BMI</h4>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<ul>
<li>bmi: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0004340" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0004340</a>
<ul>
<li><strong>7573 SNPs</strong> from <strong>155 studies</strong></li>
</ul></li>
</ul>
</div>
<<<<<<< HEAD
<div id="educational-attainment" class="section level4">
<h4><span class="header-section-number">1.5.1.3</span> Educational attainment</h4>
=======
<div id="educational-attainment" class="section level4" number="1.5.1.3">
<h4 number="1.5.1.3"><span class="header-section-number">1.5.1.3</span> Educational attainment</h4>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<ul>
<li>self reported educational attainment: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0004784" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0004784</a>
<ul>
<li><strong>3989 SNPs</strong> from <strong>24 studies</strong></li>
</ul></li>
</ul>
</div>
<<<<<<< HEAD
<div id="intelligence" class="section level4">
<h4><span class="header-section-number">1.5.1.4</span> Intelligence</h4>
=======
<div id="intelligence" class="section level4" number="1.5.1.4">
<h4 number="1.5.1.4"><span class="header-section-number">1.5.1.4</span> Intelligence</h4>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<ul>
<li>intelligence: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0004337" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0004337</a>
<ul>
<li><strong>2967 SNPs</strong> from <strong>27 studies</strong></li>
</ul></li>
</ul>
</div>
<<<<<<< HEAD
<div id="ibd" class="section level4">
<h4><span class="header-section-number">1.5.1.5</span> IBD</h4>
=======
<div id="ibd" class="section level4" number="1.5.1.5">
<h4 number="1.5.1.5"><span class="header-section-number">1.5.1.5</span> IBD</h4>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<ul>
<li>inflammatory bowel disease: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0003767" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0003767</a>
<ul>
<li><strong>536 SNPs</strong> from <strong>34 studies</strong></li>
</ul></li>
</ul>
</div>
<<<<<<< HEAD
<div id="pigmentation" class="section level4">
<h4><span class="header-section-number">1.5.1.6</span> Pigmentation</h4>
=======
<div id="pigmentation" class="section level4" number="1.5.1.6">
<h4 number="1.5.1.6"><span class="header-section-number">1.5.1.6</span> Pigmentation</h4>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<ul>
<li>skin pigmentation: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0003784" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0003784</a>
<ul>
<li><strong>102 SNPs</strong> from <strong>6 studies</strong></li>
</ul></li>
<li>skin pigmentation measurement: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0007009" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0007009</a>
<ul>
<li><strong>233 SNPs</strong> from <strong>9 studies</strong></li>
</ul></li>
<li>eye color: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0003949" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0003949</a>
<ul>
<li><strong>77 SNPs</strong> from <strong>13 studies</strong></li>
</ul></li>
<li>eye colour measurement: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0009764" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0009764</a>
<ul>
<li><strong>202 SNPs</strong> from <strong>9 studies</strong></li>
</ul></li>
<li>hair color: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0003924" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0003924</a>
<ul>
<li><strong>424 SNPs</strong> from <strong>18 studies</strong></li>
</ul></li>
<li>hair colour measurement: <a href="https://www.ebi.ac.uk/gwas/efotraits/EFO_0007822" class="uri">https://www.ebi.ac.uk/gwas/efotraits/EFO_0007822</a>
<ul>
<li><strong>541 SNPs</strong> from <strong>6 studies</strong></li>
</ul></li>
</ul>
</div>
</div>
<<<<<<< HEAD
<div id="read-into-list" class="section level3">
<h3><span class="header-section-number">1.5.2</span> Read into list</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>file_in =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210122_gwas_catalog.xlsx&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="co"># Create vector of traits</span></span>
<span id="cb7-3"><a href="#cb7-3"></a>traits =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hei&quot;</span>, <span class="st">&quot;bmi&quot;</span>, <span class="st">&quot;edu&quot;</span>, <span class="st">&quot;int&quot;</span>, <span class="st">&quot;ibd&quot;</span>, <span class="st">&quot;pig&quot;</span>)</span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="kw">names</span>(traits) =<span class="st"> </span>traits</span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co"># Assign sheets to traits</span></span>
<span id="cb7-6"><a href="#cb7-6"></a>sheets &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">11</span>)</span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="kw">names</span>(sheets) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hei&quot;</span>, <span class="st">&quot;bmi&quot;</span>, <span class="st">&quot;edu&quot;</span>, <span class="st">&quot;int&quot;</span>, <span class="st">&quot;ibd&quot;</span>, <span class="kw">rep</span>(<span class="st">&quot;pig&quot;</span>, <span class="dv">6</span>))</span>
<span id="cb7-8"><a href="#cb7-8"></a></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co"># get sheets</span></span>
<span id="cb7-10"><a href="#cb7-10"></a>sheet_names &lt;-<span class="st"> </span>readxl<span class="op">::</span><span class="kw">excel_sheets</span>(file_in)</span>
<span id="cb7-11"><a href="#cb7-11"></a></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co"># Create function to read in data</span></span>
<span id="cb7-13"><a href="#cb7-13"></a>read_catalog_data &lt;-<span class="st"> </span><span class="cf">function</span>(path, target_sheet){</span>
<span id="cb7-14"><a href="#cb7-14"></a>  <span class="co"># Read in data</span></span>
<span id="cb7-15"><a href="#cb7-15"></a>  out =<span class="st"> </span>readxl<span class="op">::</span><span class="kw">read_xlsx</span>(path, <span class="dt">sheet =</span> target_sheet) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="dt">CHR =</span> CHR_ID, </span>
<span id="cb7-17"><a href="#cb7-17"></a>                  <span class="dt">POS =</span> CHR_POS, </span>
<span id="cb7-18"><a href="#cb7-18"></a>                  <span class="dt">SNP_AL =</span> <span class="st">`</span><span class="dt">STRONGEST SNP-RISK ALLELE</span><span class="st">`</span>, </span>
<span id="cb7-19"><a href="#cb7-19"></a>                  <span class="dt">P =</span> <span class="st">`</span><span class="dt">P-VALUE</span><span class="st">`</span>, </span>
<span id="cb7-20"><a href="#cb7-20"></a>                  <span class="dt">OR_OR_BETA =</span> <span class="st">`</span><span class="dt">OR or BETA</span><span class="st">`</span>, </span>
<span id="cb7-21"><a href="#cb7-21"></a>                  MAPPED_TRAIT,</span>
<span id="cb7-22"><a href="#cb7-22"></a>                  <span class="dt">STUDY =</span> <span class="st">`</span><span class="dt">STUDY ACCESSION</span><span class="st">`</span>,</span>
<span id="cb7-23"><a href="#cb7-23"></a>                  <span class="dt">SAMPLE =</span> <span class="st">`</span><span class="dt">INITIAL SAMPLE SIZE</span><span class="st">`</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-24"><a href="#cb7-24"></a><span class="st">    </span><span class="co"># Split SNP and risk allele into separate columns</span></span>
<span id="cb7-25"><a href="#cb7-25"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">TOP_SNP =</span> stringr<span class="op">::</span><span class="kw">str_split</span>(SNP_AL, <span class="st">&quot;-&quot;</span>, <span class="dt">simplify =</span> T)[, <span class="dv">1</span>],</span>
<span id="cb7-26"><a href="#cb7-26"></a>                  <span class="dt">RISK_ALLELE =</span> stringr<span class="op">::</span><span class="kw">str_split</span>(SNP_AL, <span class="st">&quot;-&quot;</span>, <span class="dt">simplify =</span> T)[, <span class="dv">2</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-27"><a href="#cb7-27"></a><span class="st">    </span><span class="co"># Reorder and select</span></span>
<span id="cb7-28"><a href="#cb7-28"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(CHR, POS, TOP_SNP, RISK_ALLELE, P, OR_OR_BETA, MAPPED_TRAIT, STUDY, SAMPLE)</span>
<span id="cb7-29"><a href="#cb7-29"></a>  <span class="co"># Change variables to specific types</span></span>
<span id="cb7-30"><a href="#cb7-30"></a>  out<span class="op">$</span>CHR &lt;-<span class="st"> </span><span class="kw">as.integer</span>(out<span class="op">$</span>CHR)</span>
<span id="cb7-31"><a href="#cb7-31"></a>  out<span class="op">$</span>POS &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(out<span class="op">$</span>POS)</span>
<span id="cb7-32"><a href="#cb7-32"></a>  out<span class="op">$</span>P &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(out<span class="op">$</span>P)</span>
<span id="cb7-33"><a href="#cb7-33"></a>  <span class="co"># return DF</span></span>
<span id="cb7-34"><a href="#cb7-34"></a>  <span class="kw">return</span>(out)</span>
<span id="cb7-35"><a href="#cb7-35"></a>}</span>
<span id="cb7-36"><a href="#cb7-36"></a></span>
<span id="cb7-37"><a href="#cb7-37"></a><span class="co"># Read in data</span></span>
<span id="cb7-38"><a href="#cb7-38"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb7-39"><a href="#cb7-39"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(traits, <span class="cf">function</span>(trait){</span>
<span id="cb7-40"><a href="#cb7-40"></a>  <span class="co"># set counter </span></span>
<span id="cb7-41"><a href="#cb7-41"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb7-42"><a href="#cb7-42"></a>  <span class="co"># set target file</span></span>
<span id="cb7-43"><a href="#cb7-43"></a>  target_file =<span class="st"> </span>file_in</span>
<span id="cb7-44"><a href="#cb7-44"></a>  <span class="co"># get target sheet</span></span>
<span id="cb7-45"><a href="#cb7-45"></a>  target_sheet =<span class="st"> </span>sheets[<span class="kw">names</span>(sheets) <span class="op">==</span><span class="st"> </span>trait]</span>
<span id="cb7-46"><a href="#cb7-46"></a>  <span class="kw">length</span>(target_sheet)</span>
<span id="cb7-47"><a href="#cb7-47"></a>  <span class="co"># read in pigmentation data from multiple sheets and bind into single DF</span></span>
<span id="cb7-48"><a href="#cb7-48"></a>  <span class="cf">if</span> (<span class="kw">length</span>(target_sheet) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>){</span>
<span id="cb7-49"><a href="#cb7-49"></a>    <span class="co"># loop over each sheet</span></span>
<span id="cb7-50"><a href="#cb7-50"></a>    df &lt;-<span class="st"> </span><span class="kw">lapply</span>(target_sheet, <span class="cf">function</span>(sheet){</span>
<span id="cb7-51"><a href="#cb7-51"></a>      out &lt;-<span class="st"> </span><span class="kw">read_catalog_data</span>(target_file,</span>
<span id="cb7-52"><a href="#cb7-52"></a>                               <span class="dt">target_sheet =</span> sheet)</span>
<span id="cb7-53"><a href="#cb7-53"></a>    })</span>
<span id="cb7-54"><a href="#cb7-54"></a>    <span class="co"># set name of each DF to name of sheet (replacing spaces with underscores)</span></span>
<span id="cb7-55"><a href="#cb7-55"></a>    <span class="kw">names</span>(df) =<span class="st"> </span>sheet_names[target_sheet] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-56"><a href="#cb7-56"></a><span class="st">      </span>stringr<span class="op">::</span><span class="kw">str_replace_all</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>)</span>
<span id="cb7-57"><a href="#cb7-57"></a>    <span class="co"># bind DFs into single DF</span></span>
<span id="cb7-58"><a href="#cb7-58"></a>    df &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(df, <span class="dt">.id =</span> <span class="st">&quot;PIG_PHENO&quot;</span>)</span>
<span id="cb7-59"><a href="#cb7-59"></a>  } </span>
<span id="cb7-60"><a href="#cb7-60"></a>  <span class="cf">else</span> {</span>
<span id="cb7-61"><a href="#cb7-61"></a>    <span class="co"># read in other data</span></span>
<span id="cb7-62"><a href="#cb7-62"></a>    df &lt;-<span class="st"> </span><span class="kw">read_catalog_data</span>(target_file,</span>
<span id="cb7-63"><a href="#cb7-63"></a>                            <span class="dt">target_sheet =</span> target_sheet)</span>
<span id="cb7-64"><a href="#cb7-64"></a>  }</span>
<span id="cb7-65"><a href="#cb7-65"></a>  <span class="co"># Set PHENO column</span></span>
<span id="cb7-66"><a href="#cb7-66"></a>  df<span class="op">$</span>PHENO &lt;-<span class="st"> </span><span class="kw">factor</span>(trait, <span class="dt">levels =</span> trait_levels)</span>
<span id="cb7-67"><a href="#cb7-67"></a>  <span class="co"># Recode PHENO</span></span>
<span id="cb7-68"><a href="#cb7-68"></a>  df<span class="op">$</span>PHENO =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">recode</span>(df<span class="op">$</span>PHENO, <span class="op">!!!</span>recode_vec)</span>
<span id="cb7-69"><a href="#cb7-69"></a>  <span class="co"># Create list</span></span>
<span id="cb7-70"><a href="#cb7-70"></a>  out =<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb7-71"><a href="#cb7-71"></a>  <span class="co"># Return DF as &quot;raw&quot;</span></span>
<span id="cb7-72"><a href="#cb7-72"></a>  out[[<span class="st">&quot;raw&quot;</span>]] =<span class="st"> </span>df</span>
<span id="cb7-73"><a href="#cb7-73"></a>  </span>
<span id="cb7-74"><a href="#cb7-74"></a>  <span class="kw">return</span>(out)</span>
<span id="cb7-75"><a href="#cb7-75"></a>})</span></code></pre></div>
<pre><code>## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs
## introduced by coercion</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in AB1130 / R1130C28: got &#39;1E-423&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L2384 / R2384C12: got &#39;2 x 9&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in M2384 / R2384C13: got &#39;190120954 x 70383416&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L3046 / R3046C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L3070 / R3070C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L4419 / R4419C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L4420 / R4420C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L4421 / R4421C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L4422 / R4422C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L4586 / R4586C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, :
## Expecting numeric in L6641 / R6641C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs
## introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs
## introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs
## introduced by coercion</code></pre>
<pre><code>## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced
## by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced
## by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced
## by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced
## by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced
## by coercion</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="co"># How many SNPs in raw data</span></span>
<span id="cb22-2"><a href="#cb22-2"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(x) <span class="kw">nrow</span>(x[[<span class="st">&quot;raw&quot;</span>]]))</span></code></pre></div>
=======
<div id="read-into-list" class="section level3" number="1.5.2">
<h3 number="1.5.2"><span class="header-section-number">1.5.2</span> Read into list</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>file_in <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210122_gwas_catalog.xlsx&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Create vector of traits</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;hei&quot;</span>, <span class="st">&quot;bmi&quot;</span>, <span class="st">&quot;edu&quot;</span>, <span class="st">&quot;int&quot;</span>, <span class="st">&quot;ibd&quot;</span>, <span class="st">&quot;pig&quot;</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(traits) <span class="ot">=</span> traits</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign sheets to traits</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>sheets <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sheets) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;hei&quot;</span>, <span class="st">&quot;bmi&quot;</span>, <span class="st">&quot;edu&quot;</span>, <span class="st">&quot;int&quot;</span>, <span class="st">&quot;ibd&quot;</span>, <span class="fu">rep</span>(<span class="st">&quot;pig&quot;</span>, <span class="dv">6</span>))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># get sheets</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>sheet_names <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">excel_sheets</span>(file_in)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Create function to read in data</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>read_catalog_data <span class="ot">&lt;-</span> <span class="cf">function</span>(path, target_sheet){</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read in data</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(path, <span class="at">sheet =</span> target_sheet) <span class="sc">%&gt;%</span> </span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">CHR =</span> CHR_ID, </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">POS =</span> CHR_POS, </span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">SNP_AL =</span> <span class="st">`</span><span class="at">STRONGEST SNP-RISK ALLELE</span><span class="st">`</span>, </span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">P =</span> <span class="st">`</span><span class="at">P-VALUE</span><span class="st">`</span>, </span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">OR_OR_BETA =</span> <span class="st">`</span><span class="at">OR or BETA</span><span class="st">`</span>, </span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>                  MAPPED_TRAIT,</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">STUDY =</span> <span class="st">`</span><span class="at">STUDY ACCESSION</span><span class="st">`</span>,</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">SAMPLE =</span> <span class="st">`</span><span class="at">INITIAL SAMPLE SIZE</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Split SNP and risk allele into separate columns</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">TOP_SNP =</span> stringr<span class="sc">::</span><span class="fu">str_split</span>(SNP_AL, <span class="st">&quot;-&quot;</span>, <span class="at">simplify =</span> T)[, <span class="dv">1</span>],</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>                  <span class="at">RISK_ALLELE =</span> stringr<span class="sc">::</span><span class="fu">str_split</span>(SNP_AL, <span class="st">&quot;-&quot;</span>, <span class="at">simplify =</span> T)[, <span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Reorder and select</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(CHR, POS, TOP_SNP, RISK_ALLELE, P, OR_OR_BETA, MAPPED_TRAIT, STUDY, SAMPLE)</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Change variables to specific types</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  out<span class="sc">$</span>CHR <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(out<span class="sc">$</span>CHR)</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>  out<span class="sc">$</span>POS <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(out<span class="sc">$</span>POS)</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>  out<span class="sc">$</span>P <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(out<span class="sc">$</span>P)</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># return DF</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in data</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(traits, <span class="cf">function</span>(trait){</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter </span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set target file</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>  target_file <span class="ot">=</span> file_in</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get target sheet</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>  target_sheet <span class="ot">=</span> sheets[<span class="fu">names</span>(sheets) <span class="sc">==</span> trait]</span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">length</span>(target_sheet)</span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read in pigmentation data from multiple sheets and bind into single DF</span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(target_sheet) <span class="sc">&gt;</span> <span class="dv">1</span>){</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>    <span class="co"># loop over each sheet</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">lapply</span>(target_sheet, <span class="cf">function</span>(sheet){</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>      out <span class="ot">&lt;-</span> <span class="fu">read_catalog_data</span>(target_file,</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>                               <span class="at">target_sheet =</span> sheet)</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set name of each DF to name of sheet (replacing spaces with underscores)</span></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(df) <span class="ot">=</span> sheet_names[target_sheet] <span class="sc">%&gt;%</span> </span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>      stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>)</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># bind DFs into single DF</span></span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(df, <span class="at">.id =</span> <span class="st">&quot;PIG_PHENO&quot;</span>)</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># read in other data</span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">read_catalog_data</span>(target_file,</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>                            <span class="at">target_sheet =</span> target_sheet)</span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set PHENO column</span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>PHENO <span class="ot">&lt;-</span> <span class="fu">factor</span>(trait, <span class="at">levels =</span> trait_levels)</span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Recode PHENO</span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>PHENO <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">recode</span>(df<span class="sc">$</span>PHENO, <span class="sc">!!!</span>recode_vec)</span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create list</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return DF as &quot;raw&quot;</span></span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>  out[[<span class="st">&quot;raw&quot;</span>]] <span class="ot">=</span> df</span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<pre><code>## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs introduced by coercion</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in AB1130 /
## R1130C28: got &#39;1E-423&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L2384 /
## R2384C12: got &#39;2 x 9&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in M2384 /
## R2384C13: got &#39;190120954 x 70383416&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L3046 /
## R3046C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L3070 /
## R3070C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L4419 /
## R4419C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L4420 /
## R4420C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L4421 /
## R4421C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L4422 /
## R4422C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L4586 /
## R4586C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_fun(path = enc2native(normalizePath(path)), sheet_i = sheet, : Expecting numeric in L6641 /
## R6641C12: got &#39;X&#39;</code></pre>
<pre><code>## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = target_sheet): NAs introduced by coercion</code></pre>
<pre><code>## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced by coercion

## Warning in read_catalog_data(target_file, target_sheet = sheet): NAs introduced by coercion</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># How many SNPs in raw data</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(x) <span class="fu">nrow</span>(x[[<span class="st">&quot;raw&quot;</span>]]))</span></code></pre></div>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<pre><code>## $hei
## [1] 4912
## 
## $bmi
## [1] 7573
## 
## $edu
## [1] 3989
## 
## $int
## [1] 2967
## 
## $ibd
## [1] 536
## 
## $pig
## [1] 1579</code></pre>
<<<<<<< HEAD
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="co"># Clean data</span></span>
<span id="cb24-2"><a href="#cb24-2"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb24-3"><a href="#cb24-3"></a>  df_clean =<span class="st"> </span>pheno[[<span class="st">&quot;raw&quot;</span>]]</span>
<span id="cb24-4"><a href="#cb24-4"></a>  <span class="co"># Remove rows with p-value of 0 (only 32 of them, associated with suntan)</span></span>
<span id="cb24-5"><a href="#cb24-5"></a>  df_clean =<span class="st"> </span>df_clean[df_clean<span class="op">$</span>P <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, ]</span>
<span id="cb24-6"><a href="#cb24-6"></a>  <span class="co"># Remove rows with NA in CHR</span></span>
<span id="cb24-7"><a href="#cb24-7"></a>  df_clean =<span class="st"> </span>df_clean[<span class="op">!</span><span class="kw">is.na</span>(df_clean<span class="op">$</span>CHR), ]</span>
<span id="cb24-8"><a href="#cb24-8"></a>  <span class="co"># Remove duplicates</span></span>
<span id="cb24-9"><a href="#cb24-9"></a>  <span class="co">## Find SNPs that are duplicated</span></span>
<span id="cb24-10"><a href="#cb24-10"></a>  dupes =<span class="st"> </span><span class="kw">unique</span>(df_clean<span class="op">$</span>TOP_SNP[<span class="kw">duplicated</span>(df_clean<span class="op">$</span>TOP_SNP) <span class="op">|</span><span class="st"> </span><span class="kw">duplicated</span>(df_clean<span class="op">$</span>TOP_SNP, <span class="dt">fromLast =</span> T)])</span>
<span id="cb24-11"><a href="#cb24-11"></a>  <span class="co">## Select only 1 SNP from each set of duplicated SNPs</span></span>
<span id="cb24-12"><a href="#cb24-12"></a>  dupe_filt =<span class="st"> </span><span class="kw">lapply</span>(dupes, <span class="cf">function</span>(dupe){</span>
<span id="cb24-13"><a href="#cb24-13"></a>    <span class="co"># Take the one with the lowest P-value</span></span>
<span id="cb24-14"><a href="#cb24-14"></a>    min_p =<span class="st"> </span><span class="kw">min</span>(df_clean<span class="op">$</span>P[df_clean<span class="op">$</span>TOP_SNP <span class="op">==</span><span class="st"> </span>dupe])</span>
<span id="cb24-15"><a href="#cb24-15"></a>    out =<span class="st"> </span>df_clean[df_clean<span class="op">$</span>TOP_SNP <span class="op">==</span><span class="st"> </span>dupe <span class="op">&amp;</span><span class="st"> </span>df_clean<span class="op">$</span>P <span class="op">==</span><span class="st"> </span>min_p, ]</span>
<span id="cb24-16"><a href="#cb24-16"></a>    <span class="co"># If there are still duplicates due to having equal P, take the one with the largest effect size</span></span>
<span id="cb24-17"><a href="#cb24-17"></a>    <span class="cf">if</span> (<span class="kw">nrow</span>(out) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span> ){</span>
<span id="cb24-18"><a href="#cb24-18"></a>      out =<span class="st"> </span>out[<span class="kw">which.max</span>(out<span class="op">$</span>OR_OR_BETA), ]</span>
<span id="cb24-19"><a href="#cb24-19"></a>    }</span>
<span id="cb24-20"><a href="#cb24-20"></a>    <span class="kw">return</span>(out)</span>
<span id="cb24-21"><a href="#cb24-21"></a>  })</span>
<span id="cb24-22"><a href="#cb24-22"></a>  <span class="co"># Bind list into DF</span></span>
<span id="cb24-23"><a href="#cb24-23"></a>  dupe_filt =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(dupe_filt)</span>
<span id="cb24-24"><a href="#cb24-24"></a>  <span class="co"># Extract non-duplicated rows</span></span>
<span id="cb24-25"><a href="#cb24-25"></a>  non_dupe =<span class="st"> </span>df_clean[<span class="op">!</span><span class="kw">duplicated</span>(df_clean<span class="op">$</span>TOP_SNP) <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">duplicated</span>(df_clean<span class="op">$</span>TOP_SNP, <span class="dt">fromLast =</span> T), ]</span>
<span id="cb24-26"><a href="#cb24-26"></a>  <span class="co"># Bind non-duplicated rows with filtered duplicates</span></span>
<span id="cb24-27"><a href="#cb24-27"></a>  df_clean =<span class="st"> </span><span class="kw">rbind</span>(non_dupe, dupe_filt) </span>
<span id="cb24-28"><a href="#cb24-28"></a>  <span class="co"># Add to list</span></span>
<span id="cb24-29"><a href="#cb24-29"></a>  pheno[[<span class="st">&quot;clean&quot;</span>]] =<span class="st"> </span>df_clean</span>
<span id="cb24-30"><a href="#cb24-30"></a>  </span>
<span id="cb24-31"><a href="#cb24-31"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb24-32"><a href="#cb24-32"></a>})</span>
<span id="cb24-33"><a href="#cb24-33"></a></span>
<span id="cb24-34"><a href="#cb24-34"></a><span class="co"># New SNP count</span></span>
<span id="cb24-35"><a href="#cb24-35"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(x) <span class="kw">nrow</span>(x[[<span class="st">&quot;clean&quot;</span>]]))</span></code></pre></div>
=======
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean data</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  df_clean <span class="ot">=</span> pheno[[<span class="st">&quot;raw&quot;</span>]]</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove rows with p-value of 0 (only 32 of them, associated with suntan)</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  df_clean <span class="ot">=</span> df_clean[df_clean<span class="sc">$</span>P <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove rows with NA in CHR</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  df_clean <span class="ot">=</span> df_clean[<span class="sc">!</span><span class="fu">is.na</span>(df_clean<span class="sc">$</span>CHR), ]</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove duplicates</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Find SNPs that are duplicated</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  dupes <span class="ot">=</span> <span class="fu">unique</span>(df_clean<span class="sc">$</span>TOP_SNP[<span class="fu">duplicated</span>(df_clean<span class="sc">$</span>TOP_SNP) <span class="sc">|</span> <span class="fu">duplicated</span>(df_clean<span class="sc">$</span>TOP_SNP, <span class="at">fromLast =</span> T)])</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Select only 1 SNP from each set of duplicated SNPs</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  dupe_filt <span class="ot">=</span> <span class="fu">lapply</span>(dupes, <span class="cf">function</span>(dupe){</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Take the one with the lowest P-value</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>    min_p <span class="ot">=</span> <span class="fu">min</span>(df_clean<span class="sc">$</span>P[df_clean<span class="sc">$</span>TOP_SNP <span class="sc">==</span> dupe])</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>    out <span class="ot">=</span> df_clean[df_clean<span class="sc">$</span>TOP_SNP <span class="sc">==</span> dupe <span class="sc">&amp;</span> df_clean<span class="sc">$</span>P <span class="sc">==</span> min_p, ]</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If there are still duplicates due to having equal P, take the one with the largest effect size</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(out) <span class="sc">&gt;</span> <span class="dv">1</span> ){</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>      out <span class="ot">=</span> out[<span class="fu">which.max</span>(out<span class="sc">$</span>OR_OR_BETA), ]</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(out)</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Bind list into DF</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  dupe_filt <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(dupe_filt)</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract non-duplicated rows</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>  non_dupe <span class="ot">=</span> df_clean[<span class="sc">!</span><span class="fu">duplicated</span>(df_clean<span class="sc">$</span>TOP_SNP) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">duplicated</span>(df_clean<span class="sc">$</span>TOP_SNP, <span class="at">fromLast =</span> T), ]</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Bind non-duplicated rows with filtered duplicates</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>  df_clean <span class="ot">=</span> <span class="fu">rbind</span>(non_dupe, dupe_filt) </span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add to list</span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;clean&quot;</span>]] <span class="ot">=</span> df_clean</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a><span class="co"># New SNP count</span></span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(x) <span class="fu">nrow</span>(x[[<span class="st">&quot;clean&quot;</span>]]))</span></code></pre></div>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<pre><code>## $hei
## [1] 4359
## 
## $bmi
## [1] 4184
## 
## $edu
## [1] 3021
## 
## $int
## [1] 2544
## 
## $ibd
## [1] 389
## 
## $pig
## [1] 1048</code></pre>
<<<<<<< HEAD
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb26-2"><a href="#cb26-2"></a>  knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">head</span>(pheno[[<span class="st">&quot;clean&quot;</span>]]))</span>
<span id="cb26-3"><a href="#cb26-3"></a>})</span></code></pre></div>
=======
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(pheno[[<span class="st">&quot;clean&quot;</span>]]))</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<p>$hei</p>
<table>
<colgroup>
<col width="3%" />
<col width="8%" />
<col width="8%" />
<col width="9%" />
<col width="4%" />
<col width="8%" />
<col width="10%" />
<col width="8%" />
<col width="32%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="left">TOP_SNP</th>
<th align="left">RISK_ALLELE</th>
<th align="right">P</th>
<th align="right">OR_OR_BETA</th>
<th align="left">MAPPED_TRAIT</th>
<th align="left">STUDY</th>
<th align="left">SAMPLE</th>
<th align="left">PHENO</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">10</td>
<td align="right">31008555</td>
<td align="left">rs2797910</td>
<td align="left">G</td>
<td align="right">6e-06</td>
<td align="right">0.01036</td>
<td align="left">body height</td>
<td align="left">GCST010059</td>
<td align="left">up to 6,835 Korean ancestry individuals</td>
<td align="left">Height</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">68732085</td>
<td align="left">rs1156293</td>
<td align="left">G</td>
<td align="right">1e-06</td>
<td align="right">0.02157</td>
<td align="left">body height</td>
<td align="left">GCST010059</td>
<td align="left">up to 6,835 Korean ancestry individuals</td>
<td align="left">Height</td>
</tr>
<tr class="odd">
<td align="right">10</td>
<td align="right">98257696</td>
<td align="left">rs1983864</td>
<td align="left">G</td>
<td align="right">6e-06</td>
<td align="right">NA</td>
<td align="left">body height</td>
<td align="left">GCST005908</td>
<td align="left">253,288 European ancestry individuals</td>
<td align="left">Height</td>
</tr>
<tr class="even">
<td align="right">11</td>
<td align="right">117103213</td>
<td align="left">rs12269901</td>
<td align="left">C</td>
<td align="right">2e-06</td>
<td align="right">NA</td>
<td align="left">body height</td>
<td align="left">GCST005908</td>
<td align="left">253,288 European ancestry individuals</td>
<td align="left">Height</td>
</tr>
<tr class="odd">
<td align="right">11</td>
<td align="right">18624296</td>
<td align="left">rs11024739</td>
<td align="left">A</td>
<td align="right">9e-06</td>
<td align="right">NA</td>
<td align="left">body height</td>
<td align="left">GCST005908</td>
<td align="left">253,288 European ancestry individuals</td>
<td align="left">Height</td>
</tr>
<tr class="even">
<td align="right">11</td>
<td align="right">8231306</td>
<td align="left">rs110419</td>
<td align="left">G</td>
<td align="right">3e-07</td>
<td align="right">NA</td>
<td align="left">body height</td>
<td align="left">GCST005908</td>
<td align="left">253,288 European ancestry individuals</td>
<td align="left">Height</td>
</tr>
</tbody>
</table>
<p>$bmi</p>
<table style="width:100%;">
<colgroup>
<col width="1%" />
<col width="4%" />
<col width="5%" />
<col width="5%" />
<col width="2%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="60%" />
<col width="2%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="left">TOP_SNP</th>
<th align="left">RISK_ALLELE</th>
<th align="right">P</th>
<th align="right">OR_OR_BETA</th>
<th align="left">MAPPED_TRAIT</th>
<th align="left">STUDY</th>
<th align="left">SAMPLE</th>
<th align="left">PHENO</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">5</td>
<td align="right">79549808</td>
<td align="left">rs6899277</td>
<td align="left">G</td>
<td align="right">2e-06</td>
<td align="right">0.70</td>
<td align="left">body mass index</td>
<td align="left">GCST003647</td>
<td align="left">1,235 Hispanic individuals, 706 Asian ancestry individuals, 1,549 African American individuals, 2,395 European ancestry individuals</td>
<td align="left">BMI</td>
</tr>
<tr class="even">
<td align="right">5</td>
<td align="right">116877085</td>
<td align="left">rs6866721</td>
<td align="left">C</td>
<td align="right">1e-07</td>
<td align="right">0.76</td>
<td align="left">body mass index</td>
<td align="left">GCST003647</td>
<td align="left">1,235 Hispanic individuals, 706 Asian ancestry individuals, 1,549 African American individuals, 2,395 European ancestry individuals</td>
<td align="left">BMI</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="right">84154244</td>
<td align="left">rs2497155</td>
<td align="left">T</td>
<td align="right">4e-06</td>
<td align="right">1.05</td>
<td align="left">body mass index</td>
<td align="left">GCST003647</td>
<td align="left">1,235 Hispanic individuals, 706 Asian ancestry individuals, 1,549 African American individuals, 2,395 European ancestry individuals</td>
<td align="left">BMI</td>
</tr>
<tr class="even">
<td align="right">11</td>
<td align="right">17737040</td>
<td align="left">rs7926805</td>
<td align="left">C</td>
<td align="right">1e-06</td>
<td align="right">0.80</td>
<td align="left">body mass index</td>
<td align="left">GCST003647</td>
<td align="left">1,235 Hispanic individuals, 706 Asian ancestry individuals, 1,549 African American individuals, 2,395 European ancestry individuals</td>
<td align="left">BMI</td>
</tr>
<tr class="odd">
<td align="right">7</td>
<td align="right">137720035</td>
<td align="left">rs10234581</td>
<td align="left">C</td>
<td align="right">2e-06</td>
<td align="right">NA</td>
<td align="left">body mass index</td>
<td align="left">GCST003248</td>
<td align="left">40 Korean ancestry low BMI individuals, 37 Korean ancestry high BMI individuals</td>
<td align="left">BMI</td>
</tr>
<tr class="even">
<td align="right">11</td>
<td align="right">56539377</td>
<td align="left">rs2187511</td>
<td align="left">G</td>
<td align="right">1e-06</td>
<td align="right">NA</td>
<td align="left">body mass index</td>
<td align="left">GCST003248</td>
<td align="left">40 Korean ancestry low BMI individuals, 37 Korean ancestry high BMI individuals</td>
<td align="left">BMI</td>
</tr>
</tbody>
</table>
<p>$edu</p>
<table style="width:100%;">
<colgroup>
<col width="2%" />
<col width="5%" />
<col width="6%" />
<col width="6%" />
<col width="1%" />
<col width="6%" />
<col width="21%" />
<col width="6%" />
<col width="30%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="left">TOP_SNP</th>
<th align="left">RISK_ALLELE</th>
<th align="right">P</th>
<th align="right">OR_OR_BETA</th>
<th align="left">MAPPED_TRAIT</th>
<th align="left">STUDY</th>
<th align="left">SAMPLE</th>
<th align="left">PHENO</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">18</td>
<td align="right">79815871</td>
<td align="left">rs34945223</td>
<td align="left">?</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="left">self reported educational attainment</td>
<td align="left">GCST007037</td>
<td align="left">approximately 455,000 European ancestry individuals</td>
<td align="left">Educational attainment</td>
</tr>
<tr class="even">
<td align="right">19</td>
<td align="right">4474728</td>
<td align="left">rs76608582</td>
<td align="left">?</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="left">self reported educational attainment</td>
<td align="left">GCST007037</td>
<td align="left">approximately 455,000 European ancestry individuals</td>
<td align="left">Educational attainment</td>
</tr>
<tr class="odd">
<td align="right">19</td>
<td align="right">18224905</td>
<td align="left">rs4808766</td>
<td align="left">?</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="left">self reported educational attainment</td>
<td align="left">GCST007037</td>
<td align="left">approximately 455,000 European ancestry individuals</td>
<td align="left">Educational attainment</td>
</tr>
<tr class="even">
<td align="right">19</td>
<td align="right">30256794</td>
<td align="left">rs55800473</td>
<td align="left">?</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="left">self reported educational attainment</td>
<td align="left">GCST007037</td>
<td align="left">approximately 455,000 European ancestry individuals</td>
<td align="left">Educational attainment</td>
</tr>
<tr class="odd">
<td align="right">19</td>
<td align="right">35761593</td>
<td align="left">rs807478</td>
<td align="left">?</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="left">self reported educational attainment</td>
<td align="left">GCST007037</td>
<td align="left">approximately 455,000 European ancestry individuals</td>
<td align="left">Educational attainment</td>
</tr>
<tr class="even">
<td align="right">20</td>
<td align="right">38884055</td>
<td align="left">rs208795</td>
<td align="left">?</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="left">self reported educational attainment</td>
<td align="left">GCST007037</td>
<td align="left">approximately 455,000 European ancestry individuals</td>
<td align="left">Educational attainment</td>
</tr>
</tbody>
</table>
<p>$int</p>
<table>
<colgroup>
<col width="2%" />
<col width="6%" />
<col width="8%" />
<col width="8%" />
<col width="2%" />
<col width="7%" />
<col width="8%" />
<col width="7%" />
<col width="39%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="left">TOP_SNP</th>
<th align="left">RISK_ALLELE</th>
<th align="right">P</th>
<th align="right">OR_OR_BETA</th>
<th align="left">MAPPED_TRAIT</th>
<th align="left">STUDY</th>
<th align="left">SAMPLE</th>
<th align="left">PHENO</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">2</td>
<td align="right">156027428</td>
<td align="left">rs72904190</td>
<td align="left">A</td>
<td align="right">0</td>
<td align="right">0.0284595</td>
<td align="left">intelligence</td>
<td align="left">GCST005316</td>
<td align="left">120,934 British ancestry individuals, 127,548 individuals</td>
<td align="left">Intelligence</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">156055088</td>
<td align="left">rs72906064</td>
<td align="left">T</td>
<td align="right">0</td>
<td align="right">0.0284257</td>
<td align="left">intelligence</td>
<td align="left">GCST005316</td>
<td align="left">120,934 British ancestry individuals, 127,548 individuals</td>
<td align="left">Intelligence</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">156667523</td>
<td align="left">rs13023735</td>
<td align="left">A</td>
<td align="right">0</td>
<td align="right">0.0180964</td>
<td align="left">intelligence</td>
<td align="left">GCST005316</td>
<td align="left">120,934 British ancestry individuals, 127,548 individuals</td>
<td align="left">Intelligence</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">160506319</td>
<td align="left">rs62175972</td>
<td align="left">T</td>
<td align="right">0</td>
<td align="right">0.0467504</td>
<td align="left">intelligence</td>
<td align="left">GCST005316</td>
<td align="left">120,934 British ancestry individuals, 127,548 individuals</td>
<td align="left">Intelligence</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">161045556</td>
<td align="left">rs10930011</td>
<td align="left">A</td>
<td align="right">0</td>
<td align="right">0.0168607</td>
<td align="left">intelligence</td>
<td align="left">GCST005316</td>
<td align="left">120,934 British ancestry individuals, 127,548 individuals</td>
<td align="left">Intelligence</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">160955841</td>
<td align="left">rs114952970</td>
<td align="left">T</td>
<td align="right">0</td>
<td align="right">0.0512996</td>
<td align="left">intelligence</td>
<td align="left">GCST005316</td>
<td align="left">120,934 British ancestry individuals, 127,548 individuals</td>
<td align="left">Intelligence</td>
</tr>
</tbody>
</table>
<p>$ibd</p>
<table>
<colgroup>
<col width="2%" />
<col width="6%" />
<col width="6%" />
<col width="7%" />
<col width="3%" />
<col width="6%" />
<col width="17%" />
<col width="6%" />
<col width="37%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="left">TOP_SNP</th>
<th align="left">RISK_ALLELE</th>
<th align="right">P</th>
<th align="right">OR_OR_BETA</th>
<th align="left">MAPPED_TRAIT</th>
<th align="left">STUDY</th>
<th align="left">SAMPLE</th>
<th align="left">PHENO</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">154962487</td>
<td align="left">rs3766920</td>
<td align="left">A</td>
<td align="right">0e+00</td>
<td align="right">1.35</td>
<td align="left">inflammatory bowel disease</td>
<td align="left">GCST003602</td>
<td align="left">1,505 Korean ancestry cases, 4,041 Korean ancestry controls</td>
<td align="left">IBD</td>
</tr>
<tr class="even">
<td align="right">16</td>
<td align="right">80751551</td>
<td align="left">rs16953946</td>
<td align="left">C</td>
<td align="right">0e+00</td>
<td align="right">1.23</td>
<td align="left">inflammatory bowel disease</td>
<td align="left">GCST003602</td>
<td align="left">1,505 Korean ancestry cases, 4,041 Korean ancestry controls</td>
<td align="left">IBD</td>
</tr>
<tr class="odd">
<td align="right">22</td>
<td align="right">36912743</td>
<td align="left">rs4821558</td>
<td align="left">C</td>
<td align="right">8e-07</td>
<td align="right">1.16</td>
<td align="left">inflammatory bowel disease</td>
<td align="left">GCST003602</td>
<td align="left">1,505 Korean ancestry cases, 4,041 Korean ancestry controls</td>
<td align="left">IBD</td>
</tr>
<tr class="even">
<td align="right">9</td>
<td align="right">76302462</td>
<td align="left">rs11788518</td>
<td align="left">T</td>
<td align="right">7e-06</td>
<td align="right">1.39</td>
<td align="left">inflammatory bowel disease</td>
<td align="left">GCST003602</td>
<td align="left">1,505 Korean ancestry cases, 4,041 Korean ancestry controls</td>
<td align="left">IBD</td>
</tr>
<tr class="odd">
<td align="right">9</td>
<td align="right">21966222</td>
<td align="left">rs3731257</td>
<td align="left">G</td>
<td align="right">0e+00</td>
<td align="right">1.17</td>
<td align="left">inflammatory bowel disease</td>
<td align="left">GCST003602</td>
<td align="left">1,505 Korean ancestry cases, 4,041 Korean ancestry controls</td>
<td align="left">IBD</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="right">116919430</td>
<td align="left">rs1321366</td>
<td align="left">A</td>
<td align="right">5e-06</td>
<td align="right">1.15</td>
<td align="left">inflammatory bowel disease</td>
<td align="left">GCST003602</td>
<td align="left">1,505 Korean ancestry cases, 4,041 Korean ancestry controls</td>
<td align="left">IBD</td>
</tr>
</tbody>
</table>
<p>$pig</p>
<table style="width:100%;">
<colgroup>
<col width="12%" />
<col width="2%" />
<col width="6%" />
<col width="8%" />
<col width="8%" />
<col width="4%" />
<col width="7%" />
<col width="12%" />
<col width="7%" />
<col width="20%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">PIG_PHENO</th>
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="left">TOP_SNP</th>
<th align="left">RISK_ALLELE</th>
<th align="right">P</th>
<th align="right">OR_OR_BETA</th>
<th align="left">MAPPED_TRAIT</th>
<th align="left">STUDY</th>
<th align="left">SAMPLE</th>
<th align="left">PHENO</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">skin_pigmentation</td>
<td align="right">3</td>
<td align="right">187681148</td>
<td align="left">rs79592764</td>
<td align="left">T</td>
<td align="right">4e-07</td>
<td align="right">5.23</td>
<td align="left">skin pigmentation</td>
<td align="left">GCST004219</td>
<td align="left">285 Puerto Rican individuals</td>
<td align="left">Pigmentation</td>
</tr>
<tr class="even">
<td align="left">skin_pigmentation</td>
<td align="right">10</td>
<td align="right">13564490</td>
<td align="left">rs6602666</td>
<td align="left">G</td>
<td align="right">0e+00</td>
<td align="right">4.03</td>
<td align="left">skin pigmentation</td>
<td align="left">GCST004219</td>
<td align="left">285 Puerto Rican individuals</td>
<td align="left">Pigmentation</td>
</tr>
<tr class="odd">
<td align="left">skin_pigmentation</td>
<td align="right">15</td>
<td align="right">50016753</td>
<td align="left">rs8033655</td>
<td align="left">G</td>
<td align="right">2e-07</td>
<td align="right">2.29</td>
<td align="left">skin pigmentation</td>
<td align="left">GCST004219</td>
<td align="left">285 Puerto Rican individuals</td>
<td align="left">Pigmentation</td>
</tr>
<tr class="even">
<td align="left">skin_pigmentation</td>
<td align="right">20</td>
<td align="right">34116821</td>
<td align="left">rs6142102</td>
<td align="left">G</td>
<td align="right">2e-07</td>
<td align="right">2.21</td>
<td align="left">skin pigmentation</td>
<td align="left">GCST004219</td>
<td align="left">285 Puerto Rican individuals</td>
<td align="left">Pigmentation</td>
</tr>
<tr class="odd">
<td align="left">skin_pigmentation</td>
<td align="right">1</td>
<td align="right">188031112</td>
<td align="left">rs77443641</td>
<td align="left">A</td>
<td align="right">3e-06</td>
<td align="right">10.94</td>
<td align="left">skin pigmentation</td>
<td align="left">GCST004219</td>
<td align="left">285 Puerto Rican individuals</td>
<td align="left">Pigmentation</td>
</tr>
<tr class="even">
<td align="left">skin_pigmentation</td>
<td align="right">1</td>
<td align="right">196973530</td>
<td align="left">rs115105970</td>
<td align="left">T</td>
<td align="right">2e-06</td>
<td align="right">6.84</td>
<td align="left">skin pigmentation</td>
<td align="left">GCST004219</td>
<td align="left">285 Puerto Rican individuals</td>
<td align="left">Pigmentation</td>
</tr>
</tbody>
</table>
<p>Get unique mapped traits</p>
<<<<<<< HEAD
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="kw">unique</span>(pheno<span class="op">$</span>clean<span class="op">$</span>MAPPED_TRAIT)</span>
<span id="cb27-3"><a href="#cb27-3"></a>})</span></code></pre></div>
=======
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>(pheno<span class="sc">$</span>clean<span class="sc">$</span>MAPPED_TRAIT)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<pre><code>## $hei
## [1] &quot;body height&quot;                                                     
## [2] &quot;diastolic blood pressure, body height&quot;                           
## [3] &quot;body weights and measures, body height&quot;                          
## [4] &quot;anthropometric measurement, body height&quot;                         
## [5] &quot;pericardial adipose tissue measurement, body weight, body height&quot;
## 
## $bmi
##  [1] &quot;body mass index&quot;                                                                                                                                                                                                                                                                                                                                        
##  [2] &quot;systolic blood pressure, body mass index&quot;                                                                                                                                                                                                                                                                                                               
##  [3] &quot;body mass index, obesity&quot;                                                                                                                                                                                                                                                                                                                               
##  [4] &quot;response to drug, body mass index&quot;                                                                                                                                                                                                                                                                                                                      
##  [5] &quot;body mass index, body weights and measures&quot;                                                                                                                                                                                                                                                                                                             
##  [6] &quot;body mass index, fasting blood insulin measurement&quot;                                                                                                                                                                                                                                                                                                     
##  [7] &quot;alcohol consumption measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                       
##  [8] &quot;physical activity measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                         
##  [9] &quot;smoking behaviour measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                         
## [10] &quot;waist-hip ratio, body mass index&quot;                                                                                                                                                                                                                                                                                                                       
## [11] &quot;adult onset asthma, body mass index&quot;                                                                                                                                                                                                                                                                                                                    
## [12] &quot;multiple sclerosis, body mass index&quot;                                                                                                                                                                                                                                                                                                                    
## [13] &quot;anthropometric measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                            
## [14] &quot;visceral adipose tissue measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                   
## [15] &quot;body mass index, premature birth, parental genotype effect measurement&quot;                                                                                                                                                                                                                                                                                 
## [16] &quot;bipolar disorder, body mass index&quot;                                                                                                                                                                                                                                                                                                                      
## [17] &quot;total cholesterol measurement, diastolic blood pressure, triglyceride measurement, systolic blood pressure, hematocrit, ventricular rate measurement, glucose measurement, body mass index, high density lipoprotein cholesterol measurement&quot;                                                                                                           
## [18] &quot;Barrett&#39;s esophagus, body mass index&quot;                                                                                                                                                                                                                                                                                                                   
## [19] &quot;body mass index, asthma&quot;                                                                                                                                                                                                                                                                                                                                
## [20] &quot;insulin sensitivity measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                       
## [21] &quot;body mass index, visceral:subcutaneous adipose tissue ratio&quot;                                                                                                                                                                                                                                                                                            
## [22] &quot;total cholesterol measurement, hematocrit, stroke, ventricular rate measurement, body mass index, atrial fibrillation, high density lipoprotein cholesterol measurement, coronary artery disease, cancer, diastolic blood pressure, triglyceride measurement, systolic blood pressure, heart failure, diabetes mellitus, glucose measurement, mortality&quot;
## [23] &quot;fibrinogen measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                                
## [24] &quot;urate measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                                     
## [25] &quot;body mass index, fasting blood glucose measurement&quot;                                                                                                                                                                                                                                                                                                     
## [26] &quot;body mass index, age at assessment&quot;                                                                                                                                                                                                                                                                                                                     
## [27] &quot;smoking behavior, body mass index&quot;                                                                                                                                                                                                                                                                                                                      
## [28] &quot;diet measurement, body mass index&quot;                                                                                                                                                                                                                                                                                                                      
## [29] &quot;sex interaction measurement, body mass index, age at assessment&quot;                                                                                                                                                                                                                                                                                        
## [30] &quot;physical activity, body mass index&quot;                                                                                                                                                                                                                                                                                                                     
## 
## $edu
## [1] &quot;self reported educational attainment&quot;                              
## [2] &quot;intelligence, self reported educational attainment&quot;                
## [3] &quot;autism spectrum disorder, self reported educational attainment&quot;    
## [4] &quot;schizophrenia, intelligence, self reported educational attainment&quot; 
## [5] &quot;self reported educational attainment, refractive error measurement&quot;
## [6] &quot;systolic blood pressure, self reported educational attainment&quot;     
## [7] &quot;diastolic blood pressure, self reported educational attainment&quot;    
## [8] &quot;mean arterial pressure, self reported educational attainment&quot;      
## [9] &quot;pulse pressure measurement, self reported educational attainment&quot;  
## 
## $int
## [1] &quot;intelligence&quot;                                                     
## [2] &quot;cesarean section, intelligence&quot;                                   
## [3] &quot;intelligence, specific language impairment, dyslexia&quot;             
## [4] &quot;intelligence, self reported educational attainment&quot;               
## [5] &quot;schizophrenia, intelligence&quot;                                      
## [6] &quot;schizophrenia, intelligence, self reported educational attainment&quot;
## 
## $ibd
##  [1] &quot;inflammatory bowel disease&quot;                                                                           
##  [2] &quot;inflammatory bowel disease, spine bone mineral density&quot;                                               
##  [3] &quot;leukopenia, response to thiopurine, inflammatory bowel disease&quot;                                       
##  [4] &quot;inflammatory bowel disease, femoral neck bone mineral density&quot;                                        
##  [5] &quot;pyoderma gangrenosum, inflammatory bowel disease&quot;                                                     
##  [6] &quot;Erythema nodosum, inflammatory bowel disease&quot;                                                         
##  [7] &quot;myelosuppression, response to thiopurine, inflammatory bowel disease&quot;                                 
##  [8] &quot;Alopecia, response to thiopurine, inflammatory bowel disease&quot;                                         
##  [9] &quot;pancreatitis, response to thiopurine, inflammatory bowel disease&quot;                                     
## [10] &quot;response to thiopurine, thiopurine immunosuppressant-induced pancreatitis, inflammatory bowel disease&quot;
## [11] &quot;digestive system disease, response to thiopurine, inflammatory bowel disease&quot;                         
## 
## $pig
## [1] &quot;skin pigmentation&quot;                                   
## [2] &quot;freckles, skin pigmentation&quot;                         
## [3] &quot;skin pigmentation measurement&quot;                       
## [4] &quot;eye color&quot;                                           
## [5] &quot;eye colour measurement&quot;                              
## [6] &quot;hair color&quot;                                          
## [7] &quot;cutaneous melanoma, hair color&quot;                      
## [8] &quot;hair colour measurement&quot;                             
## [9] &quot;hair colour measurement, hair morphology measurement&quot;</code></pre>
<p>Are any of the OR_OR_BETAs negative?</p>
<<<<<<< HEAD
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="kw">any</span>(<span class="kw">unlist</span>(<span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno) {</span>
<span id="cb29-2"><a href="#cb29-2"></a>  <span class="kw">any</span>(pheno<span class="op">$</span>clean<span class="op">$</span>OR_OR_BETA <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>,<span class="dt">na.rm =</span> T)</span>
<span id="cb29-3"><a href="#cb29-3"></a>})))</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>This must means that the <code>RISK_ALLELE</code> always affects the trait positively.</p>
<p>But for what proportion of SNPs is the <code>RISK_ALLELE</code> not stated?</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno) {</span>
<span id="cb31-2"><a href="#cb31-2"></a>  <span class="kw">length</span>(<span class="kw">which</span>(pheno<span class="op">$</span>clean<span class="op">$</span>RISK_ALLELE <span class="op">==</span><span class="st"> &quot;?&quot;</span>)) <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(pheno<span class="op">$</span>clean)</span>
<span id="cb31-3"><a href="#cb31-3"></a>})</span></code></pre></div>
=======
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">any</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno) {</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">any</span>(pheno<span class="sc">$</span>clean<span class="sc">$</span>OR_OR_BETA <span class="sc">&lt;</span> <span class="dv">0</span>,<span class="at">na.rm =</span> T)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>})))</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>This must means that the <code>RISK_ALLELE</code> always affects the trait positively.</p>
<p>But for what proportion of SNPs is the <code>RISK_ALLELE</code> not stated?</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno) {</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">length</span>(<span class="fu">which</span>(pheno<span class="sc">$</span>clean<span class="sc">$</span>RISK_ALLELE <span class="sc">==</span> <span class="st">&quot;?&quot;</span>)) <span class="sc">/</span> <span class="fu">nrow</span>(pheno<span class="sc">$</span>clean)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<pre><code>## $hei
## [1] 0.7065841
## 
## $bmi
## [1] 0.5023901
## 
## $edu
## [1] 0.1741145
## 
## $int
## [1] 0.4382862
## 
## $ibd
## [1] 0.3213368
## 
## $pig
## [1] 0.5314885</code></pre>
<p><span style="color: red;"><strong>NOTE</strong></span>: In cases where <code>RISK_ALLELE</code> isn’t provided, treat the <code>ALT</code> allele as <code>RISK_ALLELE</code>.</p>
</div>
<<<<<<< HEAD
<div id="generate-manhattan-plots" class="section level3">
<h3><span class="header-section-number">1.5.3</span> Generate Manhattan plots</h3>
<p>Plot</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>counter =<span class="st"> </span><span class="dv">0</span></span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno_df){</span>
<span id="cb33-3"><a href="#cb33-3"></a>  <span class="co"># set counter</span></span>
<span id="cb33-4"><a href="#cb33-4"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb33-5"><a href="#cb33-5"></a>  df =<span class="st"> </span>pheno_df[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb33-6"><a href="#cb33-6"></a>  trait =<span class="st"> </span><span class="kw">unique</span>(df<span class="op">$</span>PHENO)</span>
<span id="cb33-7"><a href="#cb33-7"></a>  <span class="co"># Get title</span></span>
<span id="cb33-8"><a href="#cb33-8"></a>  title &lt;-<span class="st"> </span><span class="kw">paste</span>(trait, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, <span class="st">&quot;SNP count:&quot;</span>, <span class="kw">nrow</span>(df))</span>
<span id="cb33-9"><a href="#cb33-9"></a>  <span class="co"># Plot</span></span>
<span id="cb33-10"><a href="#cb33-10"></a>  <span class="kw">get_man</span>(df, <span class="dt">trait =</span> trait, <span class="dt">title =</span> title, <span class="dt">chr =</span> <span class="st">&quot;CHR&quot;</span>, <span class="dt">bp =</span> <span class="st">&quot;POS&quot;</span>, <span class="dt">snp =</span> <span class="st">&quot;TOP_SNP&quot;</span>, <span class="dt">p =</span> <span class="st">&quot;P&quot;</span>)</span>
<span id="cb33-11"><a href="#cb33-11"></a>})</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-12-1.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-2.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-3.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-4.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-5.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-6.png" /><!-- --></p>
<p>Save</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>output_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210122_manhattan_all_snps&quot;</span>)</span>
<span id="cb34-2"><a href="#cb34-2"></a></span>
<span id="cb34-3"><a href="#cb34-3"></a>counter =<span class="st"> </span><span class="dv">0</span></span>
<span id="cb34-4"><a href="#cb34-4"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno_df){</span>
<span id="cb34-5"><a href="#cb34-5"></a></span>
<span id="cb34-6"><a href="#cb34-6"></a>  <span class="co"># set counter</span></span>
<span id="cb34-7"><a href="#cb34-7"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb34-8"><a href="#cb34-8"></a>  df =<span class="st"> </span>pheno_df[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb34-9"><a href="#cb34-9"></a>  trait =<span class="st"> </span><span class="kw">unique</span>(df<span class="op">$</span>PHENO)</span>
<span id="cb34-10"><a href="#cb34-10"></a>  <span class="co"># Get title</span></span>
<span id="cb34-11"><a href="#cb34-11"></a>  title &lt;-<span class="st"> </span><span class="kw">paste</span>(trait, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, <span class="st">&quot;SNP count:&quot;</span>, <span class="kw">nrow</span>(df))</span>
<span id="cb34-12"><a href="#cb34-12"></a>  <span class="co"># Set file name to save</span></span>
<span id="cb34-13"><a href="#cb34-13"></a>  file =<span class="st"> </span><span class="kw">file.path</span>(output_dir,</span>
<span id="cb34-14"><a href="#cb34-14"></a>                   <span class="kw">paste</span>(<span class="st">&quot;manhattan_&quot;</span>,</span>
<span id="cb34-15"><a href="#cb34-15"></a>                         <span class="kw">names</span>(data_list)[counter],</span>
<span id="cb34-16"><a href="#cb34-16"></a>                         <span class="st">&quot;.svg&quot;</span>,</span>
<span id="cb34-17"><a href="#cb34-17"></a>                         <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb34-18"><a href="#cb34-18"></a>  <span class="co"># Set up graphics device</span></span>
<span id="cb34-19"><a href="#cb34-19"></a>  <span class="kw">svg</span>(file,</span>
<span id="cb34-20"><a href="#cb34-20"></a>      <span class="dt">width =</span> <span class="dv">10</span>,</span>
<span id="cb34-21"><a href="#cb34-21"></a>      <span class="dt">height =</span> <span class="dv">6</span>)  </span>
<span id="cb34-22"><a href="#cb34-22"></a>  </span>
<span id="cb34-23"><a href="#cb34-23"></a>  <span class="co"># Plot</span></span>
<span id="cb34-24"><a href="#cb34-24"></a>  <span class="kw">get_man</span>(df, <span class="dt">trait =</span> trait, <span class="dt">chr =</span> <span class="st">&quot;CHR&quot;</span>, <span class="dt">bp =</span> <span class="st">&quot;POS&quot;</span>, <span class="dt">snp =</span> <span class="st">&quot;TOP_SNP&quot;</span>, <span class="dt">p =</span> <span class="st">&quot;P&quot;</span>)</span>
<span id="cb34-25"><a href="#cb34-25"></a>  </span>
<span id="cb34-26"><a href="#cb34-26"></a>  <span class="kw">dev.off</span>()</span>
<span id="cb34-27"><a href="#cb34-27"></a>})</span></code></pre></div>
<div id="create-list-of-target-snps-to-extract-from-1kg" class="section level4">
<h4><span class="header-section-number">1.5.3.1</span> Create list of target SNPs to extract from 1KG</h4>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>dest_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210122_snp_hit_lists&quot;</span>)</span>
<span id="cb35-2"><a href="#cb35-2"></a></span>
<span id="cb35-3"><a href="#cb35-3"></a><span class="co"># Make directory</span></span>
<span id="cb35-4"><a href="#cb35-4"></a><span class="kw">dir.create</span>(dest_dir)</span>
<span id="cb35-5"><a href="#cb35-5"></a>  </span>
<span id="cb35-6"><a href="#cb35-6"></a><span class="co"># Just SNPs for extracting from 1KG</span></span>
<span id="cb35-7"><a href="#cb35-7"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb35-8"><a href="#cb35-8"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb35-9"><a href="#cb35-9"></a>  df =<span class="st"> </span>pheno[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb35-10"><a href="#cb35-10"></a>  <span class="co"># Set counter</span></span>
<span id="cb35-11"><a href="#cb35-11"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb35-12"><a href="#cb35-12"></a>  <span class="co"># Set file basename</span></span>
<span id="cb35-13"><a href="#cb35-13"></a>  trait =<span class="st"> </span><span class="kw">names</span>(data_list)[counter]</span>
<span id="cb35-14"><a href="#cb35-14"></a>  filename =<span class="st"> </span><span class="kw">paste</span>(trait, <span class="st">&quot;.list&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb35-15"><a href="#cb35-15"></a>  <span class="co"># Write SNPs to file</span></span>
<span id="cb35-16"><a href="#cb35-16"></a>  readr<span class="op">::</span><span class="kw">write_lines</span>(df<span class="op">$</span>TOP_SNP, <span class="kw">file.path</span>(dest_dir, filename))</span>
<span id="cb35-17"><a href="#cb35-17"></a>})</span>
<span id="cb35-18"><a href="#cb35-18"></a></span>
<span id="cb35-19"><a href="#cb35-19"></a><span class="co"># SNPs and P-values with header for clumping with Plink</span></span>
<span id="cb35-20"><a href="#cb35-20"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb35-21"><a href="#cb35-21"></a><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb35-22"><a href="#cb35-22"></a>  df =<span class="st"> </span>pheno[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb35-23"><a href="#cb35-23"></a>  <span class="co"># Set counter</span></span>
<span id="cb35-24"><a href="#cb35-24"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb35-25"><a href="#cb35-25"></a>  <span class="co"># Set file basename</span></span>
<span id="cb35-26"><a href="#cb35-26"></a>  trait =<span class="st"> </span><span class="kw">names</span>(data_list)[counter]</span>
<span id="cb35-27"><a href="#cb35-27"></a>  filename =<span class="st"> </span><span class="kw">paste</span>(trait, <span class="st">&quot;_with_P.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb35-28"><a href="#cb35-28"></a>  <span class="co"># Write SNPs to file</span></span>
<span id="cb35-29"><a href="#cb35-29"></a>  df <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb35-30"><a href="#cb35-30"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="dt">SNP =</span> TOP_SNP, P) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb35-31"><a href="#cb35-31"></a><span class="st">    </span>readr<span class="op">::</span><span class="kw">write_tsv</span>(<span class="kw">file.path</span>(dest_dir, filename))</span>
<span id="cb35-32"><a href="#cb35-32"></a>})</span></code></pre></div>
</div>
</div>
</div>
<div id="filter-1kg-vcf-for-target-snps" class="section level2">
<h2><span class="header-section-number">1.6</span> Filter 1KG VCF for target SNPs</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb36-1"><a href="#cb36-1"></a></span>
<span id="cb36-2"><a href="#cb36-2"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb36-3"><a href="#cb36-3"></a><span class="va">ref=</span>../refs/hs37d5.fa.gz</span>
<span id="cb36-4"><a href="#cb36-4"></a><span class="va">in_vcf=</span>../vcfs/1gk_all.vcf.gz</span>
<span id="cb36-5"><a href="#cb36-5"></a><span class="va">snps_dir=</span>data/20210122_snp_hit_lists</span>
<span id="cb36-6"><a href="#cb36-6"></a><span class="va">out_dir=</span>data/20210125_snp_hits_filtered</span>
<span id="cb36-7"><a href="#cb36-7"></a></span>
<span id="cb36-8"><a href="#cb36-8"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb36-9"><a href="#cb36-9"></a></span>
<span id="cb36-10"><a href="#cb36-10"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb36-11"><a href="#cb36-11"></a>  <span class="ex">bsub</span> \</span>
<span id="cb36-12"><a href="#cb36-12"></a>    -M 10000 \</span>
<span id="cb36-13"><a href="#cb36-13"></a>    -o ../log/20210122_extract_snps_<span class="va">$trait</span>.out \</span>
<span id="cb36-14"><a href="#cb36-14"></a>    -e ../log/20210122_extract_snps_<span class="va">$trait</span>.err \</span>
<span id="cb36-15"><a href="#cb36-15"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb36-16"><a href="#cb36-16"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb36-17"><a href="#cb36-17"></a><span class="st">    gatk SelectVariants \</span></span>
<span id="cb36-18"><a href="#cb36-18"></a><span class="st">      -R </span><span class="va">$ref</span><span class="st"> \</span></span>
<span id="cb36-19"><a href="#cb36-19"></a><span class="st">      -V </span><span class="va">$in_vcf</span><span class="st"> \</span></span>
<span id="cb36-20"><a href="#cb36-20"></a><span class="st">      --keep-ids </span><span class="va">$snps_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb36-21"><a href="#cb36-21"></a><span class="st">      -O </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz </span></span>
<span id="cb36-22"><a href="#cb36-22"></a><span class="st">    &quot;&quot;&quot;</span> <span class="kw">;</span></span>
<span id="cb36-23"><a href="#cb36-23"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="get-allele-frequencies-of-snp-hits-with-plink2" class="section level2">
<h2><span class="header-section-number">1.7</span> Get allele frequencies of SNP hits with <code>Plink2</code></h2>
<div id="import-1gk-metadata-for-sample-population-key" class="section level3">
<h3><span class="header-section-number">1.7.1</span> Import 1GK metadata (for sample-population key)</h3>
<p>Downloded via this page: <a href="http://www.internationalgenome.org/data" class="uri">http://www.internationalgenome.org/data</a> Download link: <a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_sample_info.xlsx" class="uri">http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_sample_info.xlsx</a>.</p>
<p>Saved here: <code>data/20130606_sample_info.xlsx</code></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>samples_file =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20130606_sample_info.xlsx&quot;</span>)</span>
<span id="cb37-2"><a href="#cb37-2"></a></span>
<span id="cb37-3"><a href="#cb37-3"></a>meta =<span class="st"> </span>readxl<span class="op">::</span><span class="kw">read_xlsx</span>(samples_file,</span>
<span id="cb37-4"><a href="#cb37-4"></a>                         <span class="dt">sheet =</span> <span class="st">&quot;Sample Info&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb37-5"><a href="#cb37-5"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Sample, Population, Gender)</span>
<span id="cb37-6"><a href="#cb37-6"></a></span>
<span id="cb37-7"><a href="#cb37-7"></a>knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">head</span>(meta))</span></code></pre></div>
=======
<div id="generate-manhattan-plots" class="section level3" number="1.5.3">
<h3 number="1.5.3"><span class="header-section-number">1.5.3</span> Generate Manhattan plots</h3>
<p>Plot</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno_df){</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> pheno_df[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">unique</span>(df<span class="sc">$</span>PHENO)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get title</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  title <span class="ot">&lt;-</span> <span class="fu">paste</span>(trait, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="st">&quot;SNP count:&quot;</span>, <span class="fu">nrow</span>(df))</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_man</span>(df, <span class="at">trait =</span> trait, <span class="at">title =</span> title, <span class="at">chr =</span> <span class="st">&quot;CHR&quot;</span>, <span class="at">bp =</span> <span class="st">&quot;POS&quot;</span>, <span class="at">snp =</span> <span class="st">&quot;TOP_SNP&quot;</span>, <span class="at">p =</span> <span class="st">&quot;P&quot;</span>)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-12-1.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-2.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-3.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-4.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-5.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-12-6.svg" /><!-- --></p>
<p>Save</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210122_manhattan_all_snps&quot;</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno_df){</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> pheno_df[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">unique</span>(df<span class="sc">$</span>PHENO)</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get title</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>  title <span class="ot">&lt;-</span> <span class="fu">paste</span>(trait, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="st">&quot;SNP count:&quot;</span>, <span class="fu">nrow</span>(df))</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set file name to save</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>  file <span class="ot">=</span> <span class="fu">file.path</span>(output_dir,</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">paste</span>(<span class="st">&quot;manhattan_&quot;</span>,</span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">names</span>(data_list)[counter],</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;.svg&quot;</span>,</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set up graphics device</span></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">svg</span>(file,</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>      <span class="at">width =</span> <span class="dv">10</span>,</span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">height =</span> <span class="dv">6</span>)  </span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_man</span>(df, <span class="at">trait =</span> trait, <span class="at">chr =</span> <span class="st">&quot;CHR&quot;</span>, <span class="at">bp =</span> <span class="st">&quot;POS&quot;</span>, <span class="at">snp =</span> <span class="st">&quot;TOP_SNP&quot;</span>, <span class="at">p =</span> <span class="st">&quot;P&quot;</span>)</span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<div id="create-list-of-target-snps-to-extract-from-1kg" class="section level4" number="1.5.3.1">
<h4 number="1.5.3.1"><span class="header-section-number">1.5.3.1</span> Create list of target SNPs to extract from 1KG</h4>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>dest_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210122_snp_hit_lists&quot;</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Make directory</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(dest_dir)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Just SNPs for extracting from 1KG</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> pheno[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set counter</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set file basename</span></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">=</span> <span class="fu">paste</span>(trait, <span class="st">&quot;.list&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Write SNPs to file</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_lines</span>(df<span class="sc">$</span>TOP_SNP, <span class="fu">file.path</span>(dest_dir, filename))</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a><span class="co"># SNPs and P-values with header for clumping with Plink</span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> pheno[[<span class="st">&quot;clean&quot;</span>]]</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set counter</span></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set file basename</span></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">=</span> <span class="fu">paste</span>(trait, <span class="st">&quot;_with_P.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Write SNPs to file</span></span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> </span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="at">SNP =</span> TOP_SNP, P) <span class="sc">%&gt;%</span> </span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>    readr<span class="sc">::</span><span class="fu">write_tsv</span>(<span class="fu">file.path</span>(dest_dir, filename))</span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
</div>
</div>
<div id="filter-1kg-vcf-for-target-snps" class="section level2" number="1.6">
<h2 number="1.6"><span class="header-section-number">1.6</span> Filter 1KG VCF for target SNPs</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="va">ref=</span>../refs/hs37d5.fa.gz</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="va">in_vcf=</span>../vcfs/1kg_all.vcf.gz</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="va">snps_dir=</span>data/20210122_snp_hit_lists</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir=</span>data/20210125_snp_hits_filtered</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bsub</span> <span class="kw">\</span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-M</span> 10000 <span class="kw">\</span></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-o</span> ../log/20210122_extract_snps_<span class="va">$trait</span>.out <span class="kw">\</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-e</span> ../log/20210122_extract_snps_<span class="va">$trait</span>.err <span class="kw">\</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a><span class="st">    gatk SelectVariants \</span></span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a><span class="st">      -R </span><span class="va">$ref</span><span class="st"> \</span></span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a><span class="st">      -V </span><span class="va">$in_vcf</span><span class="st"> \</span></span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a><span class="st">      --keep-ids </span><span class="va">$snps_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a><span class="st">      -O </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz </span></span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a><span class="st">    &quot;&quot;&quot;</span> ;</span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="get-allele-frequencies-of-snp-hits-with-plink2" class="section level2" number="1.7">
<h2 number="1.7"><span class="header-section-number">1.7</span> Get allele frequencies of SNP hits with <code>Plink2</code></h2>
<div id="import-1kg-metadata-for-sample-population-key" class="section level3" number="1.7.1">
<h3 number="1.7.1"><span class="header-section-number">1.7.1</span> Import 1KG metadata (for sample-population key)</h3>
<p>Downloded via this page: <a href="http://www.internationalgenome.org/data" class="uri">http://www.internationalgenome.org/data</a>.</p>
<p>Download link: <a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_sample_info.xlsx" class="uri">http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_sample_info.xlsx</a>.</p>
<p>Saved here: <code>data/20130606_sample_info.xlsx</code></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>samples_file <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20130606_sample_info.xlsx&quot;</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">=</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(samples_file,</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Sample Info&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Population, Gender)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(meta))</span></code></pre></div>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<table>
<thead>
<tr class="header">
<th align="left">Sample</th>
<th align="left">Population</th>
<th align="left">Gender</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HG00096</td>
<td align="left">GBR</td>
<td align="left">male</td>
</tr>
<tr class="even">
<td align="left">HG00097</td>
<td align="left">GBR</td>
<td align="left">female</td>
</tr>
<tr class="odd">
<td align="left">HG00098</td>
<td align="left">GBR</td>
<td align="left">male</td>
</tr>
<tr class="even">
<td align="left">HG00099</td>
<td align="left">GBR</td>
<td align="left">female</td>
</tr>
<tr class="odd">
<td align="left">HG00100</td>
<td align="left">GBR</td>
<td align="left">female</td>
</tr>
<tr class="even">
<td align="left">HG00101</td>
<td align="left">GBR</td>
<td align="left">male</td>
</tr>
</tbody>
</table>
</div>
<<<<<<< HEAD
<div id="write-population-file-for-plink2" class="section level3">
<h3><span class="header-section-number">1.7.2</span> Write population file for Plink2</h3>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>sample_popn_key_file =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;plink2_sample_popn_key.txt&quot;</span>)</span>
<span id="cb38-2"><a href="#cb38-2"></a></span>
<span id="cb38-3"><a href="#cb38-3"></a><span class="kw">write.table</span>(meta[, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>],</span>
<span id="cb38-4"><a href="#cb38-4"></a>            sample_popn_key_file,</span>
<span id="cb38-5"><a href="#cb38-5"></a>            <span class="dt">quote =</span> F,</span>
<span id="cb38-6"><a href="#cb38-6"></a>            <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,</span>
<span id="cb38-7"><a href="#cb38-7"></a>            <span class="dt">row.names =</span> F,</span>
<span id="cb38-8"><a href="#cb38-8"></a>            <span class="dt">col.names =</span> F)</span></code></pre></div>
</div>
<div id="run-plink2-for-snp-hits" class="section level3">
<h3><span class="header-section-number">1.7.3</span> Run <code>Plink2</code> for SNP hits</h3>
<p>(Take only biallelic SNPs.)</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1"></a></span>
<span id="cb39-2"><a href="#cb39-2"></a><span class="ex">conda</span> activate fst_env_rhel</span>
<span id="cb39-3"><a href="#cb39-3"></a></span>
<span id="cb39-4"><a href="#cb39-4"></a><span class="co"># Set variables</span></span>
<span id="cb39-5"><a href="#cb39-5"></a></span>
<span id="cb39-6"><a href="#cb39-6"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb39-7"><a href="#cb39-7"></a><span class="va">in_vcf_dir=</span>data/20210125_snp_hits_filtered</span>
<span id="cb39-8"><a href="#cb39-8"></a><span class="va">popn_file=</span>data/plink2_sample_popn_key.txt</span>
<span id="cb39-9"><a href="#cb39-9"></a><span class="va">out_dir=</span>data/20210122_snp_hits_alfreqs</span>
<span id="cb39-10"><a href="#cb39-10"></a></span>
<span id="cb39-11"><a href="#cb39-11"></a><span class="co"># Set up directories</span></span>
<span id="cb39-12"><a href="#cb39-12"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb39-13"><a href="#cb39-13"></a></span>
<span id="cb39-14"><a href="#cb39-14"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb39-15"><a href="#cb39-15"></a>  <span class="fu">mkdir</span> -p <span class="va">$out_dir</span>/<span class="va">$trait</span><span class="kw">;</span> </span>
<span id="cb39-16"><a href="#cb39-16"></a><span class="kw">done</span>   </span>
<span id="cb39-17"><a href="#cb39-17"></a></span>
<span id="cb39-18"><a href="#cb39-18"></a><span class="co"># Run Plink2</span></span>
<span id="cb39-19"><a href="#cb39-19"></a></span>
<span id="cb39-20"><a href="#cb39-20"></a><span class="co">## Get global AF</span></span>
<span id="cb39-21"><a href="#cb39-21"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb39-22"><a href="#cb39-22"></a>  <span class="ex">plink2</span> \</span>
<span id="cb39-23"><a href="#cb39-23"></a>    --vcf <span class="va">$in_vcf_dir</span>/<span class="va">$trait</span>.vcf.gz \</span>
<span id="cb39-24"><a href="#cb39-24"></a>    --freq \</span>
<span id="cb39-25"><a href="#cb39-25"></a>    --max-alleles 2 \</span>
<span id="cb39-26"><a href="#cb39-26"></a>    --snps-only \</span>
<span id="cb39-27"><a href="#cb39-27"></a>    --out <span class="va">$out_dir</span>/<span class="va">$trait</span>/<span class="va">$trait</span>.all</span>
<span id="cb39-28"><a href="#cb39-28"></a><span class="kw">done</span></span>
<span id="cb39-29"><a href="#cb39-29"></a></span>
<span id="cb39-30"><a href="#cb39-30"></a><span class="co">## Get AF per population</span></span>
<span id="cb39-31"><a href="#cb39-31"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb39-32"><a href="#cb39-32"></a>  <span class="ex">plink2</span> \</span>
<span id="cb39-33"><a href="#cb39-33"></a>    --vcf <span class="va">$in_vcf_dir</span>/<span class="va">$trait</span>.vcf.gz \</span>
<span id="cb39-34"><a href="#cb39-34"></a>    --freq \</span>
<span id="cb39-35"><a href="#cb39-35"></a>    --max-alleles 2 \</span>
<span id="cb39-36"><a href="#cb39-36"></a>    --snps-only \</span>
<span id="cb39-37"><a href="#cb39-37"></a>    --pheno iid-only <span class="va">$popn_file</span> \</span>
<span id="cb39-38"><a href="#cb39-38"></a>    --loop-cats PHENO1 \</span>
<span id="cb39-39"><a href="#cb39-39"></a>    --out <span class="va">$out_dir</span>/<span class="va">$trait</span>/<span class="va">$trait</span> <span class="kw">;</span></span>
<span id="cb39-40"><a href="#cb39-40"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="add-allele-frequency-files" class="section level3">
<h3><span class="header-section-number">1.7.4</span> Add allele frequency files</h3>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a>target_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210122_snp_hits_alfreqs&quot;</span>)</span>
<span id="cb40-2"><a href="#cb40-2"></a></span>
<span id="cb40-3"><a href="#cb40-3"></a><span class="co"># Global</span></span>
<span id="cb40-4"><a href="#cb40-4"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb40-5"><a href="#cb40-5"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb40-6"><a href="#cb40-6"></a>  <span class="co"># set counter </span></span>
<span id="cb40-7"><a href="#cb40-7"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb40-8"><a href="#cb40-8"></a>  <span class="co"># get trait name</span></span>
<span id="cb40-9"><a href="#cb40-9"></a>  trait =<span class="st"> </span><span class="kw">names</span>(data_list)[counter]</span>
<span id="cb40-10"><a href="#cb40-10"></a>  <span class="co"># get file path</span></span>
<span id="cb40-11"><a href="#cb40-11"></a>  target_path =<span class="st"> </span><span class="kw">file.path</span>(target_dir, trait, <span class="kw">paste</span>(trait, <span class="st">&quot;.all.afreq&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb40-12"><a href="#cb40-12"></a>  <span class="co"># read in data</span></span>
<span id="cb40-13"><a href="#cb40-13"></a>  clean_af =<span class="st"> </span><span class="kw">read_afreq</span>(target_path)</span>
<span id="cb40-14"><a href="#cb40-14"></a>  <span class="co"># add POPN column</span></span>
<span id="cb40-15"><a href="#cb40-15"></a>  clean_af<span class="op">$</span>POPN =<span class="st"> &quot;all&quot;</span></span>
<span id="cb40-16"><a href="#cb40-16"></a>  <span class="co"># add to list</span></span>
<span id="cb40-17"><a href="#cb40-17"></a>  pheno[[<span class="st">&quot;clean_af&quot;</span>]] =<span class="st"> </span>clean_af</span>
<span id="cb40-18"><a href="#cb40-18"></a>  </span>
<span id="cb40-19"><a href="#cb40-19"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb40-20"><a href="#cb40-20"></a>})</span>
<span id="cb40-21"><a href="#cb40-21"></a></span>
<span id="cb40-22"><a href="#cb40-22"></a><span class="co"># Per population</span></span>
<span id="cb40-23"><a href="#cb40-23"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb40-24"><a href="#cb40-24"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb40-25"><a href="#cb40-25"></a>  <span class="co"># set counter</span></span>
<span id="cb40-26"><a href="#cb40-26"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb40-27"><a href="#cb40-27"></a>  <span class="co"># get trait name</span></span>
<span id="cb40-28"><a href="#cb40-28"></a>  trait =<span class="st"> </span><span class="kw">names</span>(data_list)[counter]</span>
<span id="cb40-29"><a href="#cb40-29"></a>  <span class="co"># get file path</span></span>
<span id="cb40-30"><a href="#cb40-30"></a>  target_files =<span class="st"> </span><span class="kw">list.files</span>(<span class="kw">file.path</span>(target_dir, trait),</span>
<span id="cb40-31"><a href="#cb40-31"></a>                            <span class="dt">pattern =</span> <span class="st">&quot;.*[^all]</span><span class="ch">\\</span><span class="st">.afreq&quot;</span>,</span>
<span id="cb40-32"><a href="#cb40-32"></a>                            <span class="dt">full.names =</span> T)</span>
<span id="cb40-33"><a href="#cb40-33"></a>  <span class="co"># get popn names</span></span>
<span id="cb40-34"><a href="#cb40-34"></a>  <span class="kw">names</span>(target_files) =<span class="st"> </span><span class="kw">basename</span>(target_files) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb40-35"><a href="#cb40-35"></a><span class="st">    </span><span class="kw">str_split</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="dt">simplify =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb40-36"><a href="#cb40-36"></a><span class="st">    </span><span class="kw">subset</span>(<span class="dt">select =</span> <span class="dv">2</span>)</span>
<span id="cb40-37"><a href="#cb40-37"></a>  <span class="co"># read files and bind into single DF</span></span>
<span id="cb40-38"><a href="#cb40-38"></a>  popn_afreqs =<span class="st"> </span><span class="kw">lapply</span>(target_files, <span class="cf">function</span>(popn){</span>
<span id="cb40-39"><a href="#cb40-39"></a>    df =<span class="st"> </span><span class="kw">read_afreq</span>(popn)</span>
<span id="cb40-40"><a href="#cb40-40"></a>  }) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb40-41"><a href="#cb40-41"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;POPN&quot;</span>)<span class="co"># %&gt;% </span></span>
<span id="cb40-42"><a href="#cb40-42"></a><span class="co">#    dplyr::select(-OBS_CT) %&gt;% </span></span>
<span id="cb40-43"><a href="#cb40-43"></a><span class="co">#    tidyr::pivot_wider(id_cols = SNP, names_from = POPN, values_from = ALT_FREQS)</span></span>
<span id="cb40-44"><a href="#cb40-44"></a>  <span class="co"># combine with `clean_af`</span></span>
<span id="cb40-45"><a href="#cb40-45"></a>  pheno[[<span class="st">&quot;clean_af&quot;</span>]] =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(pheno[[<span class="st">&quot;clean_af&quot;</span>]],</span>
<span id="cb40-46"><a href="#cb40-46"></a>                                         popn_afreqs)</span>
<span id="cb40-47"><a href="#cb40-47"></a>  </span>
<span id="cb40-48"><a href="#cb40-48"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb40-49"><a href="#cb40-49"></a>})</span></code></pre></div>
</div>
</div>
<div id="set-up-negative-controls" class="section level2">
<h2><span class="header-section-number">1.8</span> Set up negative controls</h2>
<p>Pull out random SNPs with the same global allele frequencies as the GWAS SNP-hits</p>
<div id="bin-snp-hits-by-allele-frequency" class="section level3">
<h3><span class="header-section-number">1.8.1</span> Bin SNP hits by allele frequency</h3>
<p>Bind to clean DF to get AFs of risk allele</p>
<p><span style="color: red;"><strong>NOTE</strong></span>: If <code>RISK_ALLELE</code> is unknown, set the allele frequency to <code>ALT_FREQS</code></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb41-2"><a href="#cb41-2"></a>  <span class="co"># join DFs</span></span>
<span id="cb41-3"><a href="#cb41-3"></a>  df =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(pheno[[<span class="st">&quot;clean&quot;</span>]],</span>
<span id="cb41-4"><a href="#cb41-4"></a>                        dplyr<span class="op">::</span><span class="kw">select</span>(pheno[[<span class="st">&quot;clean_af&quot;</span>]],</span>
<span id="cb41-5"><a href="#cb41-5"></a>                                      <span class="op">-</span>CHR),</span>
<span id="cb41-6"><a href="#cb41-6"></a>                        <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;TOP_SNP&quot;</span> =<span class="st"> &quot;SNP&quot;</span>))</span>
<span id="cb41-7"><a href="#cb41-7"></a>  <span class="co"># get AF of risk allele</span></span>
<span id="cb41-8"><a href="#cb41-8"></a>  df<span class="op">$</span>RISK_AF =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">if_else</span>(df<span class="op">$</span>RISK_ALLELE <span class="op">==</span><span class="st"> &quot;?&quot;</span>,</span>
<span id="cb41-9"><a href="#cb41-9"></a>                              df<span class="op">$</span>ALT_FREQS,</span>
<span id="cb41-10"><a href="#cb41-10"></a>                              dplyr<span class="op">::</span><span class="kw">if_else</span>(df<span class="op">$</span>RISK_ALLELE <span class="op">==</span><span class="st"> </span>df<span class="op">$</span>ALT,</span>
<span id="cb41-11"><a href="#cb41-11"></a>                                             df<span class="op">$</span>ALT_FREQS,</span>
<span id="cb41-12"><a href="#cb41-12"></a>                                             <span class="dv">1</span> <span class="op">-</span><span class="st"> </span>df<span class="op">$</span>ALT_FREQS))</span>
<span id="cb41-13"><a href="#cb41-13"></a>  <span class="co"># add `HIT_CONTROL` column</span></span>
<span id="cb41-14"><a href="#cb41-14"></a>  df<span class="op">$</span>HIT_CONTROL =<span class="st"> &quot;hit&quot;</span></span>
<span id="cb41-15"><a href="#cb41-15"></a>  <span class="co"># add to list</span></span>
<span id="cb41-16"><a href="#cb41-16"></a>  pheno[[<span class="st">&quot;consol&quot;</span>]] =<span class="st"> </span>df</span>
<span id="cb41-17"><a href="#cb41-17"></a>  </span>
<span id="cb41-18"><a href="#cb41-18"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb41-19"><a href="#cb41-19"></a>})</span></code></pre></div>
<p>Bin by risk allele frequency</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="co"># 1% intervals</span></span>
<span id="cb42-2"><a href="#cb42-2"></a>breakpoints =<span class="st"> </span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.01</span>)</span>
<span id="cb42-3"><a href="#cb42-3"></a></span>
<span id="cb42-4"><a href="#cb42-4"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb42-5"><a href="#cb42-5"></a>  <span class="co"># choose DF</span></span>
<span id="cb42-6"><a href="#cb42-6"></a>  df =<span class="st"> </span>pheno[[<span class="st">&quot;consol&quot;</span>]]</span>
<span id="cb42-7"><a href="#cb42-7"></a>  <span class="co"># add bins</span></span>
<span id="cb42-8"><a href="#cb42-8"></a>  df<span class="op">$</span>BIN_<span class="dv">100</span> =<span class="st"> </span><span class="kw">cut</span>(df<span class="op">$</span>RISK_AF, <span class="dt">breaks =</span> breakpoints, <span class="dt">labels =</span> F)</span>
<span id="cb42-9"><a href="#cb42-9"></a>  <span class="co"># save back into list</span></span>
<span id="cb42-10"><a href="#cb42-10"></a>  pheno[[<span class="st">&quot;consol&quot;</span>]] =<span class="st"> </span>df</span>
<span id="cb42-11"><a href="#cb42-11"></a>  </span>
<span id="cb42-12"><a href="#cb42-12"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb42-13"><a href="#cb42-13"></a>})</span></code></pre></div>
<p>Extract key columns and write to file</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a>out_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210126_snp_risk_hits_binned&quot;</span>)</span>
<span id="cb43-2"><a href="#cb43-2"></a></span>
<span id="cb43-3"><a href="#cb43-3"></a><span class="kw">dir.create</span>(out_dir)</span>
<span id="cb43-4"><a href="#cb43-4"></a></span>
<span id="cb43-5"><a href="#cb43-5"></a><span class="co"># Save list</span></span>
<span id="cb43-6"><a href="#cb43-6"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb43-7"><a href="#cb43-7"></a>risk_afs =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb43-8"><a href="#cb43-8"></a>  <span class="co"># set counter </span></span>
<span id="cb43-9"><a href="#cb43-9"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb43-10"><a href="#cb43-10"></a>  <span class="co"># get target DF</span></span>
<span id="cb43-11"><a href="#cb43-11"></a>  df =<span class="st"> </span>pheno[[<span class="st">&quot;consol&quot;</span>]]</span>
<span id="cb43-12"><a href="#cb43-12"></a>  <span class="co"># filter</span></span>
<span id="cb43-13"><a href="#cb43-13"></a>  df =<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-14"><a href="#cb43-14"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(POPN <span class="op">==</span><span class="st"> &quot;all&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># take only global AFs </span></span>
<span id="cb43-15"><a href="#cb43-15"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(TOP_SNP, BIN_<span class="dv">100</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-16"><a href="#cb43-16"></a><span class="st">    </span>tidyr<span class="op">::</span><span class="kw">drop_na</span>() <span class="co"># drop NAs</span></span>
<span id="cb43-17"><a href="#cb43-17"></a>  <span class="co"># set output path</span></span>
<span id="cb43-18"><a href="#cb43-18"></a>  trait =<span class="st"> </span><span class="kw">names</span>(data_list)[counter]</span>
<span id="cb43-19"><a href="#cb43-19"></a>  path_out =<span class="st"> </span><span class="kw">file.path</span>(out_dir, <span class="kw">paste</span>(trait, <span class="st">&quot;.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb43-20"><a href="#cb43-20"></a>  <span class="co"># write to file</span></span>
<span id="cb43-21"><a href="#cb43-21"></a>  readr<span class="op">::</span><span class="kw">write_tsv</span>(df, path_out)</span>
<span id="cb43-22"><a href="#cb43-22"></a>})</span></code></pre></div>
</div>
<div id="bin-1kg-snps" class="section level3">
<h3><span class="header-section-number">1.8.2</span> Bin 1KG SNPs</h3>
<div id="get-allele-frequencies-from-1kg" class="section level4">
<h4><span class="header-section-number">1.8.2.1</span> Get allele frequencies from 1KG</h4>
<p>With <code>Plink2</code>, per chromosome for speed.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1"></a><span class="co"># set output directory</span></span>
<span id="cb44-2"><a href="#cb44-2"></a><span class="va">in_file=</span>../vcfs/1gk_all.vcf.gz</span>
<span id="cb44-3"><a href="#cb44-3"></a><span class="va">out_dir=</span>../big_data/20210125_alfreqs_all</span>
<span id="cb44-4"><a href="#cb44-4"></a></span>
<span id="cb44-5"><a href="#cb44-5"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb44-6"><a href="#cb44-6"></a></span>
<span id="cb44-7"><a href="#cb44-7"></a><span class="co"># Per chromosome</span></span>
<span id="cb44-8"><a href="#cb44-8"></a><span class="kw">for</span> <span class="ex">chr</span> in <span class="va">$(</span><span class="fu">seq</span> 1 22<span class="va">)</span> <span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb44-9"><a href="#cb44-9"></a>  <span class="co"># create allele-freq tables</span></span>
<span id="cb44-10"><a href="#cb44-10"></a>  <span class="ex">bsub</span> \</span>
<span id="cb44-11"><a href="#cb44-11"></a>    -M 10000 \</span>
<span id="cb44-12"><a href="#cb44-12"></a>    -o ../log/20210125_plink_alfreq_<span class="va">$chr</span>.out \</span>
<span id="cb44-13"><a href="#cb44-13"></a>    -e ../log/20210125_plink_alfreq_<span class="va">$chr</span>.err \</span>
<span id="cb44-14"><a href="#cb44-14"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb44-15"><a href="#cb44-15"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb44-16"><a href="#cb44-16"></a><span class="st">    plink2 \</span></span>
<span id="cb44-17"><a href="#cb44-17"></a><span class="st">      --vcf </span><span class="va">$in_file</span><span class="st"> \</span></span>
<span id="cb44-18"><a href="#cb44-18"></a><span class="st">      --freq \</span></span>
<span id="cb44-19"><a href="#cb44-19"></a><span class="st">      --chr </span><span class="va">$chr</span><span class="st"> \</span></span>
<span id="cb44-20"><a href="#cb44-20"></a><span class="st">      --max-alleles 2 \</span></span>
<span id="cb44-21"><a href="#cb44-21"></a><span class="st">      --snps-only \</span></span>
<span id="cb44-22"><a href="#cb44-22"></a><span class="st">      --out </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$chr</span><span class="st"> &quot;</span><span class="kw">;</span></span>
<span id="cb44-23"><a href="#cb44-23"></a><span class="kw">done</span> </span></code></pre></div>
</div>
<div id="bin-them-and-save-to-single-file" class="section level4">
<h4><span class="header-section-number">1.8.2.2</span> Bin them and save to single file</h4>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a><span class="co"># On cluster</span></span>
<span id="cb45-2"><a href="#cb45-2"></a></span>
<span id="cb45-3"><a href="#cb45-3"></a><span class="kw">library</span>(here)</span>
<span id="cb45-4"><a href="#cb45-4"></a><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;code&quot;</span>, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;source.R&quot;</span>))</span>
<span id="cb45-5"><a href="#cb45-5"></a></span>
<span id="cb45-6"><a href="#cb45-6"></a><span class="co"># Set variables</span></span>
<span id="cb45-7"><a href="#cb45-7"></a>in_dir =<span class="st"> &quot;../big_data/20210125_alfreqs_all&quot;</span></span>
<span id="cb45-8"><a href="#cb45-8"></a>out_dir =<span class="st"> &quot;../big_data/20210125_alfreqs_all_binned&quot;</span></span>
<span id="cb45-9"><a href="#cb45-9"></a>breakpoints =<span class="st"> </span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.01</span>) <span class="co"># 1% bins</span></span>
<span id="cb45-10"><a href="#cb45-10"></a></span>
<span id="cb45-11"><a href="#cb45-11"></a><span class="co"># Create output directory</span></span>
<span id="cb45-12"><a href="#cb45-12"></a><span class="kw">dir.create</span>(out_dir)</span>
<span id="cb45-13"><a href="#cb45-13"></a></span>
<span id="cb45-14"><a href="#cb45-14"></a><span class="co"># Get list of input files</span></span>
<span id="cb45-15"><a href="#cb45-15"></a>in_files =<span class="st"> </span><span class="kw">list.files</span>(in_dir, <span class="dt">pattern =</span> <span class="st">&quot;.afreq&quot;</span>, <span class="dt">full.names =</span> T)</span>
<span id="cb45-16"><a href="#cb45-16"></a></span>
<span id="cb45-17"><a href="#cb45-17"></a><span class="co"># Read in files, add bins, and write to output</span></span>
<span id="cb45-18"><a href="#cb45-18"></a>freq_list =<span class="st"> </span><span class="kw">lapply</span>(in_files, <span class="cf">function</span>(chr_file){</span>
<span id="cb45-19"><a href="#cb45-19"></a>  <span class="co"># read in file</span></span>
<span id="cb45-20"><a href="#cb45-20"></a>  df =<span class="st"> </span><span class="kw">read_afreq</span>(chr_file)</span>
<span id="cb45-21"><a href="#cb45-21"></a>  <span class="co"># add bins</span></span>
<span id="cb45-22"><a href="#cb45-22"></a>  df<span class="op">$</span>BIN_<span class="dv">100</span> =<span class="st"> </span><span class="kw">cut</span>(df<span class="op">$</span>ALT_FREQS, <span class="dt">breaks =</span> breakpoints, <span class="dt">labels =</span> F)</span>
<span id="cb45-23"><a href="#cb45-23"></a>  <span class="co"># write file</span></span>
<span id="cb45-24"><a href="#cb45-24"></a>  readr<span class="op">::</span><span class="kw">write_tsv</span>(df, <span class="dt">file =</span> <span class="kw">file.path</span>(out_dir, <span class="kw">basename</span>(chr_file)))</span>
<span id="cb45-25"><a href="#cb45-25"></a>})</span>
<span id="cb45-26"><a href="#cb45-26"></a></span>
<span id="cb45-27"><a href="#cb45-27"></a><span class="co"># Combine into single DF</span></span>
<span id="cb45-28"><a href="#cb45-28"></a>freq_df =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(freq_list)</span>
<span id="cb45-29"><a href="#cb45-29"></a></span>
<span id="cb45-30"><a href="#cb45-30"></a><span class="co"># Write to file</span></span>
<span id="cb45-31"><a href="#cb45-31"></a>readr<span class="op">::</span><span class="kw">write_tsv</span>(freq_df, <span class="dt">file =</span> <span class="kw">file.path</span>(out_dir, <span class="st">&quot;all.afreq&quot;</span>))</span></code></pre></div>
</div>
<div id="pull-out-random-snps-with-same-af-as-trait-risk-alleles" class="section level4">
<h4><span class="header-section-number">1.8.2.3</span> Pull out random SNPs with same AF as trait risk alleles</h4>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a><span class="co"># On cluster</span></span>
<span id="cb46-2"><a href="#cb46-2"></a></span>
<span id="cb46-3"><a href="#cb46-3"></a><span class="kw">library</span>(here)</span>
<span id="cb46-4"><a href="#cb46-4"></a><span class="kw">source</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;code&quot;</span>, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;source.R&quot;</span>))</span>
<span id="cb46-5"><a href="#cb46-5"></a></span>
<span id="cb46-6"><a href="#cb46-6"></a><span class="co"># Variables</span></span>
<span id="cb46-7"><a href="#cb46-7"></a></span>
<span id="cb46-8"><a href="#cb46-8"></a>input_risk_snp_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210126_snp_risk_hits_binned&quot;</span>)</span>
<span id="cb46-9"><a href="#cb46-9"></a>all_1kg_bins =<span class="st"> &quot;../big_data/20210125_alfreqs_all_binned/all.afreq&quot;</span></span>
<span id="cb46-10"><a href="#cb46-10"></a>initial_seed =<span class="st"> </span><span class="dv">123</span></span>
<span id="cb46-11"><a href="#cb46-11"></a>output_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210126_random_snps&quot;</span>)</span>
<span id="cb46-12"><a href="#cb46-12"></a>output_dir_snpids =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210126_random_snps_snp_ids&quot;</span>)</span>
<span id="cb46-13"><a href="#cb46-13"></a></span>
<span id="cb46-14"><a href="#cb46-14"></a><span class="kw">dir.create</span>(output_dir)</span>
<span id="cb46-15"><a href="#cb46-15"></a><span class="kw">dir.create</span>(output_dir_snpids)</span>
<span id="cb46-16"><a href="#cb46-16"></a></span>
<span id="cb46-17"><a href="#cb46-17"></a><span class="co">## Read in target SNP DF and split into list by bin</span></span>
<span id="cb46-18"><a href="#cb46-18"></a>phenos =<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.txt&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">basename</span>(<span class="kw">list.files</span>(input_risk_snp_dir)))</span>
<span id="cb46-19"><a href="#cb46-19"></a><span class="kw">names</span>(phenos) =<span class="st"> </span>phenos</span>
<span id="cb46-20"><a href="#cb46-20"></a></span>
<span id="cb46-21"><a href="#cb46-21"></a>risk_list =<span class="st"> </span><span class="kw">lapply</span>(phenos, <span class="cf">function</span>(pheno){</span>
<span id="cb46-22"><a href="#cb46-22"></a>  <span class="co"># set file path</span></span>
<span id="cb46-23"><a href="#cb46-23"></a>  file_path =<span class="st"> </span><span class="kw">file.path</span>(input_risk_snp_dir, <span class="kw">paste</span>(pheno, <span class="st">&quot;.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb46-24"><a href="#cb46-24"></a>  <span class="co"># read file</span></span>
<span id="cb46-25"><a href="#cb46-25"></a>  out =<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(file_path,</span>
<span id="cb46-26"><a href="#cb46-26"></a>                        <span class="dt">col_names =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb46-27"><a href="#cb46-27"></a><span class="st">    </span><span class="kw">split</span>(., <span class="dt">f =</span> .<span class="op">$</span>BIN_<span class="dv">100</span>) <span class="co"># split by bin</span></span>
<span id="cb46-28"><a href="#cb46-28"></a>})</span>
<span id="cb46-29"><a href="#cb46-29"></a>  </span>
<span id="cb46-30"><a href="#cb46-30"></a><span class="co">## Read in 1KG data</span></span>
<span id="cb46-31"><a href="#cb46-31"></a>freq_df =<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(all_1kg_bins)</span>
<span id="cb46-32"><a href="#cb46-32"></a></span>
<span id="cb46-33"><a href="#cb46-33"></a><span class="co"># For each bin in `risk_list`, pull out the same number of random number 1KG SNPs with the same bin</span></span>
<span id="cb46-34"><a href="#cb46-34"></a></span>
<span id="cb46-35"><a href="#cb46-35"></a><span class="co">## Set seed</span></span>
<span id="cb46-36"><a href="#cb46-36"></a><span class="kw">set.seed</span>(initial_seed)</span>
<span id="cb46-37"><a href="#cb46-37"></a></span>
<span id="cb46-38"><a href="#cb46-38"></a><span class="co">## Get seeds for each bin</span></span>
<span id="cb46-39"><a href="#cb46-39"></a>seeds =<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>, <span class="kw">length</span>(risk_list))</span>
<span id="cb46-40"><a href="#cb46-40"></a></span>
<span id="cb46-41"><a href="#cb46-41"></a><span class="co">## Run over list</span></span>
<span id="cb46-42"><a href="#cb46-42"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb46-43"><a href="#cb46-43"></a><span class="kw">lapply</span>(risk_list, <span class="cf">function</span>(pheno){</span>
<span id="cb46-44"><a href="#cb46-44"></a>  <span class="co"># set counter </span></span>
<span id="cb46-45"><a href="#cb46-45"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span>  </span>
<span id="cb46-46"><a href="#cb46-46"></a>  <span class="co"># set seed for pheno</span></span>
<span id="cb46-47"><a href="#cb46-47"></a>  <span class="kw">set.seed</span>(seeds)[counter]</span>
<span id="cb46-48"><a href="#cb46-48"></a>  <span class="co"># get seeds for bin</span></span>
<span id="cb46-49"><a href="#cb46-49"></a>  bin_seeds =<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>, <span class="kw">length</span>(pheno))</span>
<span id="cb46-50"><a href="#cb46-50"></a>  <span class="co"># get random SNPs from each bin</span></span>
<span id="cb46-51"><a href="#cb46-51"></a>  bin_counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb46-52"><a href="#cb46-52"></a>  out =<span class="st"> </span><span class="kw">lapply</span>(pheno, <span class="cf">function</span>(bin_df){</span>
<span id="cb46-53"><a href="#cb46-53"></a>    <span class="co"># set `bin_counter`</span></span>
<span id="cb46-54"><a href="#cb46-54"></a>    bin_counter &lt;&lt;-<span class="st"> </span>bin_counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb46-55"><a href="#cb46-55"></a>    <span class="co"># get target bin</span></span>
<span id="cb46-56"><a href="#cb46-56"></a>    target_bin =<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">names</span>(pheno)[bin_counter])</span>
<span id="cb46-57"><a href="#cb46-57"></a>    <span class="co"># get number of matches required</span></span>
<span id="cb46-58"><a href="#cb46-58"></a>    hits_n =<span class="st"> </span><span class="kw">nrow</span>(bin_df)</span>
<span id="cb46-59"><a href="#cb46-59"></a>    <span class="co"># set seed</span></span>
<span id="cb46-60"><a href="#cb46-60"></a>    <span class="kw">set.seed</span>(bin_seeds[bin_counter])    </span>
<span id="cb46-61"><a href="#cb46-61"></a>    <span class="co"># filter 1kg DF for SNPs with same bin and get random hits</span></span>
<span id="cb46-62"><a href="#cb46-62"></a>    random_hits =<span class="st"> </span>freq_df <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb46-63"><a href="#cb46-63"></a><span class="st">      </span><span class="co">#dplyr::select(SNP, ALT_FREQS, OBS_CT, BIN_100) %&gt;% </span></span>
<span id="cb46-64"><a href="#cb46-64"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">filter</span>(BIN_<span class="dv">100</span> <span class="op">==</span><span class="st"> </span>target_bin) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb46-65"><a href="#cb46-65"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">slice_sample</span>(<span class="dt">n =</span> hits_n) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb46-66"><a href="#cb46-66"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">RANDOM_SNP =</span> SNP,</span>
<span id="cb46-67"><a href="#cb46-67"></a>                    <span class="dt">RANDOM_BIN_100 =</span> BIN_<span class="dv">100</span>)</span>
<span id="cb46-68"><a href="#cb46-68"></a>    <span class="co"># bind `random_hits` to target SNP df</span></span>
<span id="cb46-69"><a href="#cb46-69"></a>    df_out =<span class="st"> </span><span class="kw">cbind</span>(bin_df, random_hits)</span>
<span id="cb46-70"><a href="#cb46-70"></a>    </span>
<span id="cb46-71"><a href="#cb46-71"></a>    <span class="kw">return</span>(df_out)</span>
<span id="cb46-72"><a href="#cb46-72"></a>  }) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb46-73"><a href="#cb46-73"></a><span class="st">    </span><span class="co"># bind into single data frame</span></span>
<span id="cb46-74"><a href="#cb46-74"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>()</span>
<span id="cb46-75"><a href="#cb46-75"></a>  </span>
<span id="cb46-76"><a href="#cb46-76"></a>  <span class="co"># save to file</span></span>
<span id="cb46-77"><a href="#cb46-77"></a>  <span class="co">## set output path</span></span>
<span id="cb46-78"><a href="#cb46-78"></a>  trait =<span class="st"> </span><span class="kw">names</span>(risk_list)[counter]</span>
<span id="cb46-79"><a href="#cb46-79"></a>  out_path =<span class="st"> </span><span class="kw">file.path</span>(output_dir, <span class="kw">paste</span>(trait, <span class="st">&quot;.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb46-80"><a href="#cb46-80"></a>  <span class="co">## write file</span></span>
<span id="cb46-81"><a href="#cb46-81"></a>  readr<span class="op">::</span><span class="kw">write_tsv</span>(out, out_path)</span>
<span id="cb46-82"><a href="#cb46-82"></a>  </span>
<span id="cb46-83"><a href="#cb46-83"></a>  <span class="co"># save just SNP IDs (for Plink to get per-population AFs)</span></span>
<span id="cb46-84"><a href="#cb46-84"></a>  out_path =<span class="st"> </span><span class="kw">file.path</span>(output_dir_snpids, <span class="kw">paste</span>(trait, <span class="st">&quot;.list&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb46-85"><a href="#cb46-85"></a>  <span class="co">## write file</span></span>
<span id="cb46-86"><a href="#cb46-86"></a>  readr<span class="op">::</span><span class="kw">write_lines</span>(out<span class="op">$</span>RANDOM_SNP, out_path)  </span>
<span id="cb46-87"><a href="#cb46-87"></a>})</span></code></pre></div>
</div>
<div id="filter-vcfs-for-random-snps" class="section level4">
<h4><span class="header-section-number">1.8.2.4</span> Filter VCFs for random SNPs</h4>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb47-1"><a href="#cb47-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb47-2"><a href="#cb47-2"></a><span class="va">ref=</span>../refs/hs37d5.fa.gz</span>
<span id="cb47-3"><a href="#cb47-3"></a><span class="va">in_vcf=</span>../vcfs/1gk_all.vcf.gz</span>
<span id="cb47-4"><a href="#cb47-4"></a><span class="va">snps_dir=</span>data/20210126_random_snps_snp_ids</span>
<span id="cb47-5"><a href="#cb47-5"></a><span class="va">out_dir=</span>data/20210127_snp_rndm_filtered</span>
<span id="cb47-6"><a href="#cb47-6"></a></span>
<span id="cb47-7"><a href="#cb47-7"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb47-8"><a href="#cb47-8"></a></span>
<span id="cb47-9"><a href="#cb47-9"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb47-10"><a href="#cb47-10"></a>  <span class="ex">bsub</span> \</span>
<span id="cb47-11"><a href="#cb47-11"></a>    -M 10000 \</span>
<span id="cb47-12"><a href="#cb47-12"></a>    -o ../log/20210127_extract_snps_<span class="va">$trait</span>.out \</span>
<span id="cb47-13"><a href="#cb47-13"></a>    -e ../log/20210127_extract_snps_<span class="va">$trait</span>.err \</span>
<span id="cb47-14"><a href="#cb47-14"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb47-15"><a href="#cb47-15"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb47-16"><a href="#cb47-16"></a><span class="st">    gatk SelectVariants \</span></span>
<span id="cb47-17"><a href="#cb47-17"></a><span class="st">      -R </span><span class="va">$ref</span><span class="st"> \</span></span>
<span id="cb47-18"><a href="#cb47-18"></a><span class="st">      -V </span><span class="va">$in_vcf</span><span class="st"> \</span></span>
<span id="cb47-19"><a href="#cb47-19"></a><span class="st">      --keep-ids </span><span class="va">$snps_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb47-20"><a href="#cb47-20"></a><span class="st">      -O </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz </span></span>
<span id="cb47-21"><a href="#cb47-21"></a><span class="st">    &quot;&quot;&quot;</span> <span class="kw">;</span></span>
<span id="cb47-22"><a href="#cb47-22"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="get-per-population-allele-frequencies-of-random-snps" class="section level4">
<h4><span class="header-section-number">1.8.2.5</span> Get per-population allele frequencies of random SNPs</h4>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb48-1"><a href="#cb48-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb48-2"><a href="#cb48-2"></a><span class="va">random_snps_dir=</span>data/20210126_random_snps_snp_ids</span>
<span id="cb48-3"><a href="#cb48-3"></a><span class="va">vcf_in_dir=</span>data/20210127_snp_rndm_filtered</span>
<span id="cb48-4"><a href="#cb48-4"></a><span class="va">popn_key=</span>data/plink2_sample_popn_key.txt</span>
<span id="cb48-5"><a href="#cb48-5"></a><span class="va">out_dir=</span>data/20210127_snp_rndm_alfreqs</span>
<span id="cb48-6"><a href="#cb48-6"></a></span>
<span id="cb48-7"><a href="#cb48-7"></a></span>
<span id="cb48-8"><a href="#cb48-8"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb48-9"><a href="#cb48-9"></a></span>
<span id="cb48-10"><a href="#cb48-10"></a>  <span class="fu">mkdir</span> -p <span class="va">$out_dir</span>/<span class="va">$trait</span></span>
<span id="cb48-11"><a href="#cb48-11"></a>  </span>
<span id="cb48-12"><a href="#cb48-12"></a>  <span class="ex">bsub</span> \</span>
<span id="cb48-13"><a href="#cb48-13"></a>    -M 10000 \</span>
<span id="cb48-14"><a href="#cb48-14"></a>    -o ../log/20210127_rdm_popn_afreqs_<span class="va">$trait</span>.out \</span>
<span id="cb48-15"><a href="#cb48-15"></a>    -e ../log/20210127_rdm_popn_afreqs_<span class="va">$trait</span>.err \</span>
<span id="cb48-16"><a href="#cb48-16"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb48-17"><a href="#cb48-17"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb48-18"><a href="#cb48-18"></a><span class="st">    plink2 \</span></span>
<span id="cb48-19"><a href="#cb48-19"></a><span class="st">      --vcf </span><span class="va">$vcf_in_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz \</span></span>
<span id="cb48-20"><a href="#cb48-20"></a><span class="st">      --extract </span><span class="va">$random_snps_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb48-21"><a href="#cb48-21"></a><span class="st">      --freq \</span></span>
<span id="cb48-22"><a href="#cb48-22"></a><span class="st">      --pheno iid-only </span><span class="va">$popn_key</span><span class="st"> \</span></span>
<span id="cb48-23"><a href="#cb48-23"></a><span class="st">      --loop-cats PHENO1 \</span></span>
<span id="cb48-24"><a href="#cb48-24"></a><span class="st">      --out </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">/</span><span class="va">$trait</span></span>
<span id="cb48-25"><a href="#cb48-25"></a><span class="st">    &quot;&quot;&quot;</span> <span class="kw">;</span></span>
<span id="cb48-26"><a href="#cb48-26"></a><span class="kw">done</span>  </span></code></pre></div>
</div>
<div id="bind-into-single-df" class="section level4">
<h4><span class="header-section-number">1.8.2.6</span> Bind into single DF</h4>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a>in_dir_rndm =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210126_random_snps&quot;</span>)</span>
<span id="cb49-2"><a href="#cb49-2"></a>in_dir_afreq =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210127_snp_rndm_alfreqs&quot;</span>)</span>
<span id="cb49-3"><a href="#cb49-3"></a></span>
<span id="cb49-4"><a href="#cb49-4"></a><span class="co"># Read in data</span></span>
<span id="cb49-5"><a href="#cb49-5"></a></span>
<span id="cb49-6"><a href="#cb49-6"></a><span class="co">## Random SNPs</span></span>
<span id="cb49-7"><a href="#cb49-7"></a>in_files_rndm =<span class="st"> </span><span class="kw">list.files</span>(in_dir_rndm, <span class="dt">full.names =</span> T)</span>
<span id="cb49-8"><a href="#cb49-8"></a><span class="kw">names</span>(in_files_rndm) =<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.txt&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">basename</span>(in_files_rndm))</span>
<span id="cb49-9"><a href="#cb49-9"></a></span>
<span id="cb49-10"><a href="#cb49-10"></a>random_snps =<span class="st"> </span><span class="kw">lapply</span>(in_files_rndm, <span class="cf">function</span>(file){</span>
<span id="cb49-11"><a href="#cb49-11"></a>  out =<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(file)</span>
<span id="cb49-12"><a href="#cb49-12"></a>  <span class="co"># add `POPN` column</span></span>
<span id="cb49-13"><a href="#cb49-13"></a>  out<span class="op">$</span>POPN =<span class="st"> &quot;all&quot;</span></span>
<span id="cb49-14"><a href="#cb49-14"></a>  </span>
<span id="cb49-15"><a href="#cb49-15"></a>  <span class="kw">return</span>(out)</span>
<span id="cb49-16"><a href="#cb49-16"></a>}) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-17"><a href="#cb49-17"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;PHENO&quot;</span>)</span>
<span id="cb49-18"><a href="#cb49-18"></a></span>
<span id="cb49-19"><a href="#cb49-19"></a><span class="co">## Popn afreqs</span></span>
<span id="cb49-20"><a href="#cb49-20"></a>popn_afreqs =<span class="st"> </span><span class="kw">lapply</span>(trait_levels, <span class="cf">function</span>(pheno){</span>
<span id="cb49-21"><a href="#cb49-21"></a>  target_files =<span class="st"> </span><span class="kw">list.files</span>(<span class="kw">file.path</span>(in_dir_afreq, pheno), <span class="dt">pattern =</span> <span class="st">&quot;.afreq&quot;</span>, <span class="dt">full.names =</span> T)</span>
<span id="cb49-22"><a href="#cb49-22"></a>  </span>
<span id="cb49-23"><a href="#cb49-23"></a>  <span class="kw">names</span>(target_files) =<span class="st"> </span><span class="kw">basename</span>(target_files) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-24"><a href="#cb49-24"></a><span class="st">    </span><span class="kw">str_split</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="dt">simplify =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-25"><a href="#cb49-25"></a><span class="st">    </span><span class="kw">subset</span>(<span class="dt">select =</span> <span class="dv">2</span>)</span>
<span id="cb49-26"><a href="#cb49-26"></a>  </span>
<span id="cb49-27"><a href="#cb49-27"></a>  popn_afreqs =<span class="st"> </span><span class="kw">lapply</span>(target_files, <span class="cf">function</span>(popn){</span>
<span id="cb49-28"><a href="#cb49-28"></a>    df =<span class="st"> </span><span class="kw">read_afreq</span>(popn)</span>
<span id="cb49-29"><a href="#cb49-29"></a>  }) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-30"><a href="#cb49-30"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;POPN&quot;</span>) <span class="co"># %&gt;% </span></span>
<span id="cb49-31"><a href="#cb49-31"></a>  <span class="co">#  dplyr::select(-OBS_CT) %&gt;% </span></span>
<span id="cb49-32"><a href="#cb49-32"></a>  <span class="co">#  tidyr::pivot_wider(id_cols = SNP, names_from = POPN, values_from = c(ALT_FREQS, OBS_CT))</span></span>
<span id="cb49-33"><a href="#cb49-33"></a>  </span>
<span id="cb49-34"><a href="#cb49-34"></a>  <span class="kw">return</span>(popn_afreqs)</span>
<span id="cb49-35"><a href="#cb49-35"></a>}) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb49-36"><a href="#cb49-36"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;PHENO&quot;</span>)</span>
<span id="cb49-37"><a href="#cb49-37"></a></span>
<span id="cb49-38"><a href="#cb49-38"></a><span class="co"># Bind `popn_afreqs` to `random_snps`</span></span>
<span id="cb49-39"><a href="#cb49-39"></a>random_afreqs =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">full_join</span>(dplyr<span class="op">::</span><span class="kw">select</span>(random_snps, <span class="op">-</span><span class="kw">c</span>(POPN, ALT_FREQS, OBS_CT)),</span>
<span id="cb49-40"><a href="#cb49-40"></a>                                 popn_afreqs,</span>
<span id="cb49-41"><a href="#cb49-41"></a>                                 <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;PHENO&quot;</span>, <span class="st">&quot;RANDOM_SNP&quot;</span> =<span class="st"> &quot;SNP&quot;</span>, <span class="st">&quot;REF&quot;</span>, <span class="st">&quot;ALT&quot;</span>, <span class="st">&quot;CHR&quot;</span>))</span>
<span id="cb49-42"><a href="#cb49-42"></a><span class="co"># Bind `all` AFs</span></span>
<span id="cb49-43"><a href="#cb49-43"></a>random_afreqs =<span class="st"> </span><span class="kw">rbind</span>(random_afreqs, random_snps)</span>
<span id="cb49-44"><a href="#cb49-44"></a></span>
<span id="cb49-45"><a href="#cb49-45"></a><span class="co">## Recode PHENO</span></span>
<span id="cb49-46"><a href="#cb49-46"></a><span class="co">#random_afreqs$PHENO = dplyr::recode(random_afreqs$PHENO, !!!rev_recode_vec)</span></span>
<span id="cb49-47"><a href="#cb49-47"></a></span>
<span id="cb49-48"><a href="#cb49-48"></a><span class="co">## Add `HIT_CONTROL` column</span></span>
<span id="cb49-49"><a href="#cb49-49"></a>random_afreqs<span class="op">$</span>HIT_CONTROL =<span class="st"> &quot;control&quot;</span></span></code></pre></div>
</div>
<div id="add-to-data_list" class="section level4">
<h4><span class="header-section-number">1.8.2.7</span> Add to <code>data_list</code></h4>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb50-2"><a href="#cb50-2"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb50-3"><a href="#cb50-3"></a>  <span class="co"># set counter</span></span>
<span id="cb50-4"><a href="#cb50-4"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb50-5"><a href="#cb50-5"></a>  <span class="co"># set target pheno</span></span>
<span id="cb50-6"><a href="#cb50-6"></a>  target_pheno =<span class="st"> </span><span class="kw">names</span>(data_list)[counter]</span>
<span id="cb50-7"><a href="#cb50-7"></a>  <span class="co"># add random afreqs</span></span>
<span id="cb50-8"><a href="#cb50-8"></a>  random_af =<span class="st"> </span>random_afreqs <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb50-9"><a href="#cb50-9"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(PHENO <span class="op">==</span><span class="st"> </span>target_pheno)</span>
<span id="cb50-10"><a href="#cb50-10"></a>  <span class="co"># recode PHENO</span></span>
<span id="cb50-11"><a href="#cb50-11"></a>  random_af<span class="op">$</span>PHENO =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">recode</span>(random_af<span class="op">$</span>PHENO, <span class="op">!!!</span>recode_vec)</span>
<span id="cb50-12"><a href="#cb50-12"></a>  <span class="co"># add to list</span></span>
<span id="cb50-13"><a href="#cb50-13"></a>  pheno[[<span class="st">&quot;random_af&quot;</span>]] =<span class="st"> </span>random_af</span>
<span id="cb50-14"><a href="#cb50-14"></a>  </span>
<span id="cb50-15"><a href="#cb50-15"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb50-16"><a href="#cb50-16"></a>})</span></code></pre></div>
</div>
</div>
</div>
<div id="clump-to-get-lead-snps" class="section level2">
<h2><span class="header-section-number">1.9</span> Clump to get lead SNPs</h2>
=======
<div id="write-population-file-for-plink2" class="section level3" number="1.7.2">
<h3 number="1.7.2"><span class="header-section-number">1.7.2</span> Write population file for Plink2</h3>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>sample_popn_key_file <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;plink2_sample_popn_key.txt&quot;</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(meta[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>            sample_popn_key_file,</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> F,</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> F,</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> F)</span></code></pre></div>
</div>
<div id="run-plink2-for-snp-hits" class="section level3" number="1.7.3">
<h3 number="1.7.3"><span class="header-section-number">1.7.3</span> Run <code>Plink2</code> for SNP hits</h3>
<p>(Take only biallelic SNPs.)</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate fst_env_rhel</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set variables</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="va">in_vcf_dir=</span>data/20210125_snp_hits_filtered</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="va">popn_file=</span>data/plink2_sample_popn_key.txt</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir=</span>data/20210122_snp_hits_alfreqs</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up directories</span></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mkdir</span> -p <span class="va">$out_dir</span>/<span class="va">$trait</span><span class="kw">;</span> </span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span>   </span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Plink2</span></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a><span class="co">## Get global AF</span></span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>  <span class="ex">plink2</span> <span class="kw">\</span></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--vcf</span> <span class="va">$in_vcf_dir</span>/<span class="va">$trait</span>.vcf.gz <span class="kw">\</span></span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--freq</span> <span class="kw">\</span></span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--max-alleles</span> 2 <span class="kw">\</span></span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--snps-only</span> <span class="kw">\</span></span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--out</span> <span class="va">$out_dir</span>/<span class="va">$trait</span>/<span class="va">$trait</span>.all</span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span></span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a><span class="co">## Get AF per population</span></span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>  <span class="ex">plink2</span> <span class="kw">\</span></span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--vcf</span> <span class="va">$in_vcf_dir</span>/<span class="va">$trait</span>.vcf.gz <span class="kw">\</span></span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--freq</span> <span class="kw">\</span></span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--max-alleles</span> 2 <span class="kw">\</span></span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--snps-only</span> <span class="kw">\</span></span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--pheno</span> iid-only <span class="va">$popn_file</span> <span class="kw">\</span></span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--loop-cats</span> PHENO1 <span class="kw">\</span></span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a>    <span class="ex">--out</span> <span class="va">$out_dir</span>/<span class="va">$trait</span>/<span class="va">$trait</span> <span class="kw">;</span></span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="add-allele-frequency-files" class="section level3" number="1.7.4">
<h3 number="1.7.4"><span class="header-section-number">1.7.4</span> Add allele frequency files</h3>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>target_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210122_snp_hits_alfreqs&quot;</span>)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Global</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter </span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get trait name</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get file path</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>  target_path <span class="ot">=</span> <span class="fu">file.path</span>(target_dir, trait, <span class="fu">paste</span>(trait, <span class="st">&quot;.all.afreq&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read in data</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  clean_af <span class="ot">=</span> <span class="fu">read_afreq</span>(target_path)</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add POPN column</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>  clean_af<span class="sc">$</span>POPN <span class="ot">=</span> <span class="st">&quot;all&quot;</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add to list</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;clean_af&quot;</span>]] <span class="ot">=</span> clean_af</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Per population</span></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter</span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get trait name</span></span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get file path</span></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>  target_files <span class="ot">=</span> <span class="fu">list.files</span>(<span class="fu">file.path</span>(target_dir, trait),</span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>                            <span class="at">pattern =</span> <span class="st">&quot;.*[^all]</span><span class="sc">\\</span><span class="st">.afreq&quot;</span>,</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>                            <span class="at">full.names =</span> T)</span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get popn names</span></span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(target_files) <span class="ot">=</span> <span class="fu">basename</span>(target_files) <span class="sc">%&gt;%</span> </span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_split</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="at">simplify =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(<span class="at">select =</span> <span class="dv">2</span>)</span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read files and bind into single DF</span></span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>  popn_afreqs <span class="ot">=</span> <span class="fu">lapply</span>(target_files, <span class="cf">function</span>(popn){</span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">=</span> <span class="fu">read_afreq</span>(popn)</span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span> </span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;POPN&quot;</span>)<span class="co"># %&gt;% </span></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a><span class="co">#    dplyr::select(-OBS_CT) %&gt;% </span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a><span class="co">#    tidyr::pivot_wider(id_cols = SNP, names_from = POPN, values_from = ALT_FREQS)</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># combine with `clean_af`</span></span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;clean_af&quot;</span>]] <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(pheno[[<span class="st">&quot;clean_af&quot;</span>]],</span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a>                                         popn_afreqs)</span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
</div>
<div id="set-up-negative-controls" class="section level2" number="1.8">
<h2 number="1.8"><span class="header-section-number">1.8</span> Set up negative controls</h2>
<p>Pull out random SNPs with the same global allele frequencies as the GWAS SNP-hits</p>
<div id="bin-snp-hits-by-allele-frequency" class="section level3" number="1.8.1">
<h3 number="1.8.1"><span class="header-section-number">1.8.1</span> Bin SNP hits by allele frequency</h3>
<p>Bind to clean DF to get AFs of risk allele</p>
<p><span style="color: red;"><strong>NOTE</strong></span>: If <code>RISK_ALLELE</code> is unknown, set the allele frequency to <code>ALT_FREQS</code></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># join DFs</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(pheno[[<span class="st">&quot;clean&quot;</span>]],</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>                        dplyr<span class="sc">::</span><span class="fu">select</span>(pheno[[<span class="st">&quot;clean_af&quot;</span>]],</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="sc">-</span>CHR),</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;TOP_SNP&quot;</span> <span class="ot">=</span> <span class="st">&quot;SNP&quot;</span>))</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get AF of risk allele</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>RISK_AF <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(df<span class="sc">$</span>RISK_ALLELE <span class="sc">==</span> <span class="st">&quot;?&quot;</span>,</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>                              df<span class="sc">$</span>ALT_FREQS,</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>                              dplyr<span class="sc">::</span><span class="fu">if_else</span>(df<span class="sc">$</span>RISK_ALLELE <span class="sc">==</span> df<span class="sc">$</span>ALT,</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>                                             df<span class="sc">$</span>ALT_FREQS,</span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="dv">1</span> <span class="sc">-</span> df<span class="sc">$</span>ALT_FREQS))</span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add `HIT_CONTROL` column</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>HIT_CONTROL <span class="ot">=</span> <span class="st">&quot;hit&quot;</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add to list</span></span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;consol&quot;</span>]] <span class="ot">=</span> df</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>Bin by risk allele frequency</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1% intervals</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>breakpoints <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.1</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># choose DF</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> pheno[[<span class="st">&quot;consol&quot;</span>]]</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add bins</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>BIN_10 <span class="ot">=</span> <span class="fu">cut</span>(df<span class="sc">$</span>RISK_AF, <span class="at">breaks =</span> breakpoints, <span class="at">labels =</span> F)</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save back into list</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;consol&quot;</span>]] <span class="ot">=</span> df</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>Extract key columns and write to file</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_snp_risk_hits_binned&quot;</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(out_dir)</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Save list</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>risk_afs <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter </span></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get target DF</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> pheno[[<span class="st">&quot;consol&quot;</span>]]</span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter</span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(POPN <span class="sc">==</span> <span class="st">&quot;all&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># take only global AFs </span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(TOP_SNP, BIN_10) <span class="sc">%&gt;%</span> </span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="co"># drop NAs</span></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set output path</span></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>  path_out <span class="ot">=</span> <span class="fu">file.path</span>(out_dir, <span class="fu">paste</span>(trait, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># write to file</span></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_tsv</span>(df, path_out)</span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
<div id="bin-1kg-snps" class="section level3" number="1.8.2">
<h3 number="1.8.2"><span class="header-section-number">1.8.2</span> Bin 1KG SNPs</h3>
<div id="get-allele-frequencies-from-1kg" class="section level4" number="1.8.2.1">
<h4 number="1.8.2.1"><span class="header-section-number">1.8.2.1</span> Get allele frequencies from 1KG</h4>
<p>With <code>Plink2</code>, per chromosome for speed.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set output directory</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="va">in_file=</span>../vcfs/1kg_all.vcf.gz</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir=</span>../big_data/20210125_alfreqs_all</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Per chromosome</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">chr</span> in <span class="va">$(</span><span class="fu">seq</span> 1 22<span class="va">)</span> <span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create allele-freq tables</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bsub</span> <span class="kw">\</span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-M</span> 10000 <span class="kw">\</span></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-o</span> ../log/20210125_plink_alfreq_<span class="va">$chr</span>.out <span class="kw">\</span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-e</span> ../log/20210125_plink_alfreq_<span class="va">$chr</span>.err <span class="kw">\</span></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a><span class="st">    plink2 \</span></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a><span class="st">      --vcf </span><span class="va">$in_file</span><span class="st"> \</span></span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a><span class="st">      --freq \</span></span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a><span class="st">      --chr </span><span class="va">$chr</span><span class="st"> \</span></span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a><span class="st">      --max-alleles 2 \</span></span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a><span class="st">      --snps-only \</span></span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a><span class="st">      --out </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$chr</span><span class="st"> &quot;</span>;</span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span> </span></code></pre></div>
</div>
<div id="bin-them-and-save-to-single-file" class="section level4" number="1.8.2.2">
<h4 number="1.8.2.2"><span class="header-section-number">1.8.2.2</span> Bin them and save to single file</h4>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># On cluster</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co"># `conda activate fst_env_rhel`</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;code&quot;</span>, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;source.R&quot;</span>))</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Set variables</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>in_dir <span class="ot">=</span> <span class="st">&quot;../big_data/20210125_alfreqs_all&quot;</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">=</span> <span class="st">&quot;../big_data/20210201_alfreqs_all_binned_10&quot;</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>breakpoints <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.1</span>) <span class="co"># 10% bins</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Create output directory</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(out_dir)</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Get list of input files, excluding </span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>in_files <span class="ot">=</span> <span class="fu">list.files</span>(in_dir, <span class="at">pattern =</span> <span class="st">&quot;.afreq&quot;</span>, <span class="at">full.names =</span> T)</span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in files, add bins, and write to output</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a>freq_list <span class="ot">=</span> <span class="fu">lapply</span>(in_files, <span class="cf">function</span>(chr_file){</span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read in file</span></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">read_afreq</span>(chr_file)</span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add bins</span></span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>BIN_10 <span class="ot">=</span> <span class="fu">cut</span>(df<span class="sc">$</span>ALT_FREQS, <span class="at">breaks =</span> breakpoints, <span class="at">labels =</span> F)</span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove &quot;ss&quot; SNPs</span></span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;ss&quot;</span>, SNP))</span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># write file</span></span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_tsv</span>(df, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="fu">basename</span>(chr_file)))</span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine into single DF</span></span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a>freq_df <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(freq_list)</span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Write to file</span></span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(freq_df, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="st">&quot;all.afreq&quot;</span>))</span></code></pre></div>
</div>
<div id="pull-out-random-snps-with-same-af-as-trait-risk-alleles" class="section level4" number="1.8.2.3">
<h4 number="1.8.2.3"><span class="header-section-number">1.8.2.3</span> Pull out random SNPs with same AF as trait risk alleles</h4>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># On cluster</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;code&quot;</span>, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;source.R&quot;</span>))</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Variables</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>input_risk_snp_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_snp_risk_hits_binned&quot;</span>)</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>all_1kg_bins <span class="ot">=</span> <span class="st">&quot;../big_data/20210201_alfreqs_all_binned_10/all.afreq&quot;</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>initial_seed <span class="ot">=</span> <span class="dv">123</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_random_snps&quot;</span>)</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>output_dir_snpids <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_random_snps_snp_ids&quot;</span>)</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(output_dir)</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(output_dir_snpids)</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in target SNP DF and split into list by bin</span></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>phenos <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.txt&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">basename</span>(<span class="fu">list.files</span>(input_risk_snp_dir)))</span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(phenos) <span class="ot">=</span> phenos</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>risk_list <span class="ot">=</span> <span class="fu">lapply</span>(phenos, <span class="cf">function</span>(pheno){</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set file path</span></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a>  file_path <span class="ot">=</span> <span class="fu">file.path</span>(input_risk_snp_dir, <span class="fu">paste</span>(pheno, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read file</span></span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(file_path,</span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_names =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb46-27"><a href="#cb46-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">split</span>(., <span class="at">f =</span> .<span class="sc">$</span>BIN_10) <span class="co"># split by bin</span></span>
<span id="cb46-28"><a href="#cb46-28" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb46-29"><a href="#cb46-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-30"><a href="#cb46-30" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in 1KG data</span></span>
<span id="cb46-31"><a href="#cb46-31" aria-hidden="true" tabindex="-1"></a>freq_df <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(all_1kg_bins)</span>
<span id="cb46-32"><a href="#cb46-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-33"><a href="#cb46-33" aria-hidden="true" tabindex="-1"></a><span class="co"># For each bin in `risk_list`, pull out the same number of random number 1KG SNPs with the same bin</span></span>
<span id="cb46-34"><a href="#cb46-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-35"><a href="#cb46-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Set seed</span></span>
<span id="cb46-36"><a href="#cb46-36" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(initial_seed)</span>
<span id="cb46-37"><a href="#cb46-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-38"><a href="#cb46-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Get seeds for each bin</span></span>
<span id="cb46-39"><a href="#cb46-39" aria-hidden="true" tabindex="-1"></a>seeds <span class="ot">=</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>, <span class="fu">length</span>(risk_list))</span>
<span id="cb46-40"><a href="#cb46-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-41"><a href="#cb46-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Run over list</span></span>
<span id="cb46-42"><a href="#cb46-42" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb46-43"><a href="#cb46-43" aria-hidden="true" tabindex="-1"></a>test <span class="ot">=</span> <span class="fu">lapply</span>(risk_list, <span class="cf">function</span>(pheno){</span>
<span id="cb46-44"><a href="#cb46-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter </span></span>
<span id="cb46-45"><a href="#cb46-45" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span>  </span>
<span id="cb46-46"><a href="#cb46-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set seed for pheno</span></span>
<span id="cb46-47"><a href="#cb46-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(seeds[counter])</span>
<span id="cb46-48"><a href="#cb46-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get seeds for bin</span></span>
<span id="cb46-49"><a href="#cb46-49" aria-hidden="true" tabindex="-1"></a>  bin_seeds <span class="ot">=</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>, <span class="fu">length</span>(pheno))</span>
<span id="cb46-50"><a href="#cb46-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get random SNPs from each bin</span></span>
<span id="cb46-51"><a href="#cb46-51" aria-hidden="true" tabindex="-1"></a>  bin_counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb46-52"><a href="#cb46-52" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> <span class="fu">lapply</span>(pheno, <span class="cf">function</span>(bin_df){</span>
<span id="cb46-53"><a href="#cb46-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set `bin_counter`</span></span>
<span id="cb46-54"><a href="#cb46-54" aria-hidden="true" tabindex="-1"></a>    bin_counter <span class="ot">&lt;&lt;-</span> bin_counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb46-55"><a href="#cb46-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get target bin</span></span>
<span id="cb46-56"><a href="#cb46-56" aria-hidden="true" tabindex="-1"></a>    target_bin <span class="ot">=</span> <span class="fu">as.integer</span>(<span class="fu">names</span>(pheno)[bin_counter])</span>
<span id="cb46-57"><a href="#cb46-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get number of matches required</span></span>
<span id="cb46-58"><a href="#cb46-58" aria-hidden="true" tabindex="-1"></a>    hits_n <span class="ot">=</span> <span class="fu">nrow</span>(bin_df)</span>
<span id="cb46-59"><a href="#cb46-59" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set seed</span></span>
<span id="cb46-60"><a href="#cb46-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(bin_seeds[bin_counter])    </span>
<span id="cb46-61"><a href="#cb46-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter 1kg DF for SNPs with same bin and get random hits</span></span>
<span id="cb46-62"><a href="#cb46-62" aria-hidden="true" tabindex="-1"></a>    random_hits <span class="ot">=</span> freq_df <span class="sc">%&gt;%</span> </span>
<span id="cb46-63"><a href="#cb46-63" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(BIN_10 <span class="sc">==</span> target_bin) <span class="sc">%&gt;%</span> </span>
<span id="cb46-64"><a href="#cb46-64" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> hits_n) <span class="sc">%&gt;%</span> </span>
<span id="cb46-65"><a href="#cb46-65" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">RANDOM_SNP =</span> SNP,</span>
<span id="cb46-66"><a href="#cb46-66" aria-hidden="true" tabindex="-1"></a>                    <span class="at">RANDOM_BIN_10 =</span> BIN_10)</span>
<span id="cb46-67"><a href="#cb46-67" aria-hidden="true" tabindex="-1"></a>    <span class="co"># bind `random_hits` to target SNP df</span></span>
<span id="cb46-68"><a href="#cb46-68" aria-hidden="true" tabindex="-1"></a>    df_out <span class="ot">=</span> <span class="fu">cbind</span>(bin_df, random_hits)</span>
<span id="cb46-69"><a href="#cb46-69" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb46-70"><a href="#cb46-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df_out)</span>
<span id="cb46-71"><a href="#cb46-71" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span> </span>
<span id="cb46-72"><a href="#cb46-72" aria-hidden="true" tabindex="-1"></a>    <span class="co"># bind into single data frame</span></span>
<span id="cb46-73"><a href="#cb46-73" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb46-74"><a href="#cb46-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-75"><a href="#cb46-75" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save to file</span></span>
<span id="cb46-76"><a href="#cb46-76" aria-hidden="true" tabindex="-1"></a>  <span class="do">## set output path</span></span>
<span id="cb46-77"><a href="#cb46-77" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(risk_list)[counter]</span>
<span id="cb46-78"><a href="#cb46-78" aria-hidden="true" tabindex="-1"></a>  out_path <span class="ot">=</span> <span class="fu">file.path</span>(output_dir, <span class="fu">paste</span>(trait, <span class="st">&quot;.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb46-79"><a href="#cb46-79" aria-hidden="true" tabindex="-1"></a>  <span class="do">## write file</span></span>
<span id="cb46-80"><a href="#cb46-80" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_tsv</span>(out, out_path)</span>
<span id="cb46-81"><a href="#cb46-81" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-82"><a href="#cb46-82" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save just SNP IDs (for Plink to get per-population AFs)</span></span>
<span id="cb46-83"><a href="#cb46-83" aria-hidden="true" tabindex="-1"></a>  out_path <span class="ot">=</span> <span class="fu">file.path</span>(output_dir_snpids, <span class="fu">paste</span>(trait, <span class="st">&quot;.list&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb46-84"><a href="#cb46-84" aria-hidden="true" tabindex="-1"></a>  <span class="do">## write file</span></span>
<span id="cb46-85"><a href="#cb46-85" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_lines</span>(out<span class="sc">$</span>RANDOM_SNP, out_path)</span>
<span id="cb46-86"><a href="#cb46-86" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-87"><a href="#cb46-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb46-88"><a href="#cb46-88" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
<div id="filter-vcfs-for-random-snps" class="section level4" number="1.8.2.4">
<h4 number="1.8.2.4"><span class="header-section-number">1.8.2.4</span> Filter VCFs for random SNPs</h4>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="va">ref=</span>../refs/hs37d5.fa.gz</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="va">in_vcf=</span>../vcfs/1kg_all.vcf.gz</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="va">snps_dir=</span>data/20210201_random_snps_snp_ids</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir=</span>data/20210201_snp_rndm_filtered</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bsub</span> <span class="kw">\</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-M</span> 10000 <span class="kw">\</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-o</span> ../log/20210201_extract_snps_<span class="va">$trait</span>.out <span class="kw">\</span></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-e</span> ../log/20210201_extract_snps_<span class="va">$trait</span>.err <span class="kw">\</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a><span class="st">    gatk SelectVariants \</span></span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="st">      -R </span><span class="va">$ref</span><span class="st"> \</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="st">      -V </span><span class="va">$in_vcf</span><span class="st"> \</span></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a><span class="st">      --keep-ids </span><span class="va">$snps_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a><span class="st">      -O </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz </span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a><span class="st">    &quot;&quot;&quot;</span> ;</span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="get-per-population-allele-frequencies-of-random-snps" class="section level4" number="1.8.2.5">
<h4 number="1.8.2.5"><span class="header-section-number">1.8.2.5</span> Get per-population allele frequencies of random SNPs</h4>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="va">random_snps_dir=</span>data/20210201_random_snps_snp_ids</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="va">vcf_in_dir=</span>data/20210201_snp_rndm_filtered</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="va">popn_key=</span>data/plink2_sample_popn_key.txt</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir=</span>data/20210201_snp_rndm_alfreqs</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mkdir</span> -p <span class="va">$out_dir</span>/<span class="va">$trait</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bsub</span> <span class="kw">\</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-M</span> 10000 <span class="kw">\</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-o</span> ../log/20210201_rdm_popn_afreqs_<span class="va">$trait</span>.out <span class="kw">\</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-e</span> ../log/20210201_rdm_popn_afreqs_<span class="va">$trait</span>.err <span class="kw">\</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a><span class="st">    plink2 \</span></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a><span class="st">      --vcf </span><span class="va">$vcf_in_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz \</span></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="st">      --extract </span><span class="va">$random_snps_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="st">      --freq \</span></span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a><span class="st">      --pheno iid-only </span><span class="va">$popn_key</span><span class="st"> \</span></span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a><span class="st">      --loop-cats PHENO1 \</span></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a><span class="st">      --out </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">/</span><span class="va">$trait</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a><span class="st">    &quot;&quot;&quot;</span> ;</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="bind-into-single-df" class="section level4" number="1.8.2.6">
<h4 number="1.8.2.6"><span class="header-section-number">1.8.2.6</span> Bind into single DF</h4>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>in_dir_rndm <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_random_snps&quot;</span>)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>in_dir_afreq <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_snp_rndm_alfreqs&quot;</span>)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in data</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Random SNPs</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>in_files_rndm <span class="ot">=</span> <span class="fu">list.files</span>(in_dir_rndm, <span class="at">full.names =</span> T)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(in_files_rndm) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.txt&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">basename</span>(in_files_rndm))</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>random_snps <span class="ot">=</span> <span class="fu">lapply</span>(in_files_rndm, <span class="cf">function</span>(file){</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(file)</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add `POPN` column</span></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>  out<span class="sc">$</span>POPN <span class="ot">=</span> <span class="st">&quot;all&quot;</span></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> </span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;PHENO&quot;</span>)</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Popn afreqs</span></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>popn_afreqs <span class="ot">=</span> <span class="fu">lapply</span>(trait_levels, <span class="cf">function</span>(pheno){</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>  target_files <span class="ot">=</span> <span class="fu">list.files</span>(<span class="fu">file.path</span>(in_dir_afreq, pheno), <span class="at">pattern =</span> <span class="st">&quot;.afreq&quot;</span>, <span class="at">full.names =</span> T)</span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(target_files) <span class="ot">=</span> <span class="fu">basename</span>(target_files) <span class="sc">%&gt;%</span> </span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_split</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="at">simplify =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(<span class="at">select =</span> <span class="dv">2</span>)</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>  popn_afreqs <span class="ot">=</span> <span class="fu">lapply</span>(target_files, <span class="cf">function</span>(popn){</span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">=</span> <span class="fu">read_afreq</span>(popn)</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span> </span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;POPN&quot;</span>)</span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(popn_afreqs)</span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> </span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;PHENO&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">drop_na</span>()</span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Bind `popn_afreqs` to `random_snps`</span></span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a>random_afreqs <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(random_snps, <span class="sc">-</span><span class="fu">c</span>(POPN, ALT_FREQS, OBS_CT)),</span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a>                                 popn_afreqs,</span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;PHENO&quot;</span>, <span class="st">&quot;RANDOM_SNP&quot;</span> <span class="ot">=</span> <span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;REF&quot;</span>, <span class="st">&quot;ALT&quot;</span>, <span class="st">&quot;CHR&quot;</span>))</span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Bind `all` AFs</span></span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a>random_afreqs <span class="ot">=</span> <span class="fu">rbind</span>(random_afreqs, random_snps)</span>
<span id="cb49-43"><a href="#cb49-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-44"><a href="#cb49-44" aria-hidden="true" tabindex="-1"></a><span class="do">## Add `HIT_CONTROL` column</span></span>
<span id="cb49-45"><a href="#cb49-45" aria-hidden="true" tabindex="-1"></a>random_afreqs<span class="sc">$</span>HIT_CONTROL <span class="ot">=</span> <span class="st">&quot;control&quot;</span></span></code></pre></div>
</div>
<div id="add-to-data_list" class="section level4" number="1.8.2.7">
<h4 number="1.8.2.7"><span class="header-section-number">1.8.2.7</span> Add to <code>data_list</code></h4>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set target pheno</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>  target_pheno <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add random afreqs</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>  random_af <span class="ot">=</span> random_afreqs <span class="sc">%&gt;%</span> </span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(PHENO <span class="sc">==</span> target_pheno)</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># recode PHENO</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>  random_af<span class="sc">$</span>PHENO <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">recode</span>(random_af<span class="sc">$</span>PHENO, <span class="sc">!!!</span>recode_vec)</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add to list</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;random_af&quot;</span>]] <span class="ot">=</span> random_af</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
</div>
</div>
<div id="clump-to-get-lead-snps" class="section level2" number="1.9">
<h2 number="1.9"><span class="header-section-number">1.9</span> Clump to get lead SNPs</h2>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<p>Use <code>Plink1.9</code> (<code>Plink2.0</code> doesn’t have a <code>clump</code> function.)</p>
<p>From the <code>Plink1.7</code> documentation (<a href="http://zzz.bwh.harvard.edu/plink/clump.shtml" class="uri">http://zzz.bwh.harvard.edu/plink/clump.shtml</a>), which applies to <code>Plink1.9</code>:</p>
<blockquote>
<p>The clumping procedure takes all SNPs that are significant at threshold p1 that have not already been clumped (denoting these as index SNPs) and forms clumps of all other SNPs that are within a certain kb distance from the index SNP (default 250kb) and that are in linkage disequilibrium with the index SNP, based on an r-squared threshold (default 0.50)… This is a greedy algorithm and so each SNP will only appear in a single clump, if at all.</p>
</blockquote>
<blockquote>
<p>…[t]he TOTAL field lists all SNPs that are clumped with the index SNP, irrespective of the p-value for those SNPs. This number is then split into those clumped SNPs that are not significant (p&gt;0.05) and various other groups defined by significance thresholds. For SNPs that are significant at the p2 threshold, they are listed explicitly. The (1) after each SNP name refers to the results file they came from (in this case, there is only a single result file specified, so all values are 1).</p>
</blockquote>
<p>Here, we’re taking all SNPs with <em>P</em> &lt; 1e-08 as index SNPs, and it will explicitly list all SNPs within the clump that also meet that threshold.</p>
<<<<<<< HEAD
<div class="sourceCode" id="cb51"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb51-1"><a href="#cb51-1"></a></span>
<span id="cb51-2"><a href="#cb51-2"></a><span class="co"># Activate environment</span></span>
<span id="cb51-3"><a href="#cb51-3"></a><span class="ex">conda</span> activate fst_env_rhel</span>
<span id="cb51-4"><a href="#cb51-4"></a></span>
<span id="cb51-5"><a href="#cb51-5"></a><span class="co"># Set variables</span></span>
<span id="cb51-6"><a href="#cb51-6"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb51-7"><a href="#cb51-7"></a><span class="va">in_vcf_dir=</span>data/20210125_snp_hits_filtered</span>
<span id="cb51-8"><a href="#cb51-8"></a><span class="va">snp_p_dir=</span>data/20210122_snp_hit_lists</span>
<span id="cb51-9"><a href="#cb51-9"></a><span class="va">out_dir=</span>data/20210125_clumped</span>
<span id="cb51-10"><a href="#cb51-10"></a></span>
<span id="cb51-11"><a href="#cb51-11"></a><span class="va">r2_params=$(</span><span class="bu">echo</span> 0.1 0.2 0.3<span class="va">)</span></span>
<span id="cb51-12"><a href="#cb51-12"></a><span class="va">kb_params=$(</span><span class="bu">echo</span> 500 750 1000 <span class="va">)</span></span>
<span id="cb51-13"><a href="#cb51-13"></a></span>
<span id="cb51-14"><a href="#cb51-14"></a><span class="co"># Make directory</span></span>
<span id="cb51-15"><a href="#cb51-15"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb51-16"><a href="#cb51-16"></a></span>
<span id="cb51-17"><a href="#cb51-17"></a><span class="co"># Run with different parameters</span></span>
<span id="cb51-18"><a href="#cb51-18"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb51-19"><a href="#cb51-19"></a></span>
<span id="cb51-20"><a href="#cb51-20"></a>  <span class="fu">mkdir</span> -p <span class="va">$out_dir</span>/<span class="va">$trait</span> <span class="kw">;</span></span>
<span id="cb51-21"><a href="#cb51-21"></a>  </span>
<span id="cb51-22"><a href="#cb51-22"></a>  <span class="kw">for</span> <span class="ex">r2</span> in <span class="va">$r2_params</span> <span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb51-23"><a href="#cb51-23"></a>    <span class="kw">for</span> <span class="ex">kb</span> in <span class="va">$kb_params</span> <span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb51-24"><a href="#cb51-24"></a>      <span class="ex">bsub</span> \</span>
<span id="cb51-25"><a href="#cb51-25"></a>        -o ../log/20210125_clump_<span class="va">$trait</span>\_<span class="va">$r2</span>\_<span class="va">$kb</span>.out \</span>
<span id="cb51-26"><a href="#cb51-26"></a>        -e ../log/20210125_clump_<span class="va">$trait</span>\_<span class="va">$r2</span>\_<span class="va">$kb</span>.err \</span>
<span id="cb51-27"><a href="#cb51-27"></a>        <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb51-28"><a href="#cb51-28"></a><span class="st">        conda activate fst_env_rhel ;</span></span>
<span id="cb51-29"><a href="#cb51-29"></a><span class="st">        plink \</span></span>
<span id="cb51-30"><a href="#cb51-30"></a><span class="st">          --vcf </span><span class="va">$in_vcf_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz \</span></span>
<span id="cb51-31"><a href="#cb51-31"></a><span class="st">          --clump </span><span class="va">$snp_p_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">\_with_P.txt \</span></span>
<span id="cb51-32"><a href="#cb51-32"></a><span class="st">          --clump-p1 0.00000001 \</span></span>
<span id="cb51-33"><a href="#cb51-33"></a><span class="st">          --clump-p2 0.00000001 \</span></span>
<span id="cb51-34"><a href="#cb51-34"></a><span class="st">          --clump-r2 </span><span class="va">$r2</span><span class="st"> \</span></span>
<span id="cb51-35"><a href="#cb51-35"></a><span class="st">          --clump-kb </span><span class="va">$kb</span><span class="st"> \</span></span>
<span id="cb51-36"><a href="#cb51-36"></a><span class="st">          --out </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">/r2-</span><span class="va">$r2</span><span class="st">\_kb-</span><span class="va">$kb</span><span class="st"> </span></span>
<span id="cb51-37"><a href="#cb51-37"></a><span class="st">        &quot;&quot;&quot;</span>  <span class="kw">;</span></span>
<span id="cb51-38"><a href="#cb51-38"></a>    <span class="kw">done</span> ;</span>
<span id="cb51-39"><a href="#cb51-39"></a>  <span class="kw">done</span>;  </span>
<span id="cb51-40"><a href="#cb51-40"></a><span class="kw">done</span></span></code></pre></div>
<div id="add-clumped-snp-files-to-data_list" class="section level3">
<h3><span class="header-section-number">1.9.1</span> Add clumped SNP files to <code>data_list</code></h3>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a>target_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210125_clumped&quot;</span>)</span>
<span id="cb52-2"><a href="#cb52-2"></a></span>
<span id="cb52-3"><a href="#cb52-3"></a>counter &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb52-4"><a href="#cb52-4"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb52-5"><a href="#cb52-5"></a>  <span class="co"># set counter</span></span>
<span id="cb52-6"><a href="#cb52-6"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb52-7"><a href="#cb52-7"></a>  <span class="co"># get trait name</span></span>
<span id="cb52-8"><a href="#cb52-8"></a>  trait =<span class="st"> </span><span class="kw">names</span>(data_list)[counter]</span>
<span id="cb52-9"><a href="#cb52-9"></a>  <span class="co"># get file path</span></span>
<span id="cb52-10"><a href="#cb52-10"></a>  target_files =<span class="st"> </span><span class="kw">list.files</span>(<span class="kw">file.path</span>(target_dir, trait),</span>
<span id="cb52-11"><a href="#cb52-11"></a>                            <span class="dt">pattern =</span> <span class="st">&quot;.clumped&quot;</span>,</span>
<span id="cb52-12"><a href="#cb52-12"></a>                            <span class="dt">full.names =</span> T)</span>
<span id="cb52-13"><a href="#cb52-13"></a>  <span class="kw">names</span>(target_files) =<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.clumped&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">basename</span>(target_files))</span>
<span id="cb52-14"><a href="#cb52-14"></a>  <span class="co"># read files as `clumped`</span></span>
<span id="cb52-15"><a href="#cb52-15"></a>  counter_clump &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb52-16"><a href="#cb52-16"></a>  clumped =<span class="st"> </span><span class="kw">lapply</span>(target_files, <span class="cf">function</span>(params){</span>
<span id="cb52-17"><a href="#cb52-17"></a>    <span class="co"># set counter</span></span>
<span id="cb52-18"><a href="#cb52-18"></a>    counter_clump &lt;&lt;-<span class="st"> </span>counter_clump <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb52-19"><a href="#cb52-19"></a>    <span class="co"># split params string</span></span>
<span id="cb52-20"><a href="#cb52-20"></a>    param_str =<span class="st"> </span><span class="kw">names</span>(target_files)[counter_clump] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb52-21"><a href="#cb52-21"></a><span class="st">      </span>stringr<span class="op">::</span><span class="kw">str_split</span>(<span class="st">&quot;_&quot;</span>, <span class="dt">simplify =</span> T) <span class="op">%&gt;%</span></span>
<span id="cb52-22"><a href="#cb52-22"></a><span class="st">      </span>stringr<span class="op">::</span><span class="kw">str_split</span>(<span class="st">&quot;-&quot;</span>, <span class="dt">simplify =</span> T)</span>
<span id="cb52-23"><a href="#cb52-23"></a>    <span class="co"># get params</span></span>
<span id="cb52-24"><a href="#cb52-24"></a>    r2 =<span class="st"> </span><span class="kw">as.numeric</span>(param_str[<span class="dv">1</span>,<span class="dv">2</span>])</span>
<span id="cb52-25"><a href="#cb52-25"></a>    kb =<span class="st"> </span><span class="kw">as.integer</span>(param_str[<span class="dv">2</span>,<span class="dv">2</span>])</span>
<span id="cb52-26"><a href="#cb52-26"></a>    <span class="co"># read files</span></span>
<span id="cb52-27"><a href="#cb52-27"></a>    df =<span class="st"> </span><span class="kw">read.table</span>(params, <span class="dt">header =</span> T)</span>
<span id="cb52-28"><a href="#cb52-28"></a>    <span class="co"># add params to DF</span></span>
<span id="cb52-29"><a href="#cb52-29"></a>    df<span class="op">$</span>r2 =<span class="st"> </span>r2</span>
<span id="cb52-30"><a href="#cb52-30"></a>    df<span class="op">$</span>kb =<span class="st"> </span>kb</span>
<span id="cb52-31"><a href="#cb52-31"></a>    </span>
<span id="cb52-32"><a href="#cb52-32"></a>    <span class="kw">return</span>(df)</span>
<span id="cb52-33"><a href="#cb52-33"></a>  })</span>
<span id="cb52-34"><a href="#cb52-34"></a>  <span class="co"># add `clumped` list to `data_list`</span></span>
<span id="cb52-35"><a href="#cb52-35"></a>  pheno[[<span class="st">&quot;clumped&quot;</span>]] =<span class="st"> </span>clumped</span>
<span id="cb52-36"><a href="#cb52-36"></a>  <span class="co"># bind `clumped` into single DF and add to `data_list`</span></span>
<span id="cb52-37"><a href="#cb52-37"></a>  pheno[[<span class="st">&quot;clumped_all&quot;</span>]] =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(clumped)</span>
<span id="cb52-38"><a href="#cb52-38"></a>  </span>
<span id="cb52-39"><a href="#cb52-39"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb52-40"><a href="#cb52-40"></a>})</span></code></pre></div>
</div>
</div>
<div id="consolidate-key-data-into-single-df-for-analysis" class="section level2">
<h2><span class="header-section-number">1.10</span> Consolidate key data into single DF for analysis</h2>
<p><strong>NOTE</strong>: <code>clump_param</code> is set in <code>code/scripts/source.R</code></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>data_list =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb53-2"><a href="#cb53-2"></a>  <span class="co"># Filter `consol` by the index SNPs in target `clump`</span></span>
<span id="cb53-3"><a href="#cb53-3"></a>  target_clump =<span class="st"> </span>pheno[[<span class="st">&quot;clumped&quot;</span>]][[clump_param]]</span>
<span id="cb53-4"><a href="#cb53-4"></a>  </span>
<span id="cb53-5"><a href="#cb53-5"></a>  final =<span class="st"> </span>pheno[[<span class="st">&quot;consol&quot;</span>]] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-6"><a href="#cb53-6"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">SNP =</span> TOP_SNP) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-7"><a href="#cb53-7"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(SNP <span class="op">%in%</span><span class="st"> </span>target_clump<span class="op">$</span>SNP) </span>
<span id="cb53-8"><a href="#cb53-8"></a>  </span>
<span id="cb53-9"><a href="#cb53-9"></a>  <span class="co"># Add allele frequencies of SNP hits (global and per-population)</span></span>
<span id="cb53-10"><a href="#cb53-10"></a>  </span>
<span id="cb53-11"><a href="#cb53-11"></a>  <span class="co"># Add controls</span></span>
<span id="cb53-12"><a href="#cb53-12"></a>  controls =<span class="st"> </span>pheno[[<span class="st">&quot;random_af&quot;</span>]] <span class="op">%&gt;%</span></span>
<span id="cb53-13"><a href="#cb53-13"></a><span class="st">    </span><span class="co"># filter for SNPs in target_clump</span></span>
<span id="cb53-14"><a href="#cb53-14"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(TOP_SNP <span class="op">%in%</span><span class="st"> </span>target_clump<span class="op">$</span>SNP) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-15"><a href="#cb53-15"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="kw">c</span>(TOP_SNP, BIN_<span class="dv">100</span>, RANDOM_BIN_<span class="dv">100</span>), </span>
<span id="cb53-16"><a href="#cb53-16"></a>                  <span class="dt">SNP =</span> RANDOM_SNP) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-17"><a href="#cb53-17"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">RISK_AF =</span> ALT_FREQS)</span>
<span id="cb53-18"><a href="#cb53-18"></a>  </span>
<span id="cb53-19"><a href="#cb53-19"></a>  final =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(final, controls)  </span>
<span id="cb53-20"><a href="#cb53-20"></a>  pheno[[<span class="st">&quot;final&quot;</span>]] =<span class="st"> </span>final</span>
<span id="cb53-21"><a href="#cb53-21"></a>  </span>
<span id="cb53-22"><a href="#cb53-22"></a>  <span class="kw">return</span>(pheno)</span>
<span id="cb53-23"><a href="#cb53-23"></a>})</span>
<span id="cb53-24"><a href="#cb53-24"></a></span>
<span id="cb53-25"><a href="#cb53-25"></a><span class="co"># Create final df</span></span>
<span id="cb53-26"><a href="#cb53-26"></a>final_df =<span class="st"> </span><span class="kw">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb53-27"><a href="#cb53-27"></a>  out =<span class="st"> </span>pheno[[<span class="st">&quot;final&quot;</span>]]</span>
<span id="cb53-28"><a href="#cb53-28"></a>  </span>
<span id="cb53-29"><a href="#cb53-29"></a>  <span class="kw">return</span>(out)</span>
<span id="cb53-30"><a href="#cb53-30"></a>}) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-31"><a href="#cb53-31"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>()</span>
<span id="cb53-32"><a href="#cb53-32"></a></span>
<span id="cb53-33"><a href="#cb53-33"></a><span class="co"># Set factors</span></span>
<span id="cb53-34"><a href="#cb53-34"></a>final_df<span class="op">$</span>PHENO &lt;-<span class="st"> </span><span class="kw">factor</span>(final_df<span class="op">$</span>PHENO, <span class="dt">levels =</span> trait_levels_verb)</span>
<span id="cb53-35"><a href="#cb53-35"></a>final_df<span class="op">$</span>HIT_CONTROL =<span class="st"> </span><span class="kw">factor</span>(final_df<span class="op">$</span>HIT_CONTROL, <span class="dt">levels =</span> hit_control_levels)</span>
<span id="cb53-36"><a href="#cb53-36"></a></span>
<span id="cb53-37"><a href="#cb53-37"></a><span class="co"># Create DF for plotting</span></span>
<span id="cb53-38"><a href="#cb53-38"></a>final_plt =<span class="st"> </span>final_df <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-39"><a href="#cb53-39"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(SNP, PHENO, POPN, RISK_AF, HIT_CONTROL) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-40"><a href="#cb53-40"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> POPN, <span class="dt">values_from =</span> RISK_AF) </span></code></pre></div>
</div>
</div>
<div id="analysis" class="section level1">
<h1><span class="header-section-number">2</span> Analysis</h1>
<div id="manhattan-plots" class="section level2">
<h2><span class="header-section-number">2.1</span> Manhattan plots</h2>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a>counter =<span class="st"> </span><span class="dv">0</span></span>
<span id="cb54-2"><a href="#cb54-2"></a><span class="kw">lapply</span>(<span class="kw">unique</span>(final_df<span class="op">$</span>PHENO), <span class="cf">function</span>(pheno){</span>
<span id="cb54-3"><a href="#cb54-3"></a>  <span class="co"># set counter</span></span>
<span id="cb54-4"><a href="#cb54-4"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb54-5"><a href="#cb54-5"></a>  df =<span class="st"> </span>final_df <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb54-6"><a href="#cb54-6"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(PHENO <span class="op">==</span><span class="st"> </span>pheno <span class="op">&amp;</span><span class="st"> </span>HIT_CONTROL <span class="op">==</span><span class="st"> &quot;hit&quot;</span>)</span>
<span id="cb54-7"><a href="#cb54-7"></a>  <span class="co"># Get number of SNPs</span></span>
<span id="cb54-8"><a href="#cb54-8"></a>  snp_n =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(df<span class="op">$</span>SNP))</span>
<span id="cb54-9"><a href="#cb54-9"></a>  <span class="co"># Get title</span></span>
<span id="cb54-10"><a href="#cb54-10"></a>  title &lt;-<span class="st"> </span><span class="kw">paste</span>(pheno, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>, <span class="st">&quot;SNP count:&quot;</span>, snp_n)</span>
<span id="cb54-11"><a href="#cb54-11"></a>  <span class="co"># Plot</span></span>
<span id="cb54-12"><a href="#cb54-12"></a>  <span class="kw">get_man</span>(df, <span class="dt">trait =</span> pheno, <span class="dt">title =</span> title, <span class="dt">chr =</span> <span class="st">&quot;CHR&quot;</span>, <span class="dt">bp =</span> <span class="st">&quot;POS&quot;</span>, <span class="dt">snp =</span> <span class="st">&quot;SNP&quot;</span>, <span class="dt">p =</span> <span class="st">&quot;P&quot;</span>)</span>
<span id="cb54-13"><a href="#cb54-13"></a>})</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-33-1.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-33-2.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-33-3.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-33-4.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-33-5.png" /><!-- --><img src="index_files/figure-html/unnamed-chunk-33-6.png" /><!-- --></p>
</div>
<div id="allele-frequency-distributions-of-risk-alleles" class="section level2">
<h2><span class="header-section-number">2.2</span> Allele frequency distributions of risk alleles</h2>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a>final_df <span class="op">%&gt;%</span></span>
<span id="cb55-2"><a href="#cb55-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(HIT_CONTROL <span class="op">==</span><span class="st"> &quot;hit&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-3"><a href="#cb55-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(RISK_AF, <span class="dt">fill =</span> PHENO)) <span class="op">+</span></span>
<span id="cb55-4"><a href="#cb55-4"></a><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb55-5"><a href="#cb55-5"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_primary) <span class="op">+</span></span>
<span id="cb55-6"><a href="#cb55-6"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="kw">vars</span>(PHENO), <span class="dt">nrow =</span> <span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb55-7"><a href="#cb55-7"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F) <span class="op">+</span></span>
<span id="cb55-8"><a href="#cb55-8"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Risk allele frequency&quot;</span>) <span class="op">+</span></span>
<span id="cb55-9"><a href="#cb55-9"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>) <span class="op">+</span></span>
<span id="cb55-10"><a href="#cb55-10"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb55-11"><a href="#cb55-11"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Frequency distribution of </span><span class="ch">\&quot;</span><span class="st">risk</span><span class="ch">\&quot;</span><span class="st"> alleles (all 1KG populations combined)&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 28 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-34-1.png" /><!-- --></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_af_distribution&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb57-2"><a href="#cb57-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb57-3"><a href="#cb57-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb57-4"><a href="#cb57-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb57-5"><a href="#cb57-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb57-6"><a href="#cb57-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="allele-frequency-vs-effect-size" class="section level2">
<h2><span class="header-section-number">2.3</span> Allele frequency vs effect size</h2>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a>one =<span class="st"> </span>final_df <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb58-2"><a href="#cb58-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(HIT_CONTROL <span class="op">==</span><span class="st"> &quot;hit&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POPN <span class="op">==</span><span class="st"> &quot;all&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb58-3"><a href="#cb58-3"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb58-4"><a href="#cb58-4"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(RISK_AF, OR_OR_BETA, <span class="dt">colour =</span> PHENO),</span>
<span id="cb58-5"><a href="#cb58-5"></a>               <span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span></span>
<span id="cb58-6"><a href="#cb58-6"></a><span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> pal_primary) <span class="op">+</span></span>
<span id="cb58-7"><a href="#cb58-7"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="kw">vars</span>(PHENO), <span class="dt">nrow =</span> <span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb58-8"><a href="#cb58-8"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> F, <span class="dt">alpha =</span> F) <span class="op">+</span></span>
<span id="cb58-9"><a href="#cb58-9"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb58-10"><a href="#cb58-10"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Risk allele frequency vs effect size (OR or beta)&quot;</span>)</span>
<span id="cb58-11"><a href="#cb58-11"></a></span>
<span id="cb58-12"><a href="#cb58-12"></a><span class="co"># Zoom in</span></span>
<span id="cb58-13"><a href="#cb58-13"></a>two =<span class="st"> </span>final_df <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb58-14"><a href="#cb58-14"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(HIT_CONTROL <span class="op">==</span><span class="st"> &quot;hit&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POPN <span class="op">==</span><span class="st"> &quot;all&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb58-15"><a href="#cb58-15"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb58-16"><a href="#cb58-16"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(RISK_AF, OR_OR_BETA, <span class="dt">colour =</span> PHENO),</span>
<span id="cb58-17"><a href="#cb58-17"></a>               <span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span></span>
<span id="cb58-18"><a href="#cb58-18"></a><span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> pal_primary) <span class="op">+</span></span>
<span id="cb58-19"><a href="#cb58-19"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="kw">vars</span>(PHENO), <span class="dt">nrow =</span> <span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb58-20"><a href="#cb58-20"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> F, <span class="dt">alpha =</span> F) <span class="op">+</span></span>
<span id="cb58-21"><a href="#cb58-21"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb58-22"><a href="#cb58-22"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Risk allele frequency vs effect size (OR or beta)&quot;</span>) <span class="op">+</span></span>
<span id="cb58-23"><a href="#cb58-23"></a><span class="st">    </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="dv">20</span>)</span>
<span id="cb58-24"><a href="#cb58-24"></a></span>
<span id="cb58-25"><a href="#cb58-25"></a>one</span></code></pre></div>
<pre><code>## Warning: Removed 3463 rows containing missing values (geom_point).</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1"></a>two</span></code></pre></div>
<pre><code>## Warning: Removed 3467 rows containing missing values (geom_point).</code></pre>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-36-1.png" alt="Full size and zoomed to 0 &lt; y &lt; 20" width="50%" /><img src="index_files/figure-html/unnamed-chunk-36-2.png" alt="Full size and zoomed to 0 &lt; y &lt; 20" width="50%" />
=======
<div class="sourceCode" id="cb51"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Activate environment</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate fst_env_rhel</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set variables</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="va">in_vcf_dir=</span>data/20210125_snp_hits_filtered</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="va">snp_p_dir=</span>data/20210122_snp_hit_lists</span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir=</span>data/20210125_clumped</span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="va">r2_params=$(</span><span class="bu">echo</span> 0.1 0.2 0.3<span class="va">)</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="va">kb_params=$(</span><span class="bu">echo</span> 500 750 1000 <span class="va">)</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Make directory</span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir</span></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Run with different parameters</span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mkdir</span> -p <span class="va">$out_dir</span>/<span class="va">$trait</span> <span class="kw">;</span></span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a>  <span class="kw">for</span> <span class="ex">r2</span> in <span class="va">$r2_params</span> <span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a>    <span class="kw">for</span> <span class="ex">kb</span> in <span class="va">$kb_params</span> <span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a>      <span class="ex">bsub</span> <span class="kw">\</span></span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a>        <span class="ex">-o</span> ../log/20210125_clump_<span class="va">$trait</span>\_<span class="va">$r2</span>\_<span class="va">$kb</span>.out <span class="kw">\</span></span>
<span id="cb51-26"><a href="#cb51-26" aria-hidden="true" tabindex="-1"></a>        <span class="ex">-e</span> ../log/20210125_clump_<span class="va">$trait</span>\_<span class="va">$r2</span>\_<span class="va">$kb</span>.err <span class="kw">\</span></span>
<span id="cb51-27"><a href="#cb51-27" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb51-28"><a href="#cb51-28" aria-hidden="true" tabindex="-1"></a><span class="st">        conda activate fst_env_rhel ;</span></span>
<span id="cb51-29"><a href="#cb51-29" aria-hidden="true" tabindex="-1"></a><span class="st">        plink \</span></span>
<span id="cb51-30"><a href="#cb51-30" aria-hidden="true" tabindex="-1"></a><span class="st">          --vcf </span><span class="va">$in_vcf_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz \</span></span>
<span id="cb51-31"><a href="#cb51-31" aria-hidden="true" tabindex="-1"></a><span class="st">          --clump </span><span class="va">$snp_p_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">\_with_P.txt \</span></span>
<span id="cb51-32"><a href="#cb51-32" aria-hidden="true" tabindex="-1"></a><span class="st">          --clump-p1 0.00000001 \</span></span>
<span id="cb51-33"><a href="#cb51-33" aria-hidden="true" tabindex="-1"></a><span class="st">          --clump-p2 0.00000001 \</span></span>
<span id="cb51-34"><a href="#cb51-34" aria-hidden="true" tabindex="-1"></a><span class="st">          --clump-r2 </span><span class="va">$r2</span><span class="st"> \</span></span>
<span id="cb51-35"><a href="#cb51-35" aria-hidden="true" tabindex="-1"></a><span class="st">          --clump-kb </span><span class="va">$kb</span><span class="st"> \</span></span>
<span id="cb51-36"><a href="#cb51-36" aria-hidden="true" tabindex="-1"></a><span class="st">          --out </span><span class="va">$out_dir</span><span class="st">/</span><span class="va">$trait</span><span class="st">/r2-</span><span class="va">$r2</span><span class="st">\_kb-</span><span class="va">$kb</span><span class="st"> </span></span>
<span id="cb51-37"><a href="#cb51-37" aria-hidden="true" tabindex="-1"></a><span class="st">        &quot;&quot;&quot;</span>  ;</span>
<span id="cb51-38"><a href="#cb51-38" aria-hidden="true" tabindex="-1"></a>    <span class="kw">done</span> ;</span>
<span id="cb51-39"><a href="#cb51-39" aria-hidden="true" tabindex="-1"></a>  <span class="kw">done</span>;  </span>
<span id="cb51-40"><a href="#cb51-40" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="filter-vcfs-for-clumped-snps" class="section level2" number="1.10">
<h2 number="1.10"><span class="header-section-number">1.10</span> Filter VCFs for clumped SNPs</h2>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># On cluster</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Set variables</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="va">clump_param=</span><span class="st">&quot;r2-0.1_kb-1000&quot;</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="va">in_dir_snp=</span>data/20210125_clumped</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="va">in_dir_vcf=</span>data/20210125_snp_hits_filtered</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="va">ref=</span>../refs/hs37d5.fa.gz</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir_snp_list=</span>data/20210128_clumped_snps</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir_vcfs=</span>data/20210128_hits_vcfs</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir_snp_list</span> <span class="va">$out_dir_vcfs</span></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract SNP column and write to list</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>  <span class="va">target_file=$in_dir_snp</span>/<span class="va">$trait</span>/<span class="va">$clump_param</span>.clumped ;</span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">awk</span> <span class="st">&#39;{print $3}&#39;</span> <span class="va">$target_file</span> <span class="kw">|</span> <span class="fu">tail</span> -n+2 <span class="kw">\</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">&gt;</span> <span class="va">$out_dir_snp_list</span><span class="ex">/</span><span class="va">$trait</span><span class="ex">.list</span> <span class="kw">;</span></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Make new VCFs</span></span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bsub</span> <span class="kw">\</span></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-o</span> ../log/20210128_filter_vcfs_clump_<span class="va">$trait</span>.out <span class="kw">\</span></span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-e</span> ../log/20210128_filter_vcfs_clump_<span class="va">$trait</span>.err <span class="kw">\</span></span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a><span class="st">    gatk SelectVariants \</span></span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a><span class="st">      -R </span><span class="va">$ref</span><span class="st"> \</span></span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a><span class="st">      -V </span><span class="va">$in_dir_vcf</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz \</span></span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a><span class="st">      --keep-ids </span><span class="va">$out_dir_snp_list</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a><span class="st">      -O </span><span class="va">$out_dir_vcfs</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz     </span></span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a><span class="st">    &quot;&quot;&quot;</span> ;</span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span>    </span></code></pre></div>
<div id="add-clumped-snp-files-to-data_list" class="section level3" number="1.10.1">
<h3 number="1.10.1"><span class="header-section-number">1.10.1</span> Add clumped SNP files to <code>data_list</code></h3>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>target_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210125_clumped&quot;</span>)</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get trait name</span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get file path</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>  target_files <span class="ot">=</span> <span class="fu">list.files</span>(<span class="fu">file.path</span>(target_dir, trait),</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>                            <span class="at">pattern =</span> <span class="st">&quot;.clumped&quot;</span>,</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">full.names =</span> T)</span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(target_files) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.clumped&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">basename</span>(target_files))</span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read files as `clumped`</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>  counter_clump <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>  clumped <span class="ot">=</span> <span class="fu">lapply</span>(target_files, <span class="cf">function</span>(params){</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set counter</span></span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>    counter_clump <span class="ot">&lt;&lt;-</span> counter_clump <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># split params string</span></span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>    param_str <span class="ot">=</span> <span class="fu">names</span>(target_files)[counter_clump] <span class="sc">%&gt;%</span> </span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a>      stringr<span class="sc">::</span><span class="fu">str_split</span>(<span class="st">&quot;_&quot;</span>, <span class="at">simplify =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a>      stringr<span class="sc">::</span><span class="fu">str_split</span>(<span class="st">&quot;-&quot;</span>, <span class="at">simplify =</span> T)</span>
<span id="cb53-23"><a href="#cb53-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get params</span></span>
<span id="cb53-24"><a href="#cb53-24" aria-hidden="true" tabindex="-1"></a>    r2 <span class="ot">=</span> <span class="fu">as.numeric</span>(param_str[<span class="dv">1</span>,<span class="dv">2</span>])</span>
<span id="cb53-25"><a href="#cb53-25" aria-hidden="true" tabindex="-1"></a>    kb <span class="ot">=</span> <span class="fu">as.integer</span>(param_str[<span class="dv">2</span>,<span class="dv">2</span>])</span>
<span id="cb53-26"><a href="#cb53-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># read files</span></span>
<span id="cb53-27"><a href="#cb53-27" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">=</span> <span class="fu">read.table</span>(params, <span class="at">header =</span> T)</span>
<span id="cb53-28"><a href="#cb53-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># add params to DF</span></span>
<span id="cb53-29"><a href="#cb53-29" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>r2 <span class="ot">=</span> r2</span>
<span id="cb53-30"><a href="#cb53-30" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>kb <span class="ot">=</span> kb</span>
<span id="cb53-31"><a href="#cb53-31" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb53-32"><a href="#cb53-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb53-33"><a href="#cb53-33" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb53-34"><a href="#cb53-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add `clumped` list to `data_list`</span></span>
<span id="cb53-35"><a href="#cb53-35" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;clumped&quot;</span>]] <span class="ot">=</span> clumped</span>
<span id="cb53-36"><a href="#cb53-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># bind `clumped` into single DF and add to `data_list`</span></span>
<span id="cb53-37"><a href="#cb53-37" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;clumped_all&quot;</span>]] <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(clumped)</span>
<span id="cb53-38"><a href="#cb53-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb53-39"><a href="#cb53-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb53-40"><a href="#cb53-40" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
</div>
</div>
<div id="extract-random-snps-filtered-by-those-matching-the-clumped-snps" class="section level2" number="1.11">
<h2 number="1.11"><span class="header-section-number">1.11</span> Extract random SNPs filtered by those matching the clumped SNPs</h2>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_random_snps&quot;</span>)</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(out_dir)</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>out <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter</span></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get name of trait</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">=</span> <span class="fu">names</span>(data_list)[counter]</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get clumped SNPs</span></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>  clumped_snps <span class="ot">=</span> pheno[[<span class="st">&quot;clumped&quot;</span>]][[clump_param]]<span class="sc">$</span>SNP</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># gather SNPs</span></span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>  snps <span class="ot">=</span> pheno[[<span class="st">&quot;random_af&quot;</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(TOP_SNP <span class="sc">%in%</span> clumped_snps) <span class="sc">%&gt;%</span> </span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(RANDOM_SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>(.)</span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># write to file</span></span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>  out_path <span class="ot">=</span> <span class="fu">file.path</span>(out_dir, <span class="fu">paste</span>(trait, <span class="st">&quot;.list&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_lines</span>(snps<span class="sc">$</span>RANDOM_SNP, out_path)</span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(out)</span></code></pre></div>
</div>
<div id="make-vcfs-for-random-control-snps" class="section level2" number="1.12">
<h2 number="1.12"><span class="header-section-number">1.12</span> Make VCFs for random control SNPs</h2>
<div class="sourceCode" id="cb55"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># On cluster</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Set variables</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="va">traits=$(</span><span class="bu">echo</span> hei bmi edu int ibd pig<span class="va">)</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="va">in_dir_snp_list=</span>data/20210201_random_snps</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="va">in_vcf=</span>../vcfs/1kg_all.vcf.gz</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="va">ref=</span>../refs/hs37d5.fa.gz</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="va">out_dir_vcfs=</span>data/20210201_rndm_vcfs</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p <span class="va">$out_dir_vcfs</span></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Make new VCFs</span></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a><span class="kw">for</span> <span class="ex">trait</span> in <span class="va">$(</span><span class="bu">echo</span> <span class="va">$traits</span> <span class="va">)</span><span class="kw">;</span> <span class="kw">do</span></span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>  <span class="ex">bsub</span> <span class="kw">\</span></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-M</span> 20000 <span class="kw">\</span></span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-o</span> ../log/20210201_filter_vcfs_rndm_<span class="va">$trait</span>.out <span class="kw">\</span></span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>    <span class="ex">-e</span> ../log/20210201_filter_vcfs_rndm_<span class="va">$trait</span>.err <span class="kw">\</span></span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;</span></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a><span class="st">    conda activate fst_env_rhel ;</span></span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a><span class="st">    gatk SelectVariants \</span></span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a><span class="st">      -R </span><span class="va">$ref</span><span class="st"> \</span></span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a><span class="st">      -V </span><span class="va">$in_vcf</span><span class="st"> \</span></span>
<span id="cb55-23"><a href="#cb55-23" aria-hidden="true" tabindex="-1"></a><span class="st">      --keep-ids </span><span class="va">$in_dir_snp_list</span><span class="st">/</span><span class="va">$trait</span><span class="st">.list \</span></span>
<span id="cb55-24"><a href="#cb55-24" aria-hidden="true" tabindex="-1"></a><span class="st">      -O </span><span class="va">$out_dir_vcfs</span><span class="st">/</span><span class="va">$trait</span><span class="st">.vcf.gz     </span></span>
<span id="cb55-25"><a href="#cb55-25" aria-hidden="true" tabindex="-1"></a><span class="st">    &quot;&quot;&quot;</span> ;</span>
<span id="cb55-26"><a href="#cb55-26" aria-hidden="true" tabindex="-1"></a><span class="kw">done</span>    </span></code></pre></div>
</div>
<div id="consolidate-key-data-into-single-df-for-analysis" class="section level2" number="1.13">
<h2 number="1.13"><span class="header-section-number">1.13</span> Consolidate key data into single DF for analysis</h2>
<p><strong>NOTE</strong>: <code>clump_param</code> is set in <code>code/scripts/source.R</code></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter `consol` by the index SNPs in target `clump`</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  target_clump <span class="ot">=</span> pheno[[<span class="st">&quot;clumped&quot;</span>]][[clump_param]]</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  final <span class="ot">=</span> pheno[[<span class="st">&quot;consol&quot;</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SNP =</span> TOP_SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SNP <span class="sc">%in%</span> target_clump<span class="sc">$</span>SNP) </span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add allele frequencies of SNP hits (global and per-population)</span></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add controls</span></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>  controls <span class="ot">=</span> pheno[[<span class="st">&quot;random_af&quot;</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter for SNPs in target_clump</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(TOP_SNP <span class="sc">%in%</span> target_clump<span class="sc">$</span>SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(TOP_SNP, BIN_10, RANDOM_BIN_10), </span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">SNP =</span> RANDOM_SNP) <span class="sc">%&gt;%</span> </span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">RISK_AF =</span> ALT_FREQS)</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>  final <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(final, controls)  </span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>  pheno[[<span class="st">&quot;final&quot;</span>]] <span class="ot">=</span> final</span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pheno)</span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Create final df</span></span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a>final_df <span class="ot">=</span> <span class="fu">lapply</span>(data_list, <span class="cf">function</span>(pheno){</span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> pheno[[<span class="st">&quot;final&quot;</span>]]</span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb56-30"><a href="#cb56-30" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> </span>
<span id="cb56-31"><a href="#cb56-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb56-32"><a href="#cb56-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-33"><a href="#cb56-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Set factors</span></span>
<span id="cb56-34"><a href="#cb56-34" aria-hidden="true" tabindex="-1"></a>final_df<span class="sc">$</span>PHENO <span class="ot">&lt;-</span> <span class="fu">factor</span>(final_df<span class="sc">$</span>PHENO, <span class="at">levels =</span> trait_levels_verb)</span>
<span id="cb56-35"><a href="#cb56-35" aria-hidden="true" tabindex="-1"></a>final_df<span class="sc">$</span>HIT_CONTROL <span class="ot">=</span> <span class="fu">factor</span>(final_df<span class="sc">$</span>HIT_CONTROL, <span class="at">levels =</span> hit_control_levels)</span>
<span id="cb56-36"><a href="#cb56-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-37"><a href="#cb56-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Create DF for plotting</span></span>
<span id="cb56-38"><a href="#cb56-38" aria-hidden="true" tabindex="-1"></a>final_plt <span class="ot">=</span> final_df <span class="sc">%&gt;%</span> </span>
<span id="cb56-39"><a href="#cb56-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(SNP, PHENO, POPN, RISK_AF, HIT_CONTROL) <span class="sc">%&gt;%</span> </span>
<span id="cb56-40"><a href="#cb56-40" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> POPN, <span class="at">values_from =</span> RISK_AF) </span></code></pre></div>
</div>
</div>
<div id="analysis" class="section level1" number="2">
<h1 number="2"><span class="header-section-number">2</span> Analysis</h1>
<div id="manhattan-plots" class="section level2" number="2.1">
<h2 number="2.1"><span class="header-section-number">2.1</span> Manhattan plots</h2>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="fu">unique</span>(final_df<span class="sc">$</span>PHENO), <span class="cf">function</span>(pheno){</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> final_df <span class="sc">%&gt;%</span> </span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(PHENO <span class="sc">==</span> pheno <span class="sc">&amp;</span> HIT_CONTROL <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>)</span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get number of SNPs</span></span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>  snp_n <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(df<span class="sc">$</span>SNP))</span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get title</span></span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a>  title <span class="ot">&lt;-</span> <span class="fu">paste</span>(pheno, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="st">&quot;SNP count:&quot;</span>, snp_n)</span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot</span></span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">get_man</span>(df, <span class="at">trait =</span> pheno, <span class="at">title =</span> title, <span class="at">chr =</span> <span class="st">&quot;CHR&quot;</span>, <span class="at">bp =</span> <span class="st">&quot;POS&quot;</span>, <span class="at">snp =</span> <span class="st">&quot;SNP&quot;</span>, <span class="at">p =</span> <span class="st">&quot;P&quot;</span>)</span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-36-1.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-36-2.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-36-3.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-36-4.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-36-5.svg" /><!-- --><img src="index_files/figure-html/unnamed-chunk-36-6.svg" /><!-- --></p>
</div>
<div id="allele-frequency-distributions-of-risk-alleles" class="section level2" number="2.2">
<h2 number="2.2"><span class="header-section-number">2.2</span> Allele frequency distributions of risk alleles</h2>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>final_df <span class="sc">%&gt;%</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(HIT_CONTROL <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(RISK_AF, <span class="at">fill =</span> PHENO)) <span class="sc">+</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(PHENO), <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Risk allele frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Count&quot;</span>) <span class="sc">+</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Frequency distribution of </span><span class="sc">\&quot;</span><span class="st">risk</span><span class="sc">\&quot;</span><span class="st"> alleles (all 1KG populations combined)&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 28 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-37-1.svg" /><!-- --></p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_af_distribution&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="allele-frequency-vs-effect-size" class="section level2" number="2.3">
<h2 number="2.3"><span class="header-section-number">2.3</span> Allele frequency vs effect size</h2>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>one <span class="ot">=</span> final_df <span class="sc">%&gt;%</span> </span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(HIT_CONTROL <span class="sc">==</span> <span class="st">&quot;hit&quot;</span> <span class="sc">&amp;</span> POPN <span class="sc">==</span> <span class="st">&quot;all&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(RISK_AF, OR_OR_BETA, <span class="at">colour =</span> PHENO),</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(PHENO), <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">colour =</span> F, <span class="at">alpha =</span> F) <span class="sc">+</span></span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Risk allele frequency vs effect size (OR or beta)&quot;</span>)</span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Zoom in</span></span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a>two <span class="ot">=</span> final_df <span class="sc">%&gt;%</span> </span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(HIT_CONTROL <span class="sc">==</span> <span class="st">&quot;hit&quot;</span> <span class="sc">&amp;</span> POPN <span class="sc">==</span> <span class="st">&quot;all&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb61-15"><a href="#cb61-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb61-16"><a href="#cb61-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(RISK_AF, OR_OR_BETA, <span class="at">colour =</span> PHENO),</span>
<span id="cb61-17"><a href="#cb61-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb61-18"><a href="#cb61-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb61-19"><a href="#cb61-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(PHENO), <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb61-20"><a href="#cb61-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">colour =</span> F, <span class="at">alpha =</span> F) <span class="sc">+</span></span>
<span id="cb61-21"><a href="#cb61-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb61-22"><a href="#cb61-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Risk allele frequency vs effect size (OR or beta)&quot;</span>) <span class="sc">+</span></span>
<span id="cb61-23"><a href="#cb61-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">20</span>)</span>
<span id="cb61-24"><a href="#cb61-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-25"><a href="#cb61-25" aria-hidden="true" tabindex="-1"></a>one</span></code></pre></div>
<pre><code>## Warning: Removed 3463 rows containing missing values (geom_point).</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>two</span></code></pre></div>
<pre><code>## Warning: Removed 3467 rows containing missing values (geom_point).</code></pre>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-39-1.svg" alt="Full size and zoomed to 0 &lt; y &lt; 20" width="50%" /><img src="index_files/figure-html/unnamed-chunk-39-2.svg" alt="Full size and zoomed to 0 &lt; y &lt; 20" width="50%" />
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<p class="caption">
Full size and zoomed to 0 &lt; y &lt; 20
</p>
</div>
<<<<<<< HEAD
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_af_v_effect_size&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb62-2"><a href="#cb62-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb62-3"><a href="#cb62-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb62-4"><a href="#cb62-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb62-5"><a href="#cb62-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb62-6"><a href="#cb62-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_af_v_effect_size&quot;</span>, <span class="st">&quot;20210127_hits_zoomed.png&quot;</span>),</span>
<span id="cb63-2"><a href="#cb63-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb63-3"><a href="#cb63-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb63-4"><a href="#cb63-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb63-5"><a href="#cb63-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb63-6"><a href="#cb63-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="fst" class="section level2">
<h2><span class="header-section-number">2.4</span> Fst</h2>
<div id="with-all-populations" class="section level3">
<h3><span class="header-section-number">2.4.1</span> With all populations</h3>
<div id="get-fst-stats-for-top-snps" class="section level4">
<h4><span class="header-section-number">2.4.1.1</span> Get Fst stats for top SNPs</h4>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1"></a><span class="co"># Create raw list of variants</span></span>
<span id="cb64-2"><a href="#cb64-2"></a>case =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210125_snp_hits_filtered&quot;</span>)</span>
<span id="cb64-3"><a href="#cb64-3"></a>control =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210127_snp_rndm_filtered&quot;</span>)</span>
<span id="cb64-4"><a href="#cb64-4"></a>target_dirs =<span class="st"> </span><span class="kw">c</span>(case, control)</span>
<span id="cb64-5"><a href="#cb64-5"></a><span class="kw">names</span>(target_dirs) =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;case&quot;</span>, <span class="st">&quot;control&quot;</span>)</span>
<span id="cb64-6"><a href="#cb64-6"></a></span>
<span id="cb64-7"><a href="#cb64-7"></a><span class="co"># Set seed</span></span>
<span id="cb64-8"><a href="#cb64-8"></a>initial_seed =<span class="st"> </span><span class="dv">53</span></span>
<span id="cb64-9"><a href="#cb64-9"></a><span class="kw">set.seed</span>(initial_seed)</span>
<span id="cb64-10"><a href="#cb64-10"></a>seeds =<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>, <span class="dt">size =</span> <span class="kw">length</span>(target_dirs))</span>
<span id="cb64-11"><a href="#cb64-11"></a></span>
<span id="cb64-12"><a href="#cb64-12"></a><span class="co"># Run </span></span>
<span id="cb64-13"><a href="#cb64-13"></a>counter =<span class="st"> </span><span class="dv">0</span></span>
<span id="cb64-14"><a href="#cb64-14"></a>fst_out =<span class="st"> </span><span class="kw">lapply</span>(target_dirs, <span class="cf">function</span>(target_dir){</span>
<span id="cb64-15"><a href="#cb64-15"></a>  <span class="co"># set counter </span></span>
<span id="cb64-16"><a href="#cb64-16"></a>  counter &lt;&lt;-<span class="st"> </span>counter <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb64-17"><a href="#cb64-17"></a>  </span>
<span id="cb64-18"><a href="#cb64-18"></a>  vcf_list_raw &lt;-<span class="st"> </span><span class="kw">lapply</span>(trait_levels, <span class="cf">function</span>(trait){</span>
<span id="cb64-19"><a href="#cb64-19"></a>    target_file =<span class="st"> </span><span class="kw">file.path</span>(target_dir, <span class="kw">paste</span>(trait, <span class="st">&quot;.vcf.gz&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb64-20"><a href="#cb64-20"></a>    <span class="co"># read in file</span></span>
<span id="cb64-21"><a href="#cb64-21"></a>    vcf_out &lt;-<span class="st"> </span>pegas<span class="op">::</span><span class="kw">read.vcf</span>(target_file)</span>
<span id="cb64-22"><a href="#cb64-22"></a>    <span class="co"># filter `cases` for clumped SNPs, and take same number of `control` SNPs</span></span>
<span id="cb64-23"><a href="#cb64-23"></a>    <span class="co">## get target SNP IDs</span></span>
<span id="cb64-24"><a href="#cb64-24"></a>    target_snps =<span class="st"> </span>data_list[[trait]]<span class="op">$</span>clumped[[clump_param]]<span class="op">$</span>SNP</span>
<span id="cb64-25"><a href="#cb64-25"></a>    <span class="cf">if</span> (<span class="kw">names</span>(target_dirs)[counter] <span class="op">==</span><span class="st"> &quot;case&quot;</span>){</span>
<span id="cb64-26"><a href="#cb64-26"></a>      <span class="co">## filter VCF</span></span>
<span id="cb64-27"><a href="#cb64-27"></a>      vcf_out =<span class="st"> </span>vcf_out[, <span class="kw">colnames</span>(vcf_out) <span class="op">%in%</span><span class="st"> </span>target_snps]</span>
<span id="cb64-28"><a href="#cb64-28"></a>      <span class="co">## take sample of same number of `control` SNPs</span></span>
<span id="cb64-29"><a href="#cb64-29"></a>    } <span class="cf">else</span> <span class="cf">if</span> (<span class="kw">names</span>(target_dirs)[counter] <span class="op">==</span><span class="st"> &quot;control&quot;</span>){</span>
<span id="cb64-30"><a href="#cb64-30"></a>      <span class="kw">set.seed</span>(seeds[counter])</span>
<span id="cb64-31"><a href="#cb64-31"></a>      vcf_out =<span class="st"> </span>vcf_out[, <span class="kw">sample</span>(<span class="kw">ncol</span>(vcf_out), <span class="dt">size =</span> <span class="kw">length</span>(target_snps))]</span>
<span id="cb64-32"><a href="#cb64-32"></a>    }</span>
<span id="cb64-33"><a href="#cb64-33"></a>    </span>
<span id="cb64-34"><a href="#cb64-34"></a>    <span class="kw">return</span>(vcf_out)</span>
<span id="cb64-35"><a href="#cb64-35"></a>  })</span>
<span id="cb64-36"><a href="#cb64-36"></a>  </span>
<span id="cb64-37"><a href="#cb64-37"></a>  <span class="co"># Create vector of populations</span></span>
<span id="cb64-38"><a href="#cb64-38"></a>  populations &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(<span class="kw">rownames</span>(vcf_list_raw[[<span class="dv">1</span>]]), <span class="cf">function</span>(sample){</span>
<span id="cb64-39"><a href="#cb64-39"></a>    meta<span class="op">$</span>Population[meta<span class="op">$</span>Sample <span class="op">==</span><span class="st"> </span>sample]</span>
<span id="cb64-40"><a href="#cb64-40"></a>  }))</span>
<span id="cb64-41"><a href="#cb64-41"></a>  </span>
<span id="cb64-42"><a href="#cb64-42"></a>  <span class="co"># Generate Fst stats</span></span>
<span id="cb64-43"><a href="#cb64-43"></a>  fst_out_df &lt;-<span class="st"> </span><span class="kw">lapply</span>(vcf_list_raw, <span class="cf">function</span>(pheno){</span>
<span id="cb64-44"><a href="#cb64-44"></a>    out =<span class="st"> </span><span class="kw">as.data.frame</span>(pegas<span class="op">::</span><span class="kw">Fst</span>(pheno, <span class="dt">pop =</span> populations))</span>
<span id="cb64-45"><a href="#cb64-45"></a>    <span class="co"># put rownames into separate column</span></span>
<span id="cb64-46"><a href="#cb64-46"></a>    out<span class="op">$</span>snp &lt;-<span class="st"> </span><span class="kw">rownames</span>(out)</span>
<span id="cb64-47"><a href="#cb64-47"></a>    </span>
<span id="cb64-48"><a href="#cb64-48"></a>    <span class="kw">return</span>(out)</span>
<span id="cb64-49"><a href="#cb64-49"></a>  }) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb64-50"><a href="#cb64-50"></a><span class="st">    </span><span class="co"># bind into single DF</span></span>
<span id="cb64-51"><a href="#cb64-51"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;phenotype&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb64-52"><a href="#cb64-52"></a><span class="st">    </span><span class="co"># remove NA</span></span>
<span id="cb64-53"><a href="#cb64-53"></a><span class="st">    </span>tidyr<span class="op">::</span><span class="kw">drop_na</span>()</span>
<span id="cb64-54"><a href="#cb64-54"></a>  </span>
<span id="cb64-55"><a href="#cb64-55"></a>  <span class="kw">return</span>(fst_out_df)</span>
<span id="cb64-56"><a href="#cb64-56"></a>}) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb64-57"><a href="#cb64-57"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;case_control&quot;</span>)</span>
<span id="cb64-58"><a href="#cb64-58"></a></span>
<span id="cb64-59"><a href="#cb64-59"></a><span class="co"># Recode phenotype</span></span>
<span id="cb64-60"><a href="#cb64-60"></a>fst_out<span class="op">$</span>phenotype &lt;-<span class="st"> </span><span class="kw">factor</span>(fst_out<span class="op">$</span>phenotype, <span class="dt">levels =</span> trait_levels)</span>
<span id="cb64-61"><a href="#cb64-61"></a>fst_out<span class="op">$</span>phenotype =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">recode</span>(fst_out<span class="op">$</span>phenotype, <span class="op">!!!</span>recode_vec)</span></code></pre></div>
<p>Write to file</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1"></a>out_dir =<span class="st"> </span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210127_results&quot;</span>)</span>
<span id="cb65-2"><a href="#cb65-2"></a>out_path =<span class="st"> </span><span class="kw">file.path</span>(out_dir, <span class="kw">paste</span>(<span class="st">&quot;20210127_fst&quot;</span>, <span class="st">&quot;.csv&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb65-3"><a href="#cb65-3"></a></span>
<span id="cb65-4"><a href="#cb65-4"></a><span class="kw">dir.create</span>(out_dir)</span>
<span id="cb65-5"><a href="#cb65-5"></a></span>
<span id="cb65-6"><a href="#cb65-6"></a>readr<span class="op">::</span><span class="kw">write_csv</span>(fst_out, out_path)</span></code></pre></div>
<p>Read back in</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1"></a>fst_out =<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_csv</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210127_results/20210127_fst.csv&quot;</span>))</span>
<span id="cb66-2"><a href="#cb66-2"></a>fst_out<span class="op">$</span>phenotype &lt;-<span class="st"> </span><span class="kw">factor</span>(fst_out<span class="op">$</span>phenotype, <span class="dt">levels =</span> trait_levels_verb)</span>
<span id="cb66-3"><a href="#cb66-3"></a></span>
<span id="cb66-4"><a href="#cb66-4"></a>knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">head</span>(fst_out))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">case_control</th>
=======
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_af_v_effect_size&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_af_v_effect_size&quot;</span>, <span class="st">&quot;20210127_hits_zoomed.png&quot;</span>),</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="fst" class="section level2" number="2.4">
<h2 number="2.4"><span class="header-section-number">2.4</span> Fst</h2>
<div id="with-all-populations" class="section level3" number="2.4.1">
<h3 number="2.4.1"><span class="header-section-number">2.4.1</span> With all populations</h3>
<div id="get-fst-stats-for-top-snps" class="section level4" number="2.4.1.1">
<h4 number="2.4.1.1"><span class="header-section-number">2.4.1.1</span> Get Fst stats for top SNPs</h4>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create raw list of variants</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>hit <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210128_hits_vcfs&quot;</span>)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>control <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210201_rndm_vcfs&quot;</span>)</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>target_dirs <span class="ot">=</span> <span class="fu">c</span>(hit, control)</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(target_dirs) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;hit&quot;</span>, <span class="st">&quot;control&quot;</span>)</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Run </span></span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>fst_out <span class="ot">=</span> <span class="fu">lapply</span>(target_dirs, <span class="cf">function</span>(target_dir){</span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set counter </span></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>  counter <span class="ot">&lt;&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a>  vcf_list_raw <span class="ot">&lt;-</span> <span class="fu">lapply</span>(trait_levels, <span class="cf">function</span>(trait){</span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a>    target_file <span class="ot">=</span> <span class="fu">file.path</span>(target_dir, <span class="fu">paste</span>(trait, <span class="st">&quot;.vcf.gz&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb67-15"><a href="#cb67-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># read in file</span></span>
<span id="cb67-16"><a href="#cb67-16" aria-hidden="true" tabindex="-1"></a>    vcf_out <span class="ot">&lt;-</span> pegas<span class="sc">::</span><span class="fu">read.vcf</span>(target_file)</span>
<span id="cb67-17"><a href="#cb67-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-18"><a href="#cb67-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(vcf_out)</span>
<span id="cb67-19"><a href="#cb67-19" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb67-20"><a href="#cb67-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb67-21"><a href="#cb67-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create vector of populations</span></span>
<span id="cb67-22"><a href="#cb67-22" aria-hidden="true" tabindex="-1"></a>  populations <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">rownames</span>(vcf_list_raw[[<span class="dv">1</span>]]), <span class="cf">function</span>(sample){</span>
<span id="cb67-23"><a href="#cb67-23" aria-hidden="true" tabindex="-1"></a>    meta<span class="sc">$</span>Population[meta<span class="sc">$</span>Sample <span class="sc">==</span> sample]</span>
<span id="cb67-24"><a href="#cb67-24" aria-hidden="true" tabindex="-1"></a>  }))</span>
<span id="cb67-25"><a href="#cb67-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb67-26"><a href="#cb67-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate Fst stats</span></span>
<span id="cb67-27"><a href="#cb67-27" aria-hidden="true" tabindex="-1"></a>  fst_out_df <span class="ot">&lt;-</span> <span class="fu">lapply</span>(vcf_list_raw, <span class="cf">function</span>(pheno){</span>
<span id="cb67-28"><a href="#cb67-28" aria-hidden="true" tabindex="-1"></a>    out <span class="ot">=</span> <span class="fu">as.data.frame</span>(pegas<span class="sc">::</span><span class="fu">Fst</span>(pheno, <span class="at">pop =</span> populations))</span>
<span id="cb67-29"><a href="#cb67-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># put rownames into separate column</span></span>
<span id="cb67-30"><a href="#cb67-30" aria-hidden="true" tabindex="-1"></a>    out<span class="sc">$</span>snp <span class="ot">&lt;-</span> <span class="fu">rownames</span>(out)</span>
<span id="cb67-31"><a href="#cb67-31" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-32"><a href="#cb67-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(out)</span>
<span id="cb67-33"><a href="#cb67-33" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span> </span>
<span id="cb67-34"><a href="#cb67-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># bind into single DF</span></span>
<span id="cb67-35"><a href="#cb67-35" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;phenotype&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb67-36"><a href="#cb67-36" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove NA</span></span>
<span id="cb67-37"><a href="#cb67-37" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">drop_na</span>()</span>
<span id="cb67-38"><a href="#cb67-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb67-39"><a href="#cb67-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(fst_out_df)</span>
<span id="cb67-40"><a href="#cb67-40" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> </span>
<span id="cb67-41"><a href="#cb67-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;hit_control&quot;</span>)</span>
<span id="cb67-42"><a href="#cb67-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-43"><a href="#cb67-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Recode phenotype</span></span>
<span id="cb67-44"><a href="#cb67-44" aria-hidden="true" tabindex="-1"></a>fst_out<span class="sc">$</span>phenotype <span class="ot">&lt;-</span> <span class="fu">factor</span>(fst_out<span class="sc">$</span>phenotype, <span class="at">levels =</span> trait_levels)</span>
<span id="cb67-45"><a href="#cb67-45" aria-hidden="true" tabindex="-1"></a>fst_out<span class="sc">$</span>phenotype <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">recode</span>(fst_out<span class="sc">$</span>phenotype, <span class="sc">!!!</span>recode_vec)</span></code></pre></div>
<p>Write to file</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>out_dir <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210203_results&quot;</span>)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>out_path <span class="ot">=</span> <span class="fu">file.path</span>(out_dir, <span class="fu">paste</span>(<span class="st">&quot;20210203_fst&quot;</span>, <span class="st">&quot;.csv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(out_dir)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(fst_out, out_path)</span></code></pre></div>
<p>Read back in</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>fst_out <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;20210203_results&quot;</span>, <span class="st">&quot;20210203_fst.csv&quot;</span>))</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>fst_out<span class="sc">$</span>phenotype <span class="ot">&lt;-</span> <span class="fu">factor</span>(fst_out<span class="sc">$</span>phenotype, <span class="at">levels =</span> trait_levels_verb)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>fst_out<span class="sc">$</span>hit_control <span class="ot">=</span> <span class="fu">factor</span>(fst_out<span class="sc">$</span>hit_control, <span class="at">levels =</span> hit_control_levels)</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(fst_out))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">hit_control</th>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<th align="left">phenotype</th>
<th align="right">Fit</th>
<th align="right">Fst</th>
<th align="right">Fis</th>
<th align="left">snp</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<<<<<<< HEAD
<td align="left">case</td>
=======
<td align="left">hit</td>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<td align="left">Height</td>
<td align="right">1</td>
<td align="right">0.0543812</td>
<td align="right">1</td>
<td align="left">rs2710889</td>
</tr>
<tr class="even">
<<<<<<< HEAD
<td align="left">case</td>
=======
<td align="left">hit</td>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<td align="left">Height</td>
<td align="right">1</td>
<td align="right">0.2898026</td>
<td align="right">1</td>
<td align="left">rs1240697</td>
</tr>
<tr class="odd">
<<<<<<< HEAD
<td align="left">case</td>
=======
<td align="left">hit</td>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<td align="left">Height</td>
<td align="right">1</td>
<td align="right">0.0972836</td>
<td align="right">1</td>
<td align="left">rs28401288</td>
</tr>
<tr class="even">
<<<<<<< HEAD
<td align="left">case</td>
=======
<td align="left">hit</td>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<td align="left">Height</td>
<td align="right">1</td>
<td align="right">0.0669376</td>
<td align="right">1</td>
<td align="left">rs377599</td>
</tr>
<tr class="odd">
<<<<<<< HEAD
<td align="left">case</td>
=======
<td align="left">hit</td>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<td align="left">Height</td>
<td align="right">1</td>
<td align="right">0.0274605</td>
<td align="right">1</td>
<td align="left">rs12028979</td>
</tr>
<tr class="even">
<<<<<<< HEAD
<td align="left">case</td>
=======
<td align="left">hit</td>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<td align="left">Height</td>
<td align="right">1</td>
<td align="right">0.0888646</td>
<td align="right">1</td>
<td align="left">rs5024246</td>
</tr>
</tbody>
</table>
</div>
<<<<<<< HEAD
<div id="fst-histograms" class="section level4">
<h4><span class="header-section-number">2.4.1.2</span> <em>Fst</em> histograms</h4>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1"></a>one =<span class="st"> </span>fst_out <span class="op">%&gt;%</span></span>
<span id="cb67-2"><a href="#cb67-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;case&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb67-3"><a href="#cb67-3"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb67-4"><a href="#cb67-4"></a><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype), <span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb67-5"><a href="#cb67-5"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb67-6"><a href="#cb67-6"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb67-7"><a href="#cb67-7"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_primary)  <span class="op">+</span></span>
<span id="cb67-8"><a href="#cb67-8"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb67-9"><a href="#cb67-9"></a></span>
<span id="cb67-10"><a href="#cb67-10"></a>two =<span class="st"> </span>fst_out <span class="op">%&gt;%</span></span>
<span id="cb67-11"><a href="#cb67-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb67-12"><a href="#cb67-12"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb67-13"><a href="#cb67-13"></a><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype), <span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb67-14"><a href="#cb67-14"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb67-15"><a href="#cb67-15"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb67-16"><a href="#cb67-16"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_secondary) <span class="op">+</span></span>
<span id="cb67-17"><a href="#cb67-17"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb67-18"><a href="#cb67-18"></a></span>
<span id="cb67-19"><a href="#cb67-19"></a>one</span>
<span id="cb67-20"><a href="#cb67-20"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-44-1.png" alt="Cases vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-44-2.png" alt="Cases vs controls" width="50%" />
<p class="caption">
Cases vs controls
</p>
</div>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_histograms&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb68-2"><a href="#cb68-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb68-3"><a href="#cb68-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb68-4"><a href="#cb68-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb68-5"><a href="#cb68-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb68-6"><a href="#cb68-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_histograms&quot;</span>, <span class="st">&quot;20210127_controls_all.png&quot;</span>),</span>
<span id="cb69-2"><a href="#cb69-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb69-3"><a href="#cb69-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb69-4"><a href="#cb69-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb69-5"><a href="#cb69-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb69-6"><a href="#cb69-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="fst-density" class="section level4">
<h4><span class="header-section-number">2.4.1.3</span> <em>Fst</em> density</h4>
<div id="facets" class="section level5">
<h5><span class="header-section-number">2.4.1.3.1</span> Facets</h5>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1"></a>one =<span class="st"> </span>fst_out <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb70-2"><a href="#cb70-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;case&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb70-3"><a href="#cb70-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype)) <span class="op">+</span></span>
<span id="cb70-4"><a href="#cb70-4"></a><span class="st">    </span><span class="kw">geom_density</span>() <span class="op">+</span></span>
<span id="cb70-5"><a href="#cb70-5"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Phenotype&quot;</span>) <span class="op">+</span></span>
<span id="cb70-6"><a href="#cb70-6"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb70-7"><a href="#cb70-7"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span></span>
<span id="cb70-8"><a href="#cb70-8"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb70-9"><a href="#cb70-9"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_primary) <span class="op">+</span></span>
<span id="cb70-10"><a href="#cb70-10"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb70-11"><a href="#cb70-11"></a></span>
<span id="cb70-12"><a href="#cb70-12"></a></span>
<span id="cb70-13"><a href="#cb70-13"></a>two =<span class="st"> </span>fst_out <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb70-14"><a href="#cb70-14"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb70-15"><a href="#cb70-15"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype)) <span class="op">+</span></span>
<span id="cb70-16"><a href="#cb70-16"></a><span class="st">    </span><span class="kw">geom_density</span>() <span class="op">+</span></span>
<span id="cb70-17"><a href="#cb70-17"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Phenotype&quot;</span>) <span class="op">+</span></span>
<span id="cb70-18"><a href="#cb70-18"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb70-19"><a href="#cb70-19"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span></span>
<span id="cb70-20"><a href="#cb70-20"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb70-21"><a href="#cb70-21"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_secondary) <span class="op">+</span></span>
<span id="cb70-22"><a href="#cb70-22"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb70-23"><a href="#cb70-23"></a></span>
<span id="cb70-24"><a href="#cb70-24"></a>one</span>
<span id="cb70-25"><a href="#cb70-25"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-49-1.png" alt="Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-49-2.png" alt="Hits vs controls" width="50%" />
=======
<div id="fst-histograms" class="section level4" number="2.4.1.2">
<h4 number="2.4.1.2"><span class="header-section-number">2.4.1.2</span> <em>Fst</em> histograms</h4>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>one <span class="ot">=</span> fst_out <span class="sc">%&gt;%</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(Fst, <span class="at">fill =</span> phenotype), <span class="at">bins =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary)  <span class="sc">+</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">450</span>) <span class="sc">+</span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Hits&quot;</span>)</span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>two <span class="ot">=</span> fst_out <span class="sc">%&gt;%</span></span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;control&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(Fst, <span class="at">fill =</span> phenotype), <span class="at">bins =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span></span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_secondary) <span class="sc">+</span></span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">450</span>) <span class="sc">+</span></span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Controls&quot;</span>)</span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a>one</span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-47-1.svg" alt="Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-47-2.svg" alt="Hits vs controls" width="50%" />
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<p class="caption">
Hits vs controls
</p>
</div>
<<<<<<< HEAD
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_densities&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb71-2"><a href="#cb71-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb71-3"><a href="#cb71-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb71-4"><a href="#cb71-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb71-5"><a href="#cb71-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb71-6"><a href="#cb71-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_densities&quot;</span>, <span class="st">&quot;20210127_controls_all.png&quot;</span>),</span>
<span id="cb72-2"><a href="#cb72-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb72-3"><a href="#cb72-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb72-4"><a href="#cb72-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb72-5"><a href="#cb72-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb72-6"><a href="#cb72-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="ridges" class="section level5">
<h5><span class="header-section-number">2.4.1.3.2</span> Ridges</h5>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1"></a>one =<span class="st"> </span>fst_out <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb73-2"><a href="#cb73-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;case&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb73-3"><a href="#cb73-3"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb73-4"><a href="#cb73-4"></a><span class="st">    </span><span class="kw">geom_density_ridges2</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Fst, <span class="dt">y =</span> phenotype, <span class="dt">fill =</span> phenotype),</span>
<span id="cb73-5"><a href="#cb73-5"></a>                         <span class="dt">scale =</span> <span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb73-6"><a href="#cb73-6"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_primary) <span class="op">+</span></span>
<span id="cb73-7"><a href="#cb73-7"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="dt">label =</span> <span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb73-8"><a href="#cb73-8"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb73-9"><a href="#cb73-9"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F) <span class="op">+</span></span>
<span id="cb73-10"><a href="#cb73-10"></a><span class="st">    </span><span class="kw">scale_y_discrete</span>(<span class="dt">expand =</span> <span class="kw">expand_scale</span>(<span class="dt">add =</span> <span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>)))</span>
<span id="cb73-11"><a href="#cb73-11"></a></span>
<span id="cb73-12"><a href="#cb73-12"></a>two =<span class="st"> </span>fst_out <span class="op">%&gt;%</span></span>
<span id="cb73-13"><a href="#cb73-13"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb73-14"><a href="#cb73-14"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb73-15"><a href="#cb73-15"></a><span class="st">    </span><span class="kw">geom_density_ridges2</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Fst, <span class="dt">y =</span> phenotype, <span class="dt">fill =</span> phenotype),</span>
<span id="cb73-16"><a href="#cb73-16"></a>                         <span class="dt">scale =</span> <span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb73-17"><a href="#cb73-17"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_secondary) <span class="op">+</span></span>
<span id="cb73-18"><a href="#cb73-18"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="dt">label =</span> <span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb73-19"><a href="#cb73-19"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb73-20"><a href="#cb73-20"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F) <span class="op">+</span></span>
<span id="cb73-21"><a href="#cb73-21"></a><span class="st">    </span><span class="kw">scale_y_discrete</span>(<span class="dt">expand =</span> <span class="kw">expand_scale</span>(<span class="dt">add =</span> <span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>)))</span>
<span id="cb73-22"><a href="#cb73-22"></a></span>
<span id="cb73-23"><a href="#cb73-23"></a>one</span>
<span id="cb73-24"><a href="#cb73-24"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-54-1.png" alt="Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-54-2.png" alt="Hits vs controls" width="50%" />
=======
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">file.path</span>(plot_path, <span class="st">&quot;histograms_hits.svg&quot;</span>),</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;svg&quot;</span>,</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>out_path <span class="ot">=</span> <span class="fu">file.path</span>(plot_path, <span class="st">&quot;histograms_rndm.svg&quot;</span>)</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(out_path,</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;svg&quot;</span>,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="fst-density" class="section level4" number="2.4.1.3">
<h4 number="2.4.1.3"><span class="header-section-number">2.4.1.3</span> <em>Fst</em> density</h4>
<div id="facets" class="section level5" number="2.4.1.3.1">
<h5 number="2.4.1.3.1"><span class="header-section-number">2.4.1.3.1</span> Facets</h5>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>one <span class="ot">=</span> fst_out <span class="sc">%&gt;%</span> </span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(Fst, <span class="at">fill =</span> phenotype)) <span class="sc">+</span></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">&quot;Phenotype&quot;</span>) <span class="sc">+</span></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="sc">+</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Hits&quot;</span>)</span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>two <span class="ot">=</span> fst_out <span class="sc">%&gt;%</span> </span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;control&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(Fst, <span class="at">fill =</span> phenotype)) <span class="sc">+</span></span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">&quot;Phenotype&quot;</span>) <span class="sc">+</span></span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>phenotype) <span class="sc">+</span></span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="sc">+</span></span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb73-23"><a href="#cb73-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_secondary) <span class="sc">+</span></span>
<span id="cb73-24"><a href="#cb73-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb73-25"><a href="#cb73-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb73-26"><a href="#cb73-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Controls&quot;</span>)</span>
<span id="cb73-27"><a href="#cb73-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-28"><a href="#cb73-28" aria-hidden="true" tabindex="-1"></a>one</span>
<span id="cb73-29"><a href="#cb73-29" aria-hidden="true" tabindex="-1"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-52-1.svg" alt="Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-52-2.svg" alt="Hits vs controls" width="50%" />
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<p class="caption">
Hits vs controls
</p>
</div>
<<<<<<< HEAD
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_ridges&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb74-2"><a href="#cb74-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb74-3"><a href="#cb74-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb74-4"><a href="#cb74-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb74-5"><a href="#cb74-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb74-6"><a href="#cb74-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_ridges&quot;</span>, <span class="st">&quot;20210127_controls_all.png&quot;</span>),</span>
<span id="cb75-2"><a href="#cb75-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb75-3"><a href="#cb75-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb75-4"><a href="#cb75-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb75-5"><a href="#cb75-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb75-6"><a href="#cb75-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
</div>
</div>
<div id="run-kolmogorov-smirnov-tests" class="section level3">
<h3><span class="header-section-number">2.4.2</span> Run Kolmogorov-Smirnov Tests</h3>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1"></a>case_control =<span class="st"> </span><span class="kw">unique</span>(fst_out<span class="op">$</span>case_control)</span>
<span id="cb76-2"><a href="#cb76-2"></a><span class="kw">names</span>(case_control) =<span class="st"> </span>case_control</span>
<span id="cb76-3"><a href="#cb76-3"></a></span>
<span id="cb76-4"><a href="#cb76-4"></a>ks_out =<span class="st"> </span><span class="kw">lapply</span>(case_control, <span class="cf">function</span>(dataset){</span>
<span id="cb76-5"><a href="#cb76-5"></a>  <span class="co"># filter dataset</span></span>
<span id="cb76-6"><a href="#cb76-6"></a>  target_df =<span class="st"> </span>fst_out[fst_out<span class="op">$</span>case_control <span class="op">==</span><span class="st"> </span>dataset, ]</span>
<span id="cb76-7"><a href="#cb76-7"></a>  <span class="co"># run pairwise KS tests</span></span>
<span id="cb76-8"><a href="#cb76-8"></a>  ks_out =<span class="st"> </span><span class="kw">lapply</span>(trait_levels_verb, <span class="cf">function</span>(trait_a){</span>
<span id="cb76-9"><a href="#cb76-9"></a>    out =<span class="st"> </span><span class="kw">lapply</span>(trait_levels_verb, <span class="cf">function</span>(trait_b){</span>
<span id="cb76-10"><a href="#cb76-10"></a>      results =<span class="st"> </span><span class="kw">ks.test</span>(target_df<span class="op">$</span>Fst[target_df<span class="op">$</span>phenotype <span class="op">==</span><span class="st"> </span>trait_a],</span>
<span id="cb76-11"><a href="#cb76-11"></a>                        target_df<span class="op">$</span>Fst[target_df<span class="op">$</span>phenotype <span class="op">==</span><span class="st"> </span>trait_b])</span>
<span id="cb76-12"><a href="#cb76-12"></a>      P =<span class="st"> </span>results<span class="op">$</span>p.value</span>
<span id="cb76-13"><a href="#cb76-13"></a>      </span>
<span id="cb76-14"><a href="#cb76-14"></a>      <span class="kw">return</span>(P)</span>
<span id="cb76-15"><a href="#cb76-15"></a>    }) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb76-16"><a href="#cb76-16"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;test_b&quot;</span>)</span>
<span id="cb76-17"><a href="#cb76-17"></a>    </span>
<span id="cb76-18"><a href="#cb76-18"></a>    <span class="kw">return</span>(out)</span>
<span id="cb76-19"><a href="#cb76-19"></a>  })  <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb76-20"><a href="#cb76-20"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;trait&quot;</span>)</span>
<span id="cb76-21"><a href="#cb76-21"></a>  </span>
<span id="cb76-22"><a href="#cb76-22"></a>  traits =<span class="st"> </span>ks_out<span class="op">$</span>trait</span>
<span id="cb76-23"><a href="#cb76-23"></a>  ks_out<span class="op">$</span>trait &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb76-24"><a href="#cb76-24"></a>  </span>
<span id="cb76-25"><a href="#cb76-25"></a>  <span class="kw">rownames</span>(ks_out) =<span class="st"> </span>traits</span>
<span id="cb76-26"><a href="#cb76-26"></a>  <span class="kw">return</span>(ks_out)</span>
<span id="cb76-27"><a href="#cb76-27"></a>})</span>
<span id="cb76-28"><a href="#cb76-28"></a></span>
<span id="cb76-29"><a href="#cb76-29"></a><span class="co"># convert to matrix</span></span>
<span id="cb76-30"><a href="#cb76-30"></a>ks_mat =<span class="st"> </span><span class="kw">lapply</span>(ks_out, <span class="cf">function</span>(dataset){</span>
<span id="cb76-31"><a href="#cb76-31"></a>  out =<span class="st"> </span><span class="kw">as.matrix</span>(dataset)</span>
<span id="cb76-32"><a href="#cb76-32"></a>  </span>
<span id="cb76-33"><a href="#cb76-33"></a>  <span class="kw">return</span>(out)</span>
<span id="cb76-34"><a href="#cb76-34"></a>})</span></code></pre></div>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1"></a><span class="co"># Process for plotting</span></span>
<span id="cb77-2"><a href="#cb77-2"></a>ks_out_gg =<span class="st"> </span><span class="kw">lapply</span>(ks_out, <span class="cf">function</span>(dataset){</span>
<span id="cb77-3"><a href="#cb77-3"></a>  out =<span class="st"> </span>dataset</span>
<span id="cb77-4"><a href="#cb77-4"></a>  out<span class="op">$</span>A =<span class="st"> </span><span class="kw">rownames</span>(dataset)</span>
<span id="cb77-5"><a href="#cb77-5"></a>  out =<span class="st"> </span>out <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb77-6"><a href="#cb77-6"></a><span class="st">    </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="op">-</span>A, <span class="dt">names_to =</span> <span class="st">&quot;B&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;ks_P&quot;</span>)</span>
<span id="cb77-7"><a href="#cb77-7"></a>  <span class="co"># convert P-values to -log10</span></span>
<span id="cb77-8"><a href="#cb77-8"></a>  out<span class="op">$</span>ks_P =<span class="st"> </span><span class="op">-</span><span class="kw">log10</span>(out<span class="op">$</span>ks_P)</span>
<span id="cb77-9"><a href="#cb77-9"></a>  </span>
<span id="cb77-10"><a href="#cb77-10"></a>  <span class="kw">return</span>(out)</span>
<span id="cb77-11"><a href="#cb77-11"></a>}) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb77-12"><a href="#cb77-12"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;case_control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb77-13"><a href="#cb77-13"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="kw">across</span>(<span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>),</span>
<span id="cb77-14"><a href="#cb77-14"></a>                       <span class="op">~</span><span class="kw">factor</span>(.x, <span class="dt">levels =</span> trait_levels_verb)))</span></code></pre></div>
<p>Plot</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1"></a>heat_hits_all =<span class="st"> </span>ks_out_gg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-2"><a href="#cb78-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;case&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-3"><a href="#cb78-3"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb78-4"><a href="#cb78-4"></a><span class="st">    </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(A, B, <span class="dt">fill =</span> ks_P)) <span class="op">+</span></span>
<span id="cb78-5"><a href="#cb78-5"></a><span class="st">    </span><span class="kw">scale_fill_viridis_c</span>() <span class="op">+</span></span>
<span id="cb78-6"><a href="#cb78-6"></a><span class="st">    </span><span class="kw">coord_fixed</span>() <span class="op">+</span></span>
<span id="cb78-7"><a href="#cb78-7"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb78-8"><a href="#cb78-8"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb78-9"><a href="#cb78-9"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;KS-test</span><span class="ch">\n</span><span class="st">-log(P)&quot;</span>)</span>
<span id="cb78-10"><a href="#cb78-10"></a></span>
<span id="cb78-11"><a href="#cb78-11"></a>heat_controls_all =<span class="st"> </span>ks_out_gg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-12"><a href="#cb78-12"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-13"><a href="#cb78-13"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb78-14"><a href="#cb78-14"></a><span class="st">    </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(A, B, <span class="dt">fill =</span> ks_P)) <span class="op">+</span></span>
<span id="cb78-15"><a href="#cb78-15"></a><span class="st">    </span><span class="kw">scale_fill_viridis_c</span>(<span class="dt">option =</span> <span class="st">&quot;magma&quot;</span>) <span class="op">+</span></span>
<span id="cb78-16"><a href="#cb78-16"></a><span class="st">    </span><span class="kw">coord_fixed</span>() <span class="op">+</span></span>
<span id="cb78-17"><a href="#cb78-17"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb78-18"><a href="#cb78-18"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb78-19"><a href="#cb78-19"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;KS-test</span><span class="ch">\n</span><span class="st">-log(P)&quot;</span>)</span>
<span id="cb78-20"><a href="#cb78-20"></a></span>
<span id="cb78-21"><a href="#cb78-21"></a>heat_hits_all</span>
<span id="cb78-22"><a href="#cb78-22"></a>heat_controls_all</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-61-1.png" alt="Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-61-2.png" alt="Hits vs controls" width="50%" />
=======
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">file.path</span>(plot_path, <span class="st">&quot;density_hits.svg&quot;</span>),</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;svg&quot;</span>,</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">file.path</span>(plot_path, <span class="st">&quot;density_rndm.svg&quot;</span>),</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;svg&quot;</span>,</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="ridges" class="section level5" number="2.4.1.3.2">
<h5 number="2.4.1.3.2"><span class="header-section-number">2.4.1.3.2</span> Ridges</h5>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>one <span class="ot">=</span> fst_out <span class="sc">%&gt;%</span> </span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">phenotype =</span> <span class="fu">factor</span>(phenotype, <span class="at">levels =</span> <span class="fu">rev</span>(trait_levels_verb))) <span class="sc">%&gt;%</span> </span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density_ridges2</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> Fst, <span class="at">y =</span> phenotype, <span class="at">fill =</span> phenotype),</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">scale =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="at">label =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expand_scale</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>))) <span class="sc">+</span></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Hits&quot;</span>)</span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a>two <span class="ot">=</span> fst_out <span class="sc">%&gt;%</span></span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;control&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">phenotype =</span> <span class="fu">factor</span>(phenotype, <span class="at">levels =</span> <span class="fu">rev</span>(trait_levels_verb))) <span class="sc">%&gt;%</span> </span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density_ridges2</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> Fst, <span class="at">y =</span> phenotype, <span class="at">fill =</span> phenotype),</span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>                         <span class="at">scale =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb76-20"><a href="#cb76-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_secondary) <span class="sc">+</span></span>
<span id="cb76-21"><a href="#cb76-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="at">label =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb76-22"><a href="#cb76-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb76-23"><a href="#cb76-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb76-24"><a href="#cb76-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expand_scale</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>))) <span class="sc">+</span></span>
<span id="cb76-25"><a href="#cb76-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Controls&quot;</span>)</span>
<span id="cb76-26"><a href="#cb76-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-27"><a href="#cb76-27" aria-hidden="true" tabindex="-1"></a>one</span>
<span id="cb76-28"><a href="#cb76-28" aria-hidden="true" tabindex="-1"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-57-1.svg" alt="Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-57-2.svg" alt="Hits vs controls" width="50%" />
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
<p class="caption">
Hits vs controls
</p>
</div>
<<<<<<< HEAD
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_ks_heatmaps&quot;</span>, <span class="st">&quot;20210127_hits_all.png&quot;</span>),</span>
<span id="cb79-2"><a href="#cb79-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb79-3"><a href="#cb79-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb79-4"><a href="#cb79-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb79-5"><a href="#cb79-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb79-6"><a href="#cb79-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_ks_heatmaps&quot;</span>, <span class="st">&quot;20210127_controls_all.png&quot;</span>),</span>
<span id="cb80-2"><a href="#cb80-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb80-3"><a href="#cb80-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb80-4"><a href="#cb80-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb80-5"><a href="#cb80-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb80-6"><a href="#cb80-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="try-with-same-number-of-snps" class="section level3">
<h3><span class="header-section-number">2.4.3</span> Try with same number of SNPs</h3>
<p><span style="color: red;"><strong>Strategy</strong></span>: Will it make a difference if we analyse the same number of SNPs for each trait? i.e. is the different number of SNPs for each trait affecting the KS test output?</p>
<p>IBD only has 182 hits, so remove from this part of the analysis.</p>
<p>The trait with the next least number of hits is <code>Intelligence</code>, with `480. So we’ll take a random sample of 40</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1"></a>fst_sample =<span class="st"> </span><span class="kw">split</span>(fst_out, <span class="dt">f =</span> fst_out<span class="op">$</span>case_control)</span>
<span id="cb81-2"><a href="#cb81-2"></a>fst_sample =<span class="st"> </span><span class="kw">lapply</span>(fst_sample, <span class="cf">function</span>(dataset){</span>
<span id="cb81-3"><a href="#cb81-3"></a>  out =<span class="st"> </span><span class="kw">split</span>(dataset, <span class="dt">f =</span> dataset<span class="op">$</span>phenotype)</span>
<span id="cb81-4"><a href="#cb81-4"></a>  out =<span class="st"> </span><span class="kw">lapply</span>(out, <span class="cf">function</span>(pheno){</span>
<span id="cb81-5"><a href="#cb81-5"></a>    pheno =<span class="st"> </span>pheno <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb81-6"><a href="#cb81-6"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">slice_sample</span>(<span class="dt">n =</span> <span class="dv">480</span>)</span>
<span id="cb81-7"><a href="#cb81-7"></a>    </span>
<span id="cb81-8"><a href="#cb81-8"></a>    <span class="kw">return</span>(pheno)</span>
<span id="cb81-9"><a href="#cb81-9"></a>  }) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb81-10"><a href="#cb81-10"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb81-11"><a href="#cb81-11"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(phenotype <span class="op">!=</span><span class="st"> &quot;IBD&quot;</span>)</span>
<span id="cb81-12"><a href="#cb81-12"></a>}) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb81-13"><a href="#cb81-13"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>()</span></code></pre></div>
<div id="histograms" class="section level4">
<h4><span class="header-section-number">2.4.3.1</span> Histograms</h4>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1"></a>one =<span class="st"> </span>fst_sample <span class="op">%&gt;%</span></span>
<span id="cb82-2"><a href="#cb82-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;case&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb82-3"><a href="#cb82-3"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb82-4"><a href="#cb82-4"></a><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype), <span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb82-5"><a href="#cb82-5"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb82-6"><a href="#cb82-6"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb82-7"><a href="#cb82-7"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_primary)  <span class="op">+</span></span>
<span id="cb82-8"><a href="#cb82-8"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb82-9"><a href="#cb82-9"></a></span>
<span id="cb82-10"><a href="#cb82-10"></a>two =<span class="st"> </span>fst_sample <span class="op">%&gt;%</span></span>
<span id="cb82-11"><a href="#cb82-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb82-12"><a href="#cb82-12"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb82-13"><a href="#cb82-13"></a><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype), <span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb82-14"><a href="#cb82-14"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb82-15"><a href="#cb82-15"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb82-16"><a href="#cb82-16"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_secondary) <span class="op">+</span></span>
<span id="cb82-17"><a href="#cb82-17"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb82-18"><a href="#cb82-18"></a></span>
<span id="cb82-19"><a href="#cb82-19"></a>one</span>
<span id="cb82-20"><a href="#cb82-20"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-67-1.png" alt="SAMPLED: Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-67-2.png" alt="SAMPLED: Hits vs controls" width="50%" />
<p class="caption">
SAMPLED: Hits vs controls
</p>
</div>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_histograms&quot;</span>, <span class="st">&quot;20210127_hits_sample.png&quot;</span>),</span>
<span id="cb83-2"><a href="#cb83-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb83-3"><a href="#cb83-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb83-4"><a href="#cb83-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb83-5"><a href="#cb83-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb83-6"><a href="#cb83-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_histograms&quot;</span>, <span class="st">&quot;20210127_controls_sample.png&quot;</span>),</span>
<span id="cb84-2"><a href="#cb84-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb84-3"><a href="#cb84-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb84-4"><a href="#cb84-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb84-5"><a href="#cb84-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb84-6"><a href="#cb84-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
<div id="density" class="section level4">
<h4><span class="header-section-number">2.4.3.2</span> Density</h4>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1"></a>one =<span class="st"> </span>fst_sample <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb85-2"><a href="#cb85-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;case&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb85-3"><a href="#cb85-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype)) <span class="op">+</span></span>
<span id="cb85-4"><a href="#cb85-4"></a><span class="st">    </span><span class="kw">geom_density</span>() <span class="op">+</span></span>
<span id="cb85-5"><a href="#cb85-5"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Phenotype&quot;</span>) <span class="op">+</span></span>
<span id="cb85-6"><a href="#cb85-6"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb85-7"><a href="#cb85-7"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span></span>
<span id="cb85-8"><a href="#cb85-8"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb85-9"><a href="#cb85-9"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_primary) <span class="op">+</span></span>
<span id="cb85-10"><a href="#cb85-10"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb85-11"><a href="#cb85-11"></a></span>
<span id="cb85-12"><a href="#cb85-12"></a>two =<span class="st"> </span>fst_sample <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb85-13"><a href="#cb85-13"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb85-14"><a href="#cb85-14"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(Fst, <span class="dt">fill =</span> phenotype)) <span class="op">+</span></span>
<span id="cb85-15"><a href="#cb85-15"></a><span class="st">    </span><span class="kw">geom_density</span>() <span class="op">+</span></span>
<span id="cb85-16"><a href="#cb85-16"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Phenotype&quot;</span>) <span class="op">+</span></span>
<span id="cb85-17"><a href="#cb85-17"></a><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>phenotype) <span class="op">+</span></span>
<span id="cb85-18"><a href="#cb85-18"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span></span>
<span id="cb85-19"><a href="#cb85-19"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span></span>
<span id="cb85-20"><a href="#cb85-20"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> pal_secondary) <span class="op">+</span></span>
<span id="cb85-21"><a href="#cb85-21"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">fill =</span> F)</span>
<span id="cb85-22"><a href="#cb85-22"></a></span>
<span id="cb85-23"><a href="#cb85-23"></a>one</span>
<span id="cb85-24"><a href="#cb85-24"></a>two</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-72-1.png" alt="SAMPLED: Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-72-2.png" alt="SAMPLED: Hits vs controls" width="50%" />
<p class="caption">
SAMPLED: Hits vs controls
</p>
</div>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_densities&quot;</span>, <span class="st">&quot;20210127_hits_sample.png&quot;</span>),</span>
<span id="cb86-2"><a href="#cb86-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb86-3"><a href="#cb86-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb86-4"><a href="#cb86-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb86-5"><a href="#cb86-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb86-6"><a href="#cb86-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_densities&quot;</span>, <span class="st">&quot;20210127_controls_sample.png&quot;</span>),</span>
<span id="cb87-2"><a href="#cb87-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb87-3"><a href="#cb87-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb87-4"><a href="#cb87-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb87-5"><a href="#cb87-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb87-6"><a href="#cb87-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
</div>
<div id="ks-test" class="section level3">
<h3><span class="header-section-number">2.4.4</span> KS test</h3>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1"></a>case_control =<span class="st"> </span><span class="kw">unique</span>(fst_sample<span class="op">$</span>case_control)</span>
<span id="cb88-2"><a href="#cb88-2"></a><span class="kw">names</span>(case_control) =<span class="st"> </span>case_control</span>
<span id="cb88-3"><a href="#cb88-3"></a></span>
<span id="cb88-4"><a href="#cb88-4"></a>trait_levels_verb_sample =<span class="st"> </span>trait_levels_verb[<span class="op">-</span><span class="kw">which</span>(trait_levels_verb <span class="op">==</span><span class="st"> &quot;IBD&quot;</span>)]</span>
<span id="cb88-5"><a href="#cb88-5"></a>traits_sample =<span class="st"> </span>traits[<span class="op">-</span><span class="kw">which</span>(traits <span class="op">==</span><span class="st"> &quot;ibd&quot;</span>)]</span>
<span id="cb88-6"><a href="#cb88-6"></a></span>
<span id="cb88-7"><a href="#cb88-7"></a>ks_sample =<span class="st"> </span><span class="kw">lapply</span>(case_control, <span class="cf">function</span>(dataset){</span>
<span id="cb88-8"><a href="#cb88-8"></a>  <span class="co"># filter dataset</span></span>
<span id="cb88-9"><a href="#cb88-9"></a>  target_df =<span class="st"> </span>fst_sample[fst_sample<span class="op">$</span>case_control <span class="op">==</span><span class="st"> </span>dataset, ]</span>
<span id="cb88-10"><a href="#cb88-10"></a>  <span class="co"># run pairwise KS tests</span></span>
<span id="cb88-11"><a href="#cb88-11"></a>  ks_out =<span class="st"> </span><span class="kw">lapply</span>(trait_levels_verb_sample, <span class="cf">function</span>(trait_a){</span>
<span id="cb88-12"><a href="#cb88-12"></a>    out =<span class="st"> </span><span class="kw">lapply</span>(trait_levels_verb_sample, <span class="cf">function</span>(trait_b){</span>
<span id="cb88-13"><a href="#cb88-13"></a>      results =<span class="st"> </span><span class="kw">ks.test</span>(target_df<span class="op">$</span>Fst[target_df<span class="op">$</span>phenotype <span class="op">==</span><span class="st"> </span>trait_a],</span>
<span id="cb88-14"><a href="#cb88-14"></a>                        target_df<span class="op">$</span>Fst[target_df<span class="op">$</span>phenotype <span class="op">==</span><span class="st"> </span>trait_b])</span>
<span id="cb88-15"><a href="#cb88-15"></a>      P =<span class="st"> </span>results<span class="op">$</span>p.value</span>
<span id="cb88-16"><a href="#cb88-16"></a>      </span>
<span id="cb88-17"><a href="#cb88-17"></a>      <span class="kw">return</span>(P)</span>
<span id="cb88-18"><a href="#cb88-18"></a>    }) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb88-19"><a href="#cb88-19"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;test_b&quot;</span>)</span>
<span id="cb88-20"><a href="#cb88-20"></a>    </span>
<span id="cb88-21"><a href="#cb88-21"></a>    <span class="kw">return</span>(out)</span>
<span id="cb88-22"><a href="#cb88-22"></a>  })  <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb88-23"><a href="#cb88-23"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;trait&quot;</span>)</span>
<span id="cb88-24"><a href="#cb88-24"></a>  </span>
<span id="cb88-25"><a href="#cb88-25"></a>  traits =<span class="st"> </span>ks_out<span class="op">$</span>trait_levels_verb_sample</span>
<span id="cb88-26"><a href="#cb88-26"></a>  ks_out<span class="op">$</span>trait &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb88-27"><a href="#cb88-27"></a>  </span>
<span id="cb88-28"><a href="#cb88-28"></a>  <span class="kw">rownames</span>(ks_out) =<span class="st"> </span>trait_levels_verb_sample</span>
<span id="cb88-29"><a href="#cb88-29"></a>  <span class="kw">return</span>(ks_out)</span>
<span id="cb88-30"><a href="#cb88-30"></a>})</span></code></pre></div>
<div id="heatmaps" class="section level4">
<h4><span class="header-section-number">2.4.4.1</span> Heatmaps</h4>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1"></a>ks_sample_gg =<span class="st"> </span><span class="kw">lapply</span>(ks_sample, <span class="cf">function</span>(dataset){</span>
<span id="cb89-2"><a href="#cb89-2"></a>  out =<span class="st"> </span>dataset</span>
<span id="cb89-3"><a href="#cb89-3"></a>  out<span class="op">$</span>A =<span class="st"> </span><span class="kw">rownames</span>(dataset)</span>
<span id="cb89-4"><a href="#cb89-4"></a>  out =<span class="st"> </span>out <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb89-5"><a href="#cb89-5"></a><span class="st">    </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> <span class="op">-</span>A, <span class="dt">names_to =</span> <span class="st">&quot;B&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;ks_P&quot;</span>)</span>
<span id="cb89-6"><a href="#cb89-6"></a>  <span class="co"># convert P-values to -log10</span></span>
<span id="cb89-7"><a href="#cb89-7"></a>  out<span class="op">$</span>ks_P =<span class="st"> </span><span class="op">-</span><span class="kw">log10</span>(out<span class="op">$</span>ks_P)</span>
<span id="cb89-8"><a href="#cb89-8"></a>  </span>
<span id="cb89-9"><a href="#cb89-9"></a>  <span class="kw">return</span>(out)</span>
<span id="cb89-10"><a href="#cb89-10"></a>}) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb89-11"><a href="#cb89-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(<span class="dt">.id =</span> <span class="st">&quot;case_control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb89-12"><a href="#cb89-12"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="kw">across</span>(<span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>),</span>
<span id="cb89-13"><a href="#cb89-13"></a>                       <span class="op">~</span><span class="kw">factor</span>(.x, <span class="dt">levels =</span> trait_levels_verb_sample)))</span></code></pre></div>
<p>Plot</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1"></a>heat_hits_sample =<span class="st"> </span>ks_sample_gg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb90-2"><a href="#cb90-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;case&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb90-3"><a href="#cb90-3"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb90-4"><a href="#cb90-4"></a><span class="st">    </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(A, B, <span class="dt">fill =</span> ks_P)) <span class="op">+</span></span>
<span id="cb90-5"><a href="#cb90-5"></a><span class="st">    </span><span class="kw">scale_fill_viridis_c</span>() <span class="op">+</span></span>
<span id="cb90-6"><a href="#cb90-6"></a><span class="st">    </span><span class="kw">coord_fixed</span>() <span class="op">+</span></span>
<span id="cb90-7"><a href="#cb90-7"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb90-8"><a href="#cb90-8"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb90-9"><a href="#cb90-9"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;KS-test</span><span class="ch">\n</span><span class="st">-log(P)&quot;</span>)</span>
<span id="cb90-10"><a href="#cb90-10"></a></span>
<span id="cb90-11"><a href="#cb90-11"></a>heat_controls_sample =<span class="st"> </span>ks_sample_gg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb90-12"><a href="#cb90-12"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_control <span class="op">==</span><span class="st"> &quot;control&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb90-13"><a href="#cb90-13"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span></span>
<span id="cb90-14"><a href="#cb90-14"></a><span class="st">    </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(A, B, <span class="dt">fill =</span> ks_P)) <span class="op">+</span></span>
<span id="cb90-15"><a href="#cb90-15"></a><span class="st">    </span><span class="kw">scale_fill_viridis_c</span>(<span class="dt">option =</span> <span class="st">&quot;magma&quot;</span>) <span class="op">+</span></span>
<span id="cb90-16"><a href="#cb90-16"></a><span class="st">    </span><span class="kw">coord_fixed</span>() <span class="op">+</span></span>
<span id="cb90-17"><a href="#cb90-17"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb90-18"><a href="#cb90-18"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb90-19"><a href="#cb90-19"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;KS-test</span><span class="ch">\n</span><span class="st">-log(P)&quot;</span>)</span>
<span id="cb90-20"><a href="#cb90-20"></a>   </span>
<span id="cb90-21"><a href="#cb90-21"></a>heat_hits_sample</span>
<span id="cb90-22"><a href="#cb90-22"></a>heat_controls_sample</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-79-1.png" alt="SAMPLED: Hits vs controls" width="50%" /><img src="index_files/figure-html/unnamed-chunk-79-2.png" alt="SAMPLED: Hits vs controls" width="50%" />
<p class="caption">
SAMPLED: Hits vs controls
</p>
</div>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_ks_heatmaps&quot;</span>, <span class="st">&quot;20210127_hits_sample.png&quot;</span>),</span>
<span id="cb91-2"><a href="#cb91-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb91-3"><a href="#cb91-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb91-4"><a href="#cb91-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb91-5"><a href="#cb91-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb91-6"><a href="#cb91-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1"></a><span class="kw">ggsave</span>(<span class="kw">here</span>(<span class="st">&quot;plots&quot;</span>, <span class="st">&quot;20210127_ks_heatmaps&quot;</span>, <span class="st">&quot;20210127_controls_sample.png&quot;</span>),</span>
<span id="cb92-2"><a href="#cb92-2"></a>       <span class="dt">device =</span> <span class="st">&quot;png&quot;</span>,</span>
<span id="cb92-3"><a href="#cb92-3"></a>       <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>,</span>
<span id="cb92-4"><a href="#cb92-4"></a>       <span class="dt">dpi =</span> <span class="dv">400</span>,</span>
<span id="cb92-5"><a href="#cb92-5"></a>       <span class="dt">height =</span> <span class="dv">12</span>,</span>
<span id="cb92-6"><a href="#cb92-6"></a>       <span class="dt">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
</div>
<div id="compare-effects-of-sampling-on-p-values" class="section level3">
<h3><span class="header-section-number">2.4.5</span> Compare effects of sampling on P-values</h3>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1"></a>heat_hits_all</span>
<span id="cb93-2"><a href="#cb93-2"></a>heat_hits_sample</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-84-1.png" alt="Hits: all vs sample" width="50%" /><img src="index_files/figure-html/unnamed-chunk-84-2.png" alt="Hits: all vs sample" width="50%" />
<p class="caption">
Hits: all vs sample
</p>
</div>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1"></a>heat_controls_all</span>
<span id="cb94-2"><a href="#cb94-2"></a>heat_controls_sample</span></code></pre></div>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-85-1.png" alt="Controls: all vs sample" width="50%" /><img src="index_files/figure-html/unnamed-chunk-85-2.png" alt="Controls: all vs sample" width="50%" />
<p class="caption">
Controls: all vs sample
</p>
</div>
=======
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">file.path</span>(plot_path, <span class="st">&quot;ridges_hits.svg&quot;</span>),</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;svg&quot;</span>,</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">file.path</span>(plot_path, <span class="st">&quot;ridges_rndm.svg&quot;</span>),</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;svg&quot;</span>,</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>)</span></code></pre></div>
</div>
</div>
</div>
<div id="tests" class="section level3" number="2.4.2">
<h3 number="2.4.2"><span class="header-section-number">2.4.2</span> Tests</h3>
<div id="median-80-and-90-percentiles-ks-test-and-mw-test" class="section level4" number="2.4.2.1">
<h4 number="2.4.2.1"><span class="header-section-number">2.4.2.1</span> Median, 80 and 90 percentiles, KS test, and MW test</h4>
<p><span style="color: red;">Rationale</span>:</p>
<ol style="list-style-type: decimal">
<li>Get the median, and 0.8 and 0.9 percentiles for the <em>Fst</em> density of each trait.</li>
<li>Run one-sided Mann-Whitney tests using <code>Height</code> as the reference against each of the other traits.</li>
<li>Run one-sided Kolmogorov-Smirnov tests using <code>Height</code> as the reference against each of the other traits.</li>
<li>Determine which tests to use to confirm the differences apparent in the forms of the density curves.</li>
</ol>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>final_list <span class="ot">=</span> <span class="fu">split</span>(fst_out, <span class="at">f =</span> fst_out<span class="sc">$</span>hit_control)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>final_list <span class="ot">=</span> <span class="fu">lapply</span>(final_list, <span class="cf">function</span>(dataset){</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  split_data <span class="ot">=</span> <span class="fu">split</span>(dataset, <span class="at">f =</span> dataset<span class="sc">$</span>phenotype)</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> <span class="fu">lapply</span>(split_data, <span class="cf">function</span>(pheno){</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>    stats <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get median and 90 percentile</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>    stats[[<span class="st">&quot;stats&quot;</span>]] <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;median&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(pheno<span class="sc">$</span>Fst),</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">quantile</span>(pheno<span class="sc">$</span>Fst, <span class="at">probs =</span> <span class="fl">0.8</span>),</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">quantile</span>(pheno<span class="sc">$</span>Fst, <span class="at">probs =</span> <span class="fl">0.9</span>))</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get Mann-Whitney results</span></span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a>    stats[[<span class="st">&quot;final_list&quot;</span>]] <span class="ot">=</span> <span class="fu">wilcox.test</span>(<span class="at">x =</span> split_data[[<span class="st">&quot;Height&quot;</span>]]<span class="sc">$</span>Fst,</span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">y =</span> pheno<span class="sc">$</span>Fst,</span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">alternative =</span> <span class="st">&quot;less&quot;</span>,</span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">paired =</span> F,</span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">conf.int =</span> T)</span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get KS results</span></span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a>    stats[[<span class="st">&quot;ks_out&quot;</span>]] <span class="ot">=</span> <span class="fu">ks.test</span>(<span class="at">x =</span> split_data[[<span class="st">&quot;Height&quot;</span>]]<span class="sc">$</span>Fst,</span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a>                                <span class="at">y =</span> pheno<span class="sc">$</span>Fst,</span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a>                                <span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>)</span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-25"><a href="#cb79-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(stats)</span>
<span id="cb79-26"><a href="#cb79-26" aria-hidden="true" tabindex="-1"></a>  })<span class="co"># %&gt;% </span></span>
<span id="cb79-27"><a href="#cb79-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#dplyr::bind_rows(.id = &quot;phenotype&quot;)</span></span>
<span id="cb79-28"><a href="#cb79-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb79-29"><a href="#cb79-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb79-30"><a href="#cb79-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb79-31"><a href="#cb79-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-32"><a href="#cb79-32" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb79-33"><a href="#cb79-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-34"><a href="#cb79-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Collapse into DF</span></span>
<span id="cb79-35"><a href="#cb79-35" aria-hidden="true" tabindex="-1"></a>final_df <span class="ot">=</span> <span class="fu">lapply</span>(final_list, <span class="cf">function</span>(dataset){</span>
<span id="cb79-36"><a href="#cb79-36" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">=</span> <span class="fu">lapply</span>(dataset, <span class="cf">function</span>(pheno){</span>
<span id="cb79-37"><a href="#cb79-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Put key data into data frame</span></span>
<span id="cb79-38"><a href="#cb79-38" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;MEDIAN&quot;</span> <span class="ot">=</span> pheno[[<span class="st">&quot;stats&quot;</span>]][[<span class="st">&quot;median&quot;</span>]],</span>
<span id="cb79-39"><a href="#cb79-39" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;80%&quot;</span> <span class="ot">=</span> pheno[[<span class="st">&quot;stats&quot;</span>]][[<span class="st">&quot;80%&quot;</span>]],</span>
<span id="cb79-40"><a href="#cb79-40" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;90%&quot;</span> <span class="ot">=</span> pheno[[<span class="st">&quot;stats&quot;</span>]][[<span class="st">&quot;90%&quot;</span>]],</span>
<span id="cb79-41"><a href="#cb79-41" aria-hidden="true" tabindex="-1"></a>           pheno[[<span class="st">&quot;final_list&quot;</span>]]<span class="sc">$</span>statistic,</span>
<span id="cb79-42"><a href="#cb79-42" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;P_MW&quot;</span> <span class="ot">=</span> pheno[[<span class="st">&quot;final_list&quot;</span>]]<span class="sc">$</span>p.value,</span>
<span id="cb79-43"><a href="#cb79-43" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;CONF_LOWER&quot;</span> <span class="ot">=</span> pheno[[<span class="st">&quot;final_list&quot;</span>]]<span class="sc">$</span>conf.int[<span class="dv">1</span>],</span>
<span id="cb79-44"><a href="#cb79-44" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;CONF_UPPER&quot;</span> <span class="ot">=</span> pheno[[<span class="st">&quot;final_list&quot;</span>]]<span class="sc">$</span>conf.int[<span class="dv">2</span>],</span>
<span id="cb79-45"><a href="#cb79-45" aria-hidden="true" tabindex="-1"></a>           pheno[[<span class="st">&quot;ks_out&quot;</span>]]<span class="sc">$</span>statistic,</span>
<span id="cb79-46"><a href="#cb79-46" aria-hidden="true" tabindex="-1"></a>           <span class="st">&quot;P_KS&quot;</span> <span class="ot">=</span> pheno[[<span class="st">&quot;ks_out&quot;</span>]]<span class="sc">$</span>p.value)</span>
<span id="cb79-47"><a href="#cb79-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-48"><a href="#cb79-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb79-49"><a href="#cb79-49" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span> </span>
<span id="cb79-50"><a href="#cb79-50" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;PHENO&quot;</span>)</span>
<span id="cb79-51"><a href="#cb79-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb79-52"><a href="#cb79-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb79-53"><a href="#cb79-53" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> </span>
<span id="cb79-54"><a href="#cb79-54" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;HIT_CONTROL&quot;</span>)</span>
<span id="cb79-55"><a href="#cb79-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-56"><a href="#cb79-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Factor</span></span>
<span id="cb79-57"><a href="#cb79-57" aria-hidden="true" tabindex="-1"></a>final_df<span class="sc">$</span>PHENO <span class="ot">&lt;-</span> <span class="fu">factor</span>(final_df<span class="sc">$</span>PHENO, <span class="at">levels =</span> trait_levels_verb)</span>
<span id="cb79-58"><a href="#cb79-58" aria-hidden="true" tabindex="-1"></a>final_df<span class="sc">$</span>HIT_CONTROL <span class="ot">=</span> <span class="fu">factor</span>(final_df<span class="sc">$</span>HIT_CONTROL, <span class="at">levels =</span> hit_control_levels)</span>
<span id="cb79-59"><a href="#cb79-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-60"><a href="#cb79-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Output table</span></span>
<span id="cb79-61"><a href="#cb79-61" aria-hidden="true" tabindex="-1"></a>final_df <span class="sc">%&gt;%</span> </span>
<span id="cb79-62"><a href="#cb79-62" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(PHENO <span class="sc">!=</span> <span class="st">&quot;Height&quot;</span> <span class="sc">&amp;</span> HIT_CONTROL <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb79-63"><a href="#cb79-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>HIT_CONTROL) <span class="sc">%&gt;%</span> </span>
<span id="cb79-64"><a href="#cb79-64" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="20%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="7%" />
<col width="8%" />
<col width="9%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">PHENO</th>
<th align="right">MEDIAN</th>
<th align="right">80%</th>
<th align="right">90%</th>
<th align="right">W</th>
<th align="right">P_MW</th>
<th align="right">CONF_LOWER</th>
<th align="right">CONF_UPPER</th>
<th align="right">D^+</th>
<th align="right">P_KS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">BMI</td>
<td align="right">0.0473438</td>
<td align="right">0.0951012</td>
<td align="right">0.1338033</td>
<td align="right">2014306</td>
<td align="right">0.9994353</td>
<td align="right">-Inf</td>
<td align="right">0.0069362</td>
<td align="right">0.0354889</td>
<td align="right">0.0920426</td>
</tr>
<tr class="even">
<td align="left">Educational attainment</td>
<td align="right">0.0591072</td>
<td align="right">0.1189623</td>
<td align="right">0.1617981</td>
<td align="right">1668872</td>
<td align="right">0.0000070</td>
<td align="right">-Inf</td>
<td align="right">-0.0042002</td>
<td align="right">0.0875242</td>
<td align="right">0.0000007</td>
</tr>
<tr class="odd">
<td align="left">Intelligence</td>
<td align="right">0.0640484</td>
<td align="right">0.1187706</td>
<td align="right">0.1599932</td>
<td align="right">522053</td>
<td align="right">0.0000130</td>
<td align="right">-Inf</td>
<td align="right">-0.0062020</td>
<td align="right">0.1251515</td>
<td align="right">0.0000034</td>
</tr>
<tr class="even">
<td align="left">IBD</td>
<td align="right">0.0559605</td>
<td align="right">0.1180520</td>
<td align="right">0.1704417</td>
<td align="right">205361</td>
<td align="right">0.0233731</td>
<td align="right">-Inf</td>
<td align="right">-0.0012905</td>
<td align="right">0.1019891</td>
<td align="right">0.0293960</td>
</tr>
<tr class="odd">
<td align="left">Pigmentation</td>
<td align="right">0.0513410</td>
<td align="right">0.1238692</td>
<td align="right">0.1724036</td>
<td align="right">621561</td>
<td align="right">0.8097992</td>
<td align="right">-Inf</td>
<td align="right">0.0059789</td>
<td align="right">0.0355927</td>
<td align="right">0.3547547</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="final-fst-density-plots-for-hits" class="section level3" number="2.4.3">
<h3 number="2.4.3"><span class="header-section-number">2.4.3</span> Final <em>Fst</em> density plots for hits</h3>
<p>Some options for the figure.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>fst_out <span class="sc">%&gt;%</span> </span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">phenotype =</span> <span class="fu">factor</span>(phenotype, <span class="at">levels =</span> <span class="fu">rev</span>(trait_levels_verb))) <span class="sc">%&gt;%</span> </span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(Fst, phenotype, <span class="at">fill =</span> phenotype)) <span class="sc">+</span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_density_ridges</span>(<span class="at">geom =</span> <span class="st">&quot;density_ridges_gradient&quot;</span>,</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">scale =</span> <span class="dv">2</span>,</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">calc_ecdf =</span> <span class="cn">TRUE</span>,</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantiles =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.8</span>, <span class="fl">0.9</span>),</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantile_lines =</span> T) <span class="sc">+</span></span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> F) <span class="sc">+</span></span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>))) <span class="sc">+</span></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Median, 80% and 90%; auto-bandwidth&quot;</span>)</span></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.0133</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-63-1.svg" /><!-- --></p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Automatic bandwidth</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>fst_out <span class="sc">%&gt;%</span> </span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">phenotype =</span> <span class="fu">factor</span>(phenotype, <span class="at">levels =</span> <span class="fu">rev</span>(trait_levels_verb))) <span class="sc">%&gt;%</span> </span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(Fst, phenotype, <span class="at">fill =</span> phenotype, <span class="at">colour =</span> phenotype)) <span class="sc">+</span></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density_ridges</span>(<span class="at">scale =</span> <span class="dv">2</span>,</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">calc_ecdf =</span> <span class="cn">TRUE</span>,</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantiles =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.8</span>, <span class="fl">0.9</span>),</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantile_lines =</span> T,</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>                        <span class="at">jittered_points =</span> <span class="cn">TRUE</span>,</span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">point_shape =</span> <span class="st">&#39;|&#39;</span>, <span class="at">alpha =</span> <span class="fl">0.85</span>, <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">position =</span> <span class="fu">position_points_jitter</span>(<span class="at">width =</span> <span class="fl">0.01</span>, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> pal_secondary) <span class="sc">+</span></span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> F, <span class="at">colour =</span> F) <span class="sc">+</span></span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>))) <span class="sc">+</span></span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Median, 80% and 90%; auto-bandwidth; with points&quot;</span>)  </span></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.0133</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-64-1.svg" /><!-- --></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Manually-set bandwidth </span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>fst_out <span class="sc">%&gt;%</span> </span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">phenotype =</span> <span class="fu">factor</span>(phenotype, <span class="at">levels =</span> <span class="fu">rev</span>(trait_levels_verb))) <span class="sc">%&gt;%</span> </span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(Fst, phenotype, <span class="at">fill =</span> phenotype)) <span class="sc">+</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density_ridges</span>(<span class="at">scale =</span> <span class="dv">2</span>,</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">bandwidth =</span> <span class="fl">0.003</span>,</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">calc_ecdf =</span> <span class="cn">TRUE</span>,</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantiles =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.8</span>, <span class="fl">0.9</span>),</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantile_lines =</span> T,</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">jittered_points =</span> <span class="cn">TRUE</span>,</span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">point_shape =</span> <span class="st">&#39;|&#39;</span>, <span class="at">alpha =</span> <span class="fl">0.85</span>, <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">position =</span> <span class="fu">position_points_jitter</span>(<span class="at">width =</span> <span class="fl">0.01</span>, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a> <span class="co"># scale_colour_manual(values = pal_secondary) +</span></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> F, <span class="at">colour =</span> F) <span class="sc">+</span></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>))) <span class="sc">+</span></span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Median, 80% and 90%; manual bandwidth; with points; black lines&quot;</span>)  </span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-65-1.svg" /><!-- --></p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Manually-set bandwidth </span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>fst_out <span class="sc">%&gt;%</span> </span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(hit_control <span class="sc">==</span> <span class="st">&quot;hit&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">phenotype =</span> <span class="fu">factor</span>(phenotype, <span class="at">levels =</span> <span class="fu">rev</span>(trait_levels_verb))) <span class="sc">%&gt;%</span> </span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(Fst, phenotype, <span class="at">fill =</span> phenotype, <span class="at">colour =</span> phenotype)) <span class="sc">+</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density_ridges</span>(<span class="at">scale =</span> <span class="dv">2</span>,</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">bandwidth =</span> <span class="fl">0.003</span>,</span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">calc_ecdf =</span> <span class="cn">TRUE</span>,</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantiles =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.9</span>),</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quantile_lines =</span> T,</span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">jittered_points =</span> <span class="cn">TRUE</span>,</span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">point_shape =</span> <span class="st">&#39;|&#39;</span>, <span class="at">alpha =</span> <span class="fl">0.85</span>, <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">position =</span> <span class="fu">position_points_jitter</span>(<span class="at">width =</span> <span class="fl">0.01</span>, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> pal_primary) <span class="sc">+</span></span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> pal_secondary) <span class="sc">+</span></span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> F, <span class="at">colour =</span> F) <span class="sc">+</span></span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">2.3</span>))) <span class="sc">+</span></span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Median, 80% and 90%; manual bandwidth; with points; coloured lines&quot;</span>)  </span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-66-1.svg" /><!-- --></p>
>>>>>>> e1d0d20ebc8c277bd98448a6ed43c802b1ff6638
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
